---
title: "RNAhybrid_complete"
author: "Nikita Verheyden"
date: "2023-04-26"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# setup

dir
```{r dir}
# home
#setwd("D:/Krueger_Lab/Publications/miR181_paper/Figure2/RNAhybrid")

# work
setwd("Z:/Personen/Nikita/Publications/miR181_paper/Figure2/RNAhybrid")
```

## packages
```{r packages}
#home
#source("D:/Krueger_Lab/Publications/miR181_paper_v21022023/Figure_theme/theme_paper.R")
#work
source("Z:/Personen/Nikita/Publications/miR181_paper_v21022023/Figure_theme/theme_paper.R")

library(BSgenome.Mmusculus.UCSC.mm10)
library(dplyr)
library(ggplot2)
library(circlize)
library(ComplexHeatmap)
library(seqinr)
library(GenomicRanges)
library(stringr)
```

## data

```{r data}
#home
#f2bs <- readRDS("D:/Krueger_Lab/Publications/miR181_paper/Figure2/MRE_bound_gene_and_bound_region/mir181_bs_afterFigure2B.rds")
#work
f2bs <- readRDS("Z:/Personen/Nikita/Publications/miR181_paper/Figure2/MRE_bound_gene_and_bound_region/mir181_bs_afterFigure2B.rds")

head(f2bs)

```

```{r colours}
#colours
farbeneg <- "#b4b4b4"
farbe1 <- "#0073C2FF"
farbe2 <- "#EFC000FF"
farbe3 <- "#CD534CFF"
farbe4 <- "#7AA6DCFF"
farbe5 <- "#868686FF"
farbe6 <- "#003C67FF"
farbe7 <- "#8F7700FF"
farbe8 <- "#3B3B3BFF"
farbe9 <- "#A73030FF"
farbe10 <- "#4A6990FF"
farbe11 <- "#FF6F00FF"
farbe12 <- "#C71000FF"
farbe13 <- "#008EA0FF"
farbe14 <- "#8A4198FF"
farbe15 <- "#5A9599FF"
farbe16 <- "#FF6348FF"
  
RNApcol <- "#b56504"
RNAncol <- "#027d73"
RPFpcol <- "#c4c404"
RPFncol <- "#8d0391"
```



# Get DNA sequences

```{r loadDNA}
#resize ranges

f2bsLA <- f2bs
f2bsLA$start <- f2bsLA$start -25
f2bsLA$end <- f2bsLA$end + 50
f2bsLA$n_mir181a <- as.numeric(f2bsLA$n_mir181a)


df181A <- mutate(f2bsLA, Sequence = as.character(getSeq(BSgenome.Mmusculus.UCSC.mm10, seqnames, start, end)))
df181A$rownum <- rownames(df181A)
df181A <- df181A[as.numeric(df181A$n_mir181a) > 0,]

#and turn T into Us
df181A$Sequence <- gsub('T', 'U', df181A$Sequence)

head(df181A)

```

# find seed
```{r seed}
#find both seeds
seed1 <- df181A %>% filter(str_detect(Sequence, "GAAUGU"))
seed2 <- df181A %>% filter(str_detect(Sequence, "GAUUGU"))

#combine
seedm <- rbind(seed1, seed2)
#remove duplicates
seedm <- seedm[!duplicated(seedm$rownum),]
#remove NAs in gene name
seedm <- seedm[!is.na(seedm$geneName),]

head(seedm)
```


# Write to .fasta

this is deactivated for now because we only need it once right now
just remove the eval if needed

```{r writefasta, eval=F}
candgeneNameA <- as.list(seedm$geneName)
candrnameA <- as.list(seedm$rownum)
condgeneSeqA <- as.list(seedm$Sequence) 


#change to output directory

setwd("D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/fastafiles_complete/A")

for (i in 1:length(candgeneNameA)) {
  write.fasta(condgeneSeqA[i],candrnameA[i],paste(candrnameA[i], candgeneNameA[i], "miR_181a", 'fasta', sep = '.'))
}

```

```{r reimport}
Personalized_Reader <- function(lambda){
 read.table(lambda, sep = ":") %>% select(V1, V5, V6, V7, V10, V11)}

#remove NA file...I just dont get it....where is it coming from?

#File lists
reslistA <- list.files(path = "D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/results_mir181a_complete", full.names=TRUE)



#import
myfilelistA <- lapply(reslistA, Personalized_Reader)


resframeA <- bind_rows(myfilelistA)



#colnames
colnames(resframeA) <- c("rownumber", "mfs", "pvalue", "start_position", "binding_bases", "non_binding_bases")

resframeA[is.na(resframeA$non_binding_bases),"non_binding_bases"] <- "                       "



head(resframeA)
```


## merge with original df

```{r merge}
# make seperate objects for each mature mirna just to see if they are much different

seedm$rownumber <- as.character(seedm$rownum)
resframeA$rownumber <- as.character(resframeA$rownumber)



bsseqHA <- left_join(seedm, resframeA, by="rownumber")

head(bsseqHA)

```

# -------------------------------------------------------------------continue here

## Process data (remove gaps)

Due to the loops in the mRNA there are additional spaces in the mirna. We only want the binding and non binding bases of hte mirna in te correct order.
For that we will remove all gaps that origin in the mRNA loops.
```{r remove_loop}
#binding and non binding bases as characters in a list
Alistbb <- strsplit(resframeA$binding_bases,"")
Alistnb <- strsplit(resframeA$non_binding_bases,"")

#combine the two lists
Alist <- Map(cbind, Alistbb, Alistnb)
Alist <- lapply(Alist, as.data.frame)

#remove all empty rows (mRNA loops)
Alist0 <- lapply(Alist, function(x){
  x[!(x[,1]== " " & x[,2] == " "),]
})

#rewrite as characters
AlistF <- lapply(Alist0, function(x){
  paste(x[,1],  collapse = '')
})


#Attach lists back onto original data.frame as new column
resframeA$binding_nospace <-unlist(AlistF)
head(resframeA$binding_nospace)



```

# Transform into Numbers

## add 0s

replace all gaps with 0 and all letters with 1
```{r nummerify}
#0
resframeA$binding_nospace <- chartr(" ", "0", resframeA$binding_nospace)

#1
resframeA$binding_nospace <- mgsub::mgsub(resframeA$binding_nospace, c("A", "U", "C", "G"), c(rep("1", 4)))

head(resframeA)


```



## seperate into columns

for each base make 1 column so it can be added and also put into a heatmap
```{r heatframes}
#for the heatmap with every binding site
heatframeA <- do.call(rbind.data.frame, strsplit(resframeA$binding_nospace,""))
heatframeA <- sapply( heatframeA, as.numeric )
colnames(heatframeA) <- c(23:1)
rownames(heatframeA) <- resframeA[,1]
head(heatframeA)


#reverse column order
heatframeA <-heatframeA[,23:1]
```



# Heatmap

Colours
```{r hm_colours}
hmcols1 <- c("white", "black")
hmcols2 <- colorRamp2(c(-2, 2), c("white", "red"))

```

## Heatmap of all the single reads

make heatmap without column clustering but with row clustering
```{r allmap,  fig.dim = c(7, 5)}
set.seed(123)

HMA <- Heatmap(heatframeA, cluster_columns = F, col = hmcols1, row_km = 5, show_row_names = F, show_row_dend = F, border = TRUE)

```

```{r A}
HMA
```





# session info
```{r sessioninfo}
sessionInfo()
```
