---
title: "RNAhybrid fig 2-25"
author: "Nikita Verheyden"
date: "2023-04-17"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


directory
```{r dir}
setwd("D:/Krueger_Lab/Publications/miR181_paper/Figure2/RNAhybrid")
```

```{r packages}
source("D:/Krueger_Lab/Publications/miR181_paper_v21022023/Figure_theme/theme_paper.R")
library(BSgenome.Mmusculus.UCSC.mm10)
library(dplyr)
library(ggplot2)
library(circlize)
library(ComplexHeatmap)
library(seqinr)
library(GenomicRanges)
library(rGADEM)
```

```{r data}
mir181bs <- makeGRangesFromDataFrame(readRDS("D:/Krueger_Lab/Publications/miR181_paper/Figure1/mir181_binding_sites__venn_types/mir181_bs.rds"), keep.extra.columns = T)
names(mir181bs) <- 1:length(mir181bs)
```

```{r colours}
#colours
farbeneg <- "#b4b4b4"
farbe1 <- "#0073C2FF"
farbe2 <- "#EFC000FF"
farbe3 <- "#CD534CFF"
farbe4 <- "#7AA6DCFF"
farbe5 <- "#868686FF"
farbe6 <- "#003C67FF"
farbe7 <- "#8F7700FF"
farbe8 <- "#3B3B3BFF"
farbe9 <- "#A73030FF"
farbe10 <- "#4A6990FF"
farbe11 <- "#FF6F00FF"
farbe12 <- "#C71000FF"
farbe13 <- "#008EA0FF"
farbe14 <- "#8A4198FF"
farbe15 <- "#5A9599FF"
farbe16 <- "#FF6348FF"
  
RNApcol <- "#b56504"
RNAncol <- "#027d73"
RPFpcol <- "#c4c404"
RPFncol <- "#8d0391"
```



# Get DNA sequences

```{r loadDNA}
#resize ranges

mir181bs <- resize(mir181bs, width = width(mir181bs + 25), fix = "center")
mir181bs <- resize(mir181bs, width = width(mir181bs + 25), fix = "start")


# bsGene <- resize(bsGene, width = width(bsGene + 5), fix = "end")

df181 <- mutate(as.data.frame(mir181bs), Sequence = as.character(getSeq(BSgenome.Mmusculus.UCSC.mm10, seqnames, start, end)))

#and turn T into Us
df181$Sequence <- gsub('T', 'U', df181$Sequence)

head(df181)
```

# Write to .fasta

this is deactivated for now because we only need it once right now
just remove the eval if needed

```{r writefasta, eval=F}
candgeneName <- as.list(df181$geneName)
candrname <- as.list(rownames(df181))
condgeneSeq <- as.list(df181$Sequence) 


#change to output directory
setwd("D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/fastafiles-25")

for (i in 1:length(candgeneName)) {
  write.fasta(condgeneSeq[i],candrname[i],paste(candrname[i], candgeneName[i], "miR_181", 'fasta', sep = '.'))
}

```
# Import RNAhybrid results

files that are imported here were run with maximum loop of 3 in both sequences

```{r import}
Personalized_Reader <- function(lambda){
 read.table(lambda, sep = ":") %>% select(V1, V5, V6, V7, V9)}


reslistA <- list.files(path = "D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/results_mir181a_u3-25", full.names=TRUE)
reslistB <- list.files(path = "D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/results_mir181a_u3-25", full.names=TRUE)

myfilelistA <- lapply(reslistA, Personalized_Reader)
myfilelistB <- lapply(reslistB, Personalized_Reader)

resframeA <- bind_rows(myfilelistA)
resframeB <- bind_rows(myfilelistB)


colnames(resframeA) <- c("rownumber", "mfs", "pvalue", "start_position", "binding_bases")
colnames(resframeB) <- c("rownumber", "mfs", "pvalue", "start_position", "binding_bases")

head(resframeA)
head(resframeB)
```

## merge with original df

```{r merge}
# make seperate objects for each mature mirna just to see if they are much different

df181$rownumber <- as.character(rownames(df181))
resframeA$rownumber <- as.character(resframeA$rownumber)
resframeB$rownumber <- as.character(resframeB$rownumber)



bsseqHA <- left_join(df181, resframeA, by="rownumber")
bsseqHB <- left_join(df181, resframeB, by="rownumber")

head(bsseqHA)
head(bsseqHB)

```


## adjust binding site info
```{r adjustBS}
bsseqHA$binding_bases <- chartr(" ", "N", bsseqHA$binding_bases)
bsseqHB$binding_bases <- chartr(" ", "N", bsseqHB$binding_bases)

head(bsseqHA)
head(bsseqHB)
```

# Plot pValues and mfs
```{r pval}
pmfsplotA <- ggplot(bsseqHA, aes(y=-log10(pvalue), x=mfs)) + geom_point() +
  geom_hline(yintercept=-log10(0.05))


pmfsplotA

pmfsplotB <- ggplot(bsseqHA, aes(y=-log10(pvalue), x=mfs)) + geom_point() +
  geom_hline(yintercept=-log10(0.05))

pmfsplotB
```

# find motivs

## make DNAStringSet
```{r stringset}
#make a DNA sting set as well because the motif finder requires it
b_baseSetDNAA <- DNAStringSet(chartr("U", "T", bsseqHA[bsseqHA$pvalue <= 0.05, "binding_bases"]))
b_baseSetDNAB <- DNAStringSet(chartr("U", "T", bsseqHB[bsseqHB$pvalue <= 0.05, "binding_bases"]))

head(b_baseSetDNAA)
head(b_baseSetDNAB)

```
# Run rGadem

```{r rGadem}
enrichmotA <- GADEM(b_baseSetDNAA, seed = 123)
enrichmotB <- GADEM(b_baseSetDNAB, seed = 123)
```
## miR181a
```{r mir181a}
rGADEM::plot(enrichmotA)
```
## miR181b
```{r mir181b}
rGADEM::plot(enrichmotB)
```

# Session info

```{r sessioninfo}
sessionInfo()
```