---
title: "RNAhybrid_complete_testing_c_vs_nc"
author: "Nikita Verheyden"
date: "2023-04-28"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# setup

dir
```{r dir}
# home
setwd("D:/Krueger_Lab/Publications/miR181_paper/Figure2/RNAhybrid")

# work
#setwd("Z:/Personen/Nikita/Publications/miR181_paper/Figure2/RNAhybrid")
```

## packages
```{r packages}
#home
source("D:/Krueger_Lab/Publications/miR181_paper_v21022023/Figure_theme/theme_paper.R")
#work
#source("Z:/Personen/Nikita/Publications/miR181_paper_v21022023/Figure_theme/theme_paper.R")

library(BSgenome.Mmusculus.UCSC.mm10)
library(dplyr)
library(ggplot2)
library(circlize)
library(ComplexHeatmap)
library(seqinr)
library(GenomicRanges)
library(stringr)
library(gridExtra)
```

## data

```{r data}
#home
f2bs <- readRDS("D:/Krueger_Lab/Publications/miR181_paper/Figure2/MRE_bound_gene_and_bound_region/mir181_bs_afterFigure2B.rds")
#work
#f2bs <- readRDS("Z:/Personen/Nikita/Publications/miR181_paper/Figure2/MRE_bound_gene_and_bound_region/mir181_bs_afterFigure2B.rds")

head(f2bs)

```

```{r colours}
#colours
farbeneg <- "#b4b4b4"
farbe1 <- "#0073C2FF"
farbe2 <- "#EFC000FF"
farbe3 <- "#CD534CFF"
farbe4 <- "#7AA6DCFF"
farbe5 <- "#868686FF"
farbe6 <- "#003C67FF"
farbe7 <- "#8F7700FF"
farbe8 <- "#3B3B3BFF"
farbe9 <- "#A73030FF"
farbe10 <- "#4A6990FF"
farbe11 <- "#FF6F00FF"
farbe12 <- "#C71000FF"
farbe13 <- "#008EA0FF"
farbe14 <- "#8A4198FF"
farbe15 <- "#5A9599FF"
farbe16 <- "#FF6348FF"
  
RNApcol <- "#b56504"
RNAncol <- "#027d73"
RPFpcol <- "#c4c404"
RPFncol <- "#8d0391"
```



# Get DNA sequences

```{r loadDNA}
#resize ranges

f2bsLA <- f2bs
f2bsLA$start <- f2bsLA$start -25
f2bsLA$end <- f2bsLA$end + 50
f2bsLA$n_mir181a <- as.numeric(f2bsLA$n_mir181a)


df181A <- mutate(f2bsLA, Sequence = as.character(getSeq(BSgenome.Mmusculus.UCSC.mm10, seqnames, start, end)))
df181A$rownum <- rownames(df181A)
df181A <- df181A[as.numeric(df181A$n_mir181a) > 0,]

#and turn T into Us
df181A$Sequence <- gsub('T', 'U', df181A$Sequence)

head(df181A)

```

# find seed
```{r seed}
#find both seeds
seed1 <- df181A %>% filter(str_detect(Sequence, "GAAUGU"))
seed2 <- df181A %>% filter(str_detect(Sequence, "GAUUGU"))


#remove NAs in gene name
seed1 <- seed1[!is.na(seed1$geneName),]
seed2 <- seed2[!is.na(seed2$geneName),]

head(seed1)
head(seed2)
```


# Write to .fasta

this is deactivated for now because we only need it once right now
just remove the eval if needed

```{r writefasta, eval=F}
candgeneNameAc <- as.list(seed1$geneName)
candrnameAc <- as.list(seed1$rownum)
condgeneSeqAc <- as.list(seed1$Sequence)

candgeneNameAnc <- as.list(seed2$geneName)
candrnameAnc <- as.list(seed2$rownum)
condgeneSeqAnc <- as.list(seed2$Sequence) 


#change to output directory

setwd("D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/fastafiles_complete/Ac")

for (i in 1:length(candgeneNameAc)) {
  write.fasta(condgeneSeqAc[i],candrnameAc[i],paste(candrnameAc[i], candgeneNameAc[i], "miR_181a", 'fasta', sep = '.'))
}

setwd("D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/fastafiles_complete/Anc")

for (i in 1:length(candgeneNameAnc)) {
  write.fasta(condgeneSeqAnc[i],candrnameAnc[i],paste(candrnameAnc[i], candgeneNameAnc[i], "miR_181a", 'fasta', sep = '.'))
}

```

```{r reimport}
Personalized_Reader <- function(lambda){
 read.table(lambda, sep = ":") %>% select(V1, V5, V6, V7, V10, V11)}

#remove NA file...I just dont get it....where is it coming from?

#File lists
reslistAc0 <- list.files(path = "D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/results_mir181a_complete_c0", full.names=TRUE)
reslistAc1 <- list.files(path = "D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/results_mir181a_complete_c1", full.names=TRUE)
reslistAc2 <- list.files(path = "D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/results_mir181a_complete_c2", full.names=TRUE)
reslistAc3 <- list.files(path = "D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/results_mir181a_complete_c3", full.names=TRUE)
reslistAc4 <- list.files(path = "D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/results_mir181a_complete_c4", full.names=TRUE)
reslistAc5 <- list.files(path = "D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/results_mir181a_complete_c5", full.names=TRUE)


reslistAnc0 <- list.files(path = "D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/results_mir181a_complete_nc0", full.names=TRUE)
reslistAnc1 <- list.files(path = "D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/results_mir181a_complete_nc1", full.names=TRUE)
reslistAnc2 <- list.files(path = "D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/results_mir181a_complete_nc2", full.names=TRUE)
reslistAnc3 <- list.files(path = "D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/results_mir181a_complete_nc3", full.names=TRUE)
reslistAnc4 <- list.files(path = "D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/results_mir181a_complete_nc4", full.names=TRUE)
reslistAnc5 <- list.files(path = "D:/Krueger_Lab/Publications/miR181_paper_nongithub/Figure2/RNAhybrid/results_mir181a_complete_nc5", full.names=TRUE)


#import
myfilelistAc0 <- lapply(reslistAc0, Personalized_Reader)
myfilelistAc1 <- lapply(reslistAc1, Personalized_Reader)
myfilelistAc2 <- lapply(reslistAc2, Personalized_Reader)
myfilelistAc3 <- lapply(reslistAc3, Personalized_Reader)
myfilelistAc4 <- lapply(reslistAc4, Personalized_Reader)
myfilelistAc5 <- lapply(reslistAc5, Personalized_Reader)

myfilelistAnc0 <- lapply(reslistAnc0, Personalized_Reader)
myfilelistAnc1 <- lapply(reslistAnc1, Personalized_Reader)
myfilelistAnc2 <- lapply(reslistAnc2, Personalized_Reader)
myfilelistAnc3 <- lapply(reslistAnc3, Personalized_Reader)
myfilelistAnc4 <- lapply(reslistAnc4, Personalized_Reader)
myfilelistAnc5 <- lapply(reslistAnc5, Personalized_Reader)


resframeAc0 <- bind_rows(myfilelistAc0)
resframeAc1 <- bind_rows(myfilelistAc1)
resframeAc2 <- bind_rows(myfilelistAc2)
resframeAc3 <- bind_rows(myfilelistAc3)
resframeAc4 <- bind_rows(myfilelistAc4)
resframeAc5 <- bind_rows(myfilelistAc5)

resframeAnc0 <- bind_rows(myfilelistAnc0)
resframeAnc1 <- bind_rows(myfilelistAnc1)
resframeAnc2 <- bind_rows(myfilelistAnc2)
resframeAnc3 <- bind_rows(myfilelistAnc3)
resframeAnc4 <- bind_rows(myfilelistAnc4)
resframeAnc5 <- bind_rows(myfilelistAnc5)


#colnames
colnames(resframeAc0) <- c("rownumber", "mfs", "pvalue", "start_position", "binding_bases", "non_binding_bases")
colnames(resframeAc1) <- c("rownumber", "mfs", "pvalue", "start_position", "binding_bases", "non_binding_bases")
colnames(resframeAc2) <- c("rownumber", "mfs", "pvalue", "start_position", "binding_bases", "non_binding_bases")
colnames(resframeAc3) <- c("rownumber", "mfs", "pvalue", "start_position", "binding_bases", "non_binding_bases")
colnames(resframeAc4) <- c("rownumber", "mfs", "pvalue", "start_position", "binding_bases", "non_binding_bases")
colnames(resframeAc5) <- c("rownumber", "mfs", "pvalue", "start_position", "binding_bases", "non_binding_bases")

colnames(resframeAnc0) <- c("rownumber", "mfs", "pvalue", "start_position", "binding_bases", "non_binding_bases")
colnames(resframeAnc1) <- c("rownumber", "mfs", "pvalue", "start_position", "binding_bases", "non_binding_bases")
colnames(resframeAnc2) <- c("rownumber", "mfs", "pvalue", "start_position", "binding_bases", "non_binding_bases")
colnames(resframeAnc3) <- c("rownumber", "mfs", "pvalue", "start_position", "binding_bases", "non_binding_bases")
colnames(resframeAnc4) <- c("rownumber", "mfs", "pvalue", "start_position", "binding_bases", "non_binding_bases")
colnames(resframeAnc5) <- c("rownumber", "mfs", "pvalue", "start_position", "binding_bases", "non_binding_bases")


resframeAc0[is.na(resframeAc0$non_binding_bases),"non_binding_bases"] <- "                       "
resframeAc1[is.na(resframeAc1$non_binding_bases),"non_binding_bases"] <- "                       "
resframeAc2[is.na(resframeAc2$non_binding_bases),"non_binding_bases"] <- "                       "
resframeAc3[is.na(resframeAc3$non_binding_bases),"non_binding_bases"] <- "                       "
resframeAc4[is.na(resframeAc4$non_binding_bases),"non_binding_bases"] <- "                       "
resframeAc5[is.na(resframeAc5$non_binding_bases),"non_binding_bases"] <- "                       "

resframeAnc0[is.na(resframeAnc0$non_binding_bases),"non_binding_bases"] <- "                       "
resframeAnc1[is.na(resframeAnc1$non_binding_bases),"non_binding_bases"] <- "                       "
resframeAnc2[is.na(resframeAnc2$non_binding_bases),"non_binding_bases"] <- "                       "
resframeAnc3[is.na(resframeAnc3$non_binding_bases),"non_binding_bases"] <- "                       "
resframeAnc4[is.na(resframeAnc4$non_binding_bases),"non_binding_bases"] <- "                       "
resframeAnc5[is.na(resframeAnc5$non_binding_bases),"non_binding_bases"] <- "                       "


```


## merge with original df

```{r merge}
# make seperate objects for each mature mirna just to see if they are much different

seed1$rownumber <- as.character(seed1$rownum)
seed2$rownumber <- as.character(seed2$rownum)

resframeAc0$rownumber <- as.character(resframeAc0$rownumber)
resframeAc1$rownumber <- as.character(resframeAc1$rownumber)
resframeAc2$rownumber <- as.character(resframeAc2$rownumber)
resframeAc3$rownumber <- as.character(resframeAc3$rownumber)
resframeAc4$rownumber <- as.character(resframeAc4$rownumber)
resframeAc5$rownumber <- as.character(resframeAc5$rownumber)


resframeAnc0$rownumber <- as.character(resframeAnc0$rownumber)
resframeAnc1$rownumber <- as.character(resframeAnc1$rownumber)
resframeAnc2$rownumber <- as.character(resframeAnc2$rownumber)
resframeAnc3$rownumber <- as.character(resframeAnc3$rownumber)
resframeAnc4$rownumber <- as.character(resframeAnc4$rownumber)
resframeAnc5$rownumber <- as.character(resframeAnc5$rownumber)



bsseqHAc0 <- left_join(seed1, resframeAc0, by="rownumber")
bsseqHAc1 <- left_join(seed1, resframeAc1, by="rownumber")
bsseqHAc2 <- left_join(seed1, resframeAc2, by="rownumber")
bsseqHAc3 <- left_join(seed1, resframeAc3, by="rownumber")
bsseqHAc4 <- left_join(seed1, resframeAc4, by="rownumber")
bsseqHAc5 <- left_join(seed1, resframeAc5, by="rownumber")

bsseqHAnc0 <- left_join(seed2, resframeAnc0, by="rownumber")
bsseqHAnc1 <- left_join(seed2, resframeAnc1, by="rownumber")
bsseqHAnc2 <- left_join(seed2, resframeAnc2, by="rownumber")
bsseqHAnc3 <- left_join(seed2, resframeAnc3, by="rownumber")
bsseqHAnc4 <- left_join(seed2, resframeAnc4, by="rownumber")
bsseqHAnc5 <- left_join(seed2, resframeAnc5, by="rownumber")


```


## Process data (remove gaps)

Due to the loops in the mRNA there are additional spaces in the mirna. We only want the binding and non binding bases of hte mirna in te correct order.
For that we will remove all gaps that origin in the mRNA loops.
```{r remove_loop}
#binding and non binding bases as characters in a list
Alistbbc0 <- strsplit(resframeAc0$binding_bases,"")
Alistbbc1 <- strsplit(resframeAc1$binding_bases,"")
Alistbbc2 <- strsplit(resframeAc2$binding_bases,"")
Alistbbc3 <- strsplit(resframeAc3$binding_bases,"")
Alistbbc4 <- strsplit(resframeAc4$binding_bases,"")
Alistbbc5 <- strsplit(resframeAc5$binding_bases,"")

Alistbbnc0 <- strsplit(resframeAnc0$binding_bases,"")
Alistbbnc1 <- strsplit(resframeAnc1$binding_bases,"")
Alistbbnc2 <- strsplit(resframeAnc2$binding_bases,"")
Alistbbnc3 <- strsplit(resframeAnc3$binding_bases,"")
Alistbbnc4 <- strsplit(resframeAnc4$binding_bases,"")
Alistbbnc5 <- strsplit(resframeAnc5$binding_bases,"")

Alistnbc0 <- strsplit(resframeAc0$non_binding_bases,"")
Alistnbc1 <- strsplit(resframeAc1$non_binding_bases,"")
Alistnbc2 <- strsplit(resframeAc2$non_binding_bases,"")
Alistnbc3 <- strsplit(resframeAc3$non_binding_bases,"")
Alistnbc4 <- strsplit(resframeAc4$non_binding_bases,"")
Alistnbc5 <- strsplit(resframeAc5$non_binding_bases,"")


Alistnbnc0 <- strsplit(resframeAnc0$non_binding_bases,"")
Alistnbnc1 <- strsplit(resframeAnc1$non_binding_bases,"")
Alistnbnc2 <- strsplit(resframeAnc2$non_binding_bases,"")
Alistnbnc3 <- strsplit(resframeAnc3$non_binding_bases,"")
Alistnbnc4 <- strsplit(resframeAnc4$non_binding_bases,"")
Alistnbnc5 <- strsplit(resframeAnc5$non_binding_bases,"")


#combine the two lists
Alistc0 <- Map(cbind, Alistbbc0, Alistnbc0)
Alistc1 <- Map(cbind, Alistbbc1, Alistnbc1)
Alistc2 <- Map(cbind, Alistbbc2, Alistnbc2)
Alistc3 <- Map(cbind, Alistbbc3, Alistnbc3)
Alistc4 <- Map(cbind, Alistbbc4, Alistnbc4)
Alistc5 <- Map(cbind, Alistbbc5, Alistnbc5)

Alistnc0 <- Map(cbind, Alistbbnc0, Alistnbnc0)
Alistnc1 <- Map(cbind, Alistbbnc1, Alistnbnc1)
Alistnc2 <- Map(cbind, Alistbbnc2, Alistnbnc2)
Alistnc3 <- Map(cbind, Alistbbnc3, Alistnbnc3)
Alistnc4 <- Map(cbind, Alistbbnc4, Alistnbnc4)
Alistnc5 <- Map(cbind, Alistbbnc5, Alistnbnc5)

Alistc0 <- lapply(Alistc0, as.data.frame)
Alistc1 <- lapply(Alistc1, as.data.frame)
Alistc2 <- lapply(Alistc2, as.data.frame)
Alistc3 <- lapply(Alistc3, as.data.frame)
Alistc4 <- lapply(Alistc4, as.data.frame)
Alistc5 <- lapply(Alistc5, as.data.frame)

Alistnc0 <- lapply(Alistnc0, as.data.frame)
Alistnc1 <- lapply(Alistnc1, as.data.frame)
Alistnc2 <- lapply(Alistnc2, as.data.frame)
Alistnc3 <- lapply(Alistnc3, as.data.frame)
Alistnc4 <- lapply(Alistnc4, as.data.frame)
Alistnc5 <- lapply(Alistnc5, as.data.frame)

#remove all empty rows (mRNA loops)
Alist0c0 <- lapply(Alistc0, function(x){
  x[!(x[,1]== " " & x[,2] == " "),]
})
Alist0c1 <- lapply(Alistc1, function(x){
  x[!(x[,1]== " " & x[,2] == " "),]
})
Alist0c2 <- lapply(Alistc2, function(x){
  x[!(x[,1]== " " & x[,2] == " "),]
})
Alist0c3 <- lapply(Alistc3, function(x){
  x[!(x[,1]== " " & x[,2] == " "),]
})
Alist0c4 <- lapply(Alistc4, function(x){
  x[!(x[,1]== " " & x[,2] == " "),]
})
Alist0c5 <- lapply(Alistc5, function(x){
  x[!(x[,1]== " " & x[,2] == " "),]
})

Alist0nc0 <- lapply(Alistnc0, function(x){
  x[!(x[,1]== " " & x[,2] == " "),]
})
Alist0nc1 <- lapply(Alistnc1, function(x){
  x[!(x[,1]== " " & x[,2] == " "),]
})
Alist0nc2 <- lapply(Alistnc2, function(x){
  x[!(x[,1]== " " & x[,2] == " "),]
})
Alist0nc3 <- lapply(Alistnc3, function(x){
  x[!(x[,1]== " " & x[,2] == " "),]
})
Alist0nc4 <- lapply(Alistnc4, function(x){
  x[!(x[,1]== " " & x[,2] == " "),]
})
Alist0nc5 <- lapply(Alistnc5, function(x){
  x[!(x[,1]== " " & x[,2] == " "),]
})

#rewrite as characters
AlistFc0 <- lapply(Alist0c0, function(x){
  paste(x[,1],  collapse = '')
})
AlistFc1 <- lapply(Alist0c1, function(x){
  paste(x[,1],  collapse = '')
})
AlistFc2 <- lapply(Alist0c2, function(x){
  paste(x[,1],  collapse = '')
})
AlistFc3 <- lapply(Alist0c3, function(x){
  paste(x[,1],  collapse = '')
})
AlistFc4 <- lapply(Alist0c4, function(x){
  paste(x[,1],  collapse = '')
})
AlistFc5 <- lapply(Alist0c5, function(x){
  paste(x[,1],  collapse = '')
})

AlistFnc0 <- lapply(Alist0nc0, function(x){
  paste(x[,1],  collapse = '')
})
AlistFnc1 <- lapply(Alist0nc1, function(x){
  paste(x[,1],  collapse = '')
})
AlistFnc2 <- lapply(Alist0nc2, function(x){
  paste(x[,1],  collapse = '')
})
AlistFnc3 <- lapply(Alist0nc3, function(x){
  paste(x[,1],  collapse = '')
})
AlistFnc4 <- lapply(Alist0nc4, function(x){
  paste(x[,1],  collapse = '')
})
AlistFnc5 <- lapply(Alist0nc5, function(x){
  paste(x[,1],  collapse = '')
})


#Attach lists back onto original data.frame as new column
resframeAc0$binding_nospace <-unlist(AlistFc0)
resframeAc1$binding_nospace <-unlist(AlistFc1)
resframeAc2$binding_nospace <-unlist(AlistFc2)
resframeAc3$binding_nospace <-unlist(AlistFc3)
resframeAc4$binding_nospace <-unlist(AlistFc4)
resframeAc5$binding_nospace <-unlist(AlistFc5)

resframeAnc0$binding_nospace <-unlist(AlistFnc0)
resframeAnc1$binding_nospace <-unlist(AlistFnc1)
resframeAnc2$binding_nospace <-unlist(AlistFnc2)
resframeAnc3$binding_nospace <-unlist(AlistFnc3)
resframeAnc4$binding_nospace <-unlist(AlistFnc4)
resframeAnc5$binding_nospace <-unlist(AlistFnc5)


```

# Transform into Numbers

## add 0s

replace all gaps with 0 and all letters with 1
```{r nummerify}
#0
resframeAc0$binding_nospace <- chartr(" ", "0", resframeAc0$binding_nospace)
resframeAc1$binding_nospace <- chartr(" ", "0", resframeAc1$binding_nospace)
resframeAc2$binding_nospace <- chartr(" ", "0", resframeAc2$binding_nospace)
resframeAc3$binding_nospace <- chartr(" ", "0", resframeAc3$binding_nospace)
resframeAc4$binding_nospace <- chartr(" ", "0", resframeAc4$binding_nospace)
resframeAc5$binding_nospace <- chartr(" ", "0", resframeAc5$binding_nospace)

resframeAnc0$binding_nospace <- chartr(" ", "0", resframeAnc0$binding_nospace)
resframeAnc1$binding_nospace <- chartr(" ", "0", resframeAnc1$binding_nospace)
resframeAnc2$binding_nospace <- chartr(" ", "0", resframeAnc2$binding_nospace)
resframeAnc3$binding_nospace <- chartr(" ", "0", resframeAnc3$binding_nospace)
resframeAnc4$binding_nospace <- chartr(" ", "0", resframeAnc4$binding_nospace)
resframeAnc5$binding_nospace <- chartr(" ", "0", resframeAnc5$binding_nospace)


#1
resframeAc0$binding_nospace <- mgsub::mgsub(resframeAc0$binding_nospace, c("A", "U", "C", "G"), c(rep("1", 4)))
resframeAc1$binding_nospace <- mgsub::mgsub(resframeAc1$binding_nospace, c("A", "U", "C", "G"), c(rep("1", 4)))
resframeAc2$binding_nospace <- mgsub::mgsub(resframeAc2$binding_nospace, c("A", "U", "C", "G"), c(rep("1", 4)))
resframeAc3$binding_nospace <- mgsub::mgsub(resframeAc3$binding_nospace, c("A", "U", "C", "G"), c(rep("1", 4)))
resframeAc4$binding_nospace <- mgsub::mgsub(resframeAc4$binding_nospace, c("A", "U", "C", "G"), c(rep("1", 4)))
resframeAc5$binding_nospace <- mgsub::mgsub(resframeAc5$binding_nospace, c("A", "U", "C", "G"), c(rep("1", 4)))

resframeAnc0$binding_nospace <- mgsub::mgsub(resframeAnc0$binding_nospace, c("A", "U", "C", "G"), c(rep("1", 4)))
resframeAnc1$binding_nospace <- mgsub::mgsub(resframeAnc1$binding_nospace, c("A", "U", "C", "G"), c(rep("1", 4)))
resframeAnc2$binding_nospace <- mgsub::mgsub(resframeAnc2$binding_nospace, c("A", "U", "C", "G"), c(rep("1", 4)))
resframeAnc3$binding_nospace <- mgsub::mgsub(resframeAnc3$binding_nospace, c("A", "U", "C", "G"), c(rep("1", 4)))
resframeAnc4$binding_nospace <- mgsub::mgsub(resframeAnc4$binding_nospace, c("A", "U", "C", "G"), c(rep("1", 4)))
resframeAnc5$binding_nospace <- mgsub::mgsub(resframeAnc5$binding_nospace, c("A", "U", "C", "G"), c(rep("1", 4)))


```



## seperate into columns

for each base make 1 column so it can be added and also put into a heatmap
```{r heatframes}
#for the heatmap with every binding site
heatframeAc0 <- do.call(rbind.data.frame, strsplit(resframeAc0$binding_nospace,""))
heatframeAc1 <- do.call(rbind.data.frame, strsplit(resframeAc1$binding_nospace,""))
heatframeAc2 <- do.call(rbind.data.frame, strsplit(resframeAc2$binding_nospace,""))
heatframeAc3 <- do.call(rbind.data.frame, strsplit(resframeAc3$binding_nospace,""))
heatframeAc4 <- do.call(rbind.data.frame, strsplit(resframeAc4$binding_nospace,""))
heatframeAc5 <- do.call(rbind.data.frame, strsplit(resframeAc5$binding_nospace,""))

heatframeAnc0 <- do.call(rbind.data.frame, strsplit(resframeAnc0$binding_nospace,""))
heatframeAnc1 <- do.call(rbind.data.frame, strsplit(resframeAnc1$binding_nospace,""))
heatframeAnc2 <- do.call(rbind.data.frame, strsplit(resframeAnc2$binding_nospace,""))
heatframeAnc3 <- do.call(rbind.data.frame, strsplit(resframeAnc3$binding_nospace,""))
heatframeAnc4 <- do.call(rbind.data.frame, strsplit(resframeAnc4$binding_nospace,""))
heatframeAnc5 <- do.call(rbind.data.frame, strsplit(resframeAnc5$binding_nospace,""))

heatframeAc0 <- sapply( heatframeAc0, as.numeric )
heatframeAc1 <- sapply( heatframeAc1, as.numeric )
heatframeAc2 <- sapply( heatframeAc2, as.numeric )
heatframeAc3 <- sapply( heatframeAc3, as.numeric )
heatframeAc4 <- sapply( heatframeAc4, as.numeric )
heatframeAc5 <- sapply( heatframeAc5, as.numeric )

heatframeAnc0 <- sapply( heatframeAnc0, as.numeric )
heatframeAnc1 <- sapply( heatframeAnc1, as.numeric )
heatframeAnc2 <- sapply( heatframeAnc2, as.numeric )
heatframeAnc3 <- sapply( heatframeAnc3, as.numeric )
heatframeAnc4 <- sapply( heatframeAnc4, as.numeric )
heatframeAnc5 <- sapply( heatframeAnc5, as.numeric )


colnames(heatframeAc0) <- c(23:1)
colnames(heatframeAc1) <- c(23:1)
colnames(heatframeAc2) <- c(23:1)
colnames(heatframeAc3) <- c(23:1)
colnames(heatframeAc4) <- c(23:1)
colnames(heatframeAc5) <- c(23:1)

colnames(heatframeAnc0) <- c(23:1)
colnames(heatframeAnc1) <- c(23:1)
colnames(heatframeAnc2) <- c(23:1)
colnames(heatframeAnc3) <- c(23:1)
colnames(heatframeAnc4) <- c(23:1)
colnames(heatframeAnc5) <- c(23:1)

rownames(heatframeAc0) <- resframeAc0[,1]
rownames(heatframeAc1) <- resframeAc1[,1]
rownames(heatframeAc2) <- resframeAc2[,1]
rownames(heatframeAc3) <- resframeAc3[,1]
rownames(heatframeAc4) <- resframeAc4[,1]
rownames(heatframeAc5) <- resframeAc5[,1]

rownames(heatframeAnc0) <- resframeAnc0[,1]
rownames(heatframeAnc1) <- resframeAnc1[,1]
rownames(heatframeAnc2) <- resframeAnc2[,1]
rownames(heatframeAnc3) <- resframeAnc3[,1]
rownames(heatframeAnc4) <- resframeAnc4[,1]
rownames(heatframeAnc5) <- resframeAnc5[,1]

#reverse column order
heatframeAc0 <-heatframeAc0[,23:1]
heatframeAc1 <-heatframeAc1[,23:1]
heatframeAc2 <-heatframeAc2[,23:1]
heatframeAc3 <-heatframeAc3[,23:1]
heatframeAc4 <-heatframeAc4[,23:1]
heatframeAc5 <-heatframeAc5[,23:1]

heatframeAnc0 <-heatframeAnc0[,23:1]
heatframeAnc1 <-heatframeAnc1[,23:1]
heatframeAnc2 <-heatframeAnc2[,23:1]
heatframeAnc3 <-heatframeAnc3[,23:1]
heatframeAnc4 <-heatframeAnc4[,23:1]
heatframeAnc5 <-heatframeAnc5[,23:1]

```



# Heatmap

Colours
```{r hm_colours}
hmcols1 <- c("white", "black")

```

## Heatmap of all the single reads

make heatmap without column clustering but with row clustering
```{r allmap,  fig.dim = c(7, 5)}
set.seed(123)
HMAc0 <- Heatmap(heatframeAc0, cluster_columns = F, col = hmcols1, row_km = 5, show_row_names = F, show_row_dend = F, border = TRUE, column_title = "canonical seed, loop=0")
set.seed(123)
HMAc1 <- Heatmap(heatframeAc1, cluster_columns = F, col = hmcols1, row_km = 5, show_row_names = F, show_row_dend = F, border = TRUE, column_title = "canonical seed, loop=1")
set.seed(123)
HMAc2 <- Heatmap(heatframeAc2, cluster_columns = F, col = hmcols1, row_km = 5, show_row_names = F, show_row_dend = F, border = TRUE, column_title = "canonical seed, loop=2")
set.seed(123)
HMAc3 <- Heatmap(heatframeAc3, cluster_columns = F, col = hmcols1, row_km = 5, show_row_names = F, show_row_dend = F, border = TRUE, column_title = "canonical seed, loop=3")
set.seed(123)
HMAc4 <- Heatmap(heatframeAc4, cluster_columns = F, col = hmcols1, row_km = 5, show_row_names = F, show_row_dend = F, border = TRUE, column_title = "canonical seed, loop=4")
set.seed(123)
HMAc5 <- Heatmap(heatframeAc5, cluster_columns = F, col = hmcols1, row_km = 5, show_row_names = F, show_row_dend = F, border = TRUE, column_title = "canonical seed, loop=5")


set.seed(123)
HMAnc0 <- Heatmap(heatframeAnc0, cluster_columns = F, col = hmcols1, row_km = 5, show_row_names = F, show_row_dend = F, border = TRUE, column_title = "wobble seed, loop=0")
set.seed(123)
HMAnc1 <- Heatmap(heatframeAnc1, cluster_columns = F, col = hmcols1, row_km = 5, show_row_names = F, show_row_dend = F, border = TRUE, column_title = "wobble seed, loop=1")
set.seed(123)
HMAnc2 <- Heatmap(heatframeAnc2, cluster_columns = F, col = hmcols1, row_km = 5, show_row_names = F, show_row_dend = F, border = TRUE, column_title = "wobble seed, loop=2")
set.seed(123)
HMAnc3 <- Heatmap(heatframeAnc3, cluster_columns = F, col = hmcols1, row_km = 5, show_row_names = F, show_row_dend = F, border = TRUE, column_title = "wobble seed, loop=3")
set.seed(123)
HMAnc4 <- Heatmap(heatframeAnc4, cluster_columns = F, col = hmcols1, row_km = 5, show_row_names = F, show_row_dend = F, border = TRUE, column_title = "wobble seed, loop=4")
set.seed(123)
HMAnc5 <- Heatmap(heatframeAnc5, cluster_columns = F, col = hmcols1, row_km = 5, show_row_names = F, show_row_dend = F, border = TRUE, column_title = "wobble seed, loop=5")
```

```{r A}
# canonical seed, loop=0
HMAc0
# canonical seed, loop=1
HMAc1
# canonical seed, loop=2
HMAc2
# canonical seed, loop=3
HMAc3
# canonical seed, loop=4
HMAc4
# canonical seed, loop=5
HMAc5

# wobble seed, loop=0
HMAnc0
# wobble seed, loop=1
HMAnc1
# wobble seed, loop=2
HMAnc2
# wobble seed, loop=3
HMAnc3
# wobble seed, loop=4
HMAnc4
# wobble seed, loop=5
HMAnc5

```





# session info
```{r sessioninfo}
sessionInfo()
```
