---
title: "Seed motifs on transcripts"
author: "Melina Klostermann"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  BiocStyle::html_document:
    toc_float: TRUE
    code_folding: hide
    toc: TRUE
    number_sections: yes
    fig_caption: yes
  # pdf_document:
  #   fig_caption: true
  #   fig_crop: true
  #   toc: true
  #   toc_depth: 1
  #   number_sections: true
---

```{r setup, include=FALSE}
require("knitr")
knitr::opts_chunk$set(warning=FALSE, message=FALSE, cache=F, cache.lazy = FALSE)
tidy.opts=list(width.cutoff=80,tidy=TRUE, echo=FALSE)
```

# Libraries and settings

```{r libraries}
# ----------------------------------------
# libraries
# ----------------------------------------
library(tidyverse)
library(GenomicRanges)
library(colorspace)
library(gghalves)
library(BSgenome.Mmusculus.UCSC.mm10)
library(Biostrings)
library(plyranges)

here <- here::here()

source(paste0(here,"/Supporting_scripts/themes/theme_paper.R"))

# ----------------------------------------
# settings
# ----------------------------------------

out <- paste0(here,"/Figure5/01_Seed_motifes/")
source(paste0(here,"/Supporting_scripts/themes/theme_paper.R"))
source(paste0(here,"/Supporting_scripts/themes/CustomThemes.R"))

```

# What was done?

- I count different versions of the miR181 seed in the 200nt before and after mir181 binding sites.

- I use the seed 6mer, 7mers with one adjecent nt, and a 8mer with two adjecent nts.



# Files
```{r}
# -----------------------------------
# transcript sequences
# ------------------------------------
transcript_fasta <- readDNAStringSet("/Users/melinaklostermann/Documents/projects/anno/gencodevM23/gencode.vM23.transcripts.fa.gz") %>% RNAStringSet()

transcript_anno_meta <- names(transcript_fasta) 
transcript_anno_meta <- data.frame(all = transcript_anno_meta) %>%
  tidyr::separate(., col = all,
                  into = c("transcript_id", "gene_id", "a", "b", "isoform_name", "gene_name", "entrez_gene_id", "gene_type"), sep = "\\|")

names_transcript_fasta <- sub("\\..*", "", transcript_anno_meta$transcript_id)

# add N in beginning in end to not run out of transcripts when search motif
n200 <- c(rep("N",200)) %>% 
  paste(., collapse = "") %>%
  RNAStringSet()

transcript_fasta <- xscat(n200, transcript_fasta, n200)
names(transcript_fasta) <- names_transcript_fasta

transcript_fasta_df <- data.frame(tx_name = names(transcript_fasta), width = width(transcript_fasta))

# ----------------------------------------
# MREs
# ----------------------------------------

mir181_bs <- readRDS(paste0(here,"/Figure2/02_assign_transcripts/mir181_bs_on_transcripts.rds"))

mir_crosslinks <- readRDS("/Users/melinaklostermann/Documents/projects/AgoCLIP_miR181/xx_down_stream_R/01_Characterisation_of_chimeric_reads/out/mir_crosslinks_per_cond.rds")

load("/Users/melinaklostermann/Documents/projects/AgoCLIP_miR181/xx_down_stream_R/02_BS_definition_WT_rerun_Mirko/out/bsTranscript.rda")

# move bs annotation because transcripts got 200N in beginning
# Achtung! this needs to be shifted back in the end of the script!!!
mir181_bs <- makeGRangesFromDataFrame(mir181_bs, keep.extra.columns = T) %>% 
  shift(., 200) %>%
  as.data.frame(.)

mir181_enriched_set <- mir181_bs %>% 
  subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched"))

nrow(mir181_enriched_set)


mir181_bs_gene_anno <- readRDS("/Users/melinaklostermann/Documents/projects/AgoCLIP_miR181/R_github/miR181_paper/Figure2/01_MRE_bound_gene_and_bound_region/mir181_bs_afterFigure2B.rds")


# mir181_bs%>% subset(mir181BS_ID == 7129)
# 
# mir181_bs_gene_anno %>% subset(mir181BS_ID == 7129)
# 
# 38040667+18

```

# XSTREME de novo motif discovery

```{r include=TRUE, eval=FALSE}
# get seqeunce 200nt around binding sites

mir181_bs_200_both_sides <- as.data.frame(mir181_enriched_set) %>%
  left_join(transcript_fasta_df, by= c(seqnames = "tx_name"), suffix = c(".bs", ".tx")) %>% 
  mutate(end = end + 197, start = start -197) %>%
  dplyr::filter((end <  width.tx)  & (start > 0)) %>%
  makeGRangesFromDataFrame(., keep.extra.columns = T) 

mir181_bs_200_both_sides_seq <- getSeq(mir181_bs_200_both_sides, x = transcript_fasta) %>% 
  RNAStringSet()

names(mir181_bs_200_both_sides_seq) <- 1:NROW(mir181_bs_200_both_sides_seq)

# write fasta file for XSTREME
writeXStringSet(mir181_bs_200_both_sides_seq, filepath = paste0(out, "mirBS_200_both_sides_transcripts.fasta" ))

```


XSTREME is executed on the fasta file from above via the MEME SUITE webpage (https://meme-suite.org/meme/tools/xstreme) with the following parameters:

- E-value <= 0.05
- Width 5-10
- background: model control sequences
- STREME limit: Number of motifes = 20
- MEME options: Default E-value, Zero or one occurence per sequence
- SEA: 	Output the matching sequences in a TSV file


```{r}
# plot logo
motif <- read.table("/Users/melinaklostermann/Documents/projects/AgoCLIP_miR181/R_github/miR181_paper/Figure2/03_Seed_motifes/appXSTREME_5.5.316890727386511624164212/xstreme_motif1.txt")

colnames(motif) <- c("A", "C", "G", "U")
motif <- t(motif)

logo <- ggseqlogo::ggseqlogo(motif)

logo

ggsave(logo, filename = paste0(out, "xstreme_logo.pdf"), height = 6, width = 8, units = "cm")

# plot all motifs found
xstreame <- read_table("/Users/melinaklostermann/Documents/projects/AgoCLIP_miR181/R_github/miR181_paper/Figure2/03_Seed_motifes/appXSTREME_5.5.316890727386511624164212/xstreme.tsv")
xstreame <- xstreame[1:23,]
xstreame$EVALUE <- as.numeric(xstreame$EVALUE)

# ggplot(xstreame, aes(x = as.numeric(SITES), y = EVALUE, label = CONSENSUS))+
#   geom_point()+
#   ggrepel::geom_label_repel()


```

```{r}
#################
# the mir181 seed and interesting seed variations
#################

seed_8mer <- "UGAAUGUA"
seed_7mer_m8 <- "UGAAUGU"
seed_7mer_a1 <- "GAAUGUA"
seed_6mer <- "GAAUGU"
seed_6mer_wobble <- "GAUUGU"
seed_8mer_wobble <- "UGAUUGUA"
seed_7mer_m8_wobble <- "UGAUUGU"
seed_7mer_a1_wobble <- "GAUUGUA"

seed_alt_38 <- "UGAAUG"
seed_alt_38_wobble <- "UGAUUG"


# make a list of all seeds
seed_list <- list(seed_8mer, seed_7mer_m8, seed_7mer_a1,seed_6mer, seed_6mer_wobble, seed_8mer_wobble, seed_7mer_m8_wobble, seed_7mer_a1_wobble, seed_alt_38, seed_alt_38_wobble  )

seed_names_list <- list( "seed_8mer",  "seed_7mer_m8", "seed_7mer_a1",  "seed_6mer", "seed_6mer_wobble", "seed_8mer_wobble", "seed_7mer_m8_wobble", "seed_7mer_a1_wobble", "seed_alt_38", "seed_alt_38_wobble" )

# hierarchy order, to decide which seed to use if several ar present
seed_importance_order <- c("seed_8mer",  "seed_7mer_m8", "seed_7mer_a1",  "seed_6mer", "seed_alt_38", "seed_alt_38_wobble")

```

# Seed position and distribution

## 200nt after the binding site

```{r}
#######################
# get seed in 200er window
#######################

mir181_bs_200down <- as.data.frame(mir181_enriched_set) %>%
  left_join(transcript_fasta_df, by= c(seqnames = "tx_name"), suffix = c(".bs", ".tx")) %>% 
  mutate(end = end + 200) %>%
  dplyr::filter((end <  width.tx)  & (start > 0)) %>%
  makeGRangesFromDataFrame(., keep.extra.columns =T) 

mir181_bs_200down_seq <- getSeq(mir181_bs_200down, x = transcript_fasta) %>% 
  RNAStringSet()

# count occurences of all seed variations
Seeds_200down <- lapply(seed_list, function(x) {
  vmatchPattern(pattern = x, mir181_bs_200down_seq) %>% 
  lapply(., function(x) as.data.frame(x))})

# add the binding site id to the seeds and make a df per seed type
BS_ID_list <- as.list(mir181_bs_200down$mir181BS_ID)

Seeds_200down <- map(Seeds_200down, 
                     ~map2(.x, BS_ID_list, ~mutate(.x, mir181BS_ID = .y) ) %>% 
               map_dfr(~.x))


# add the seed type names and make one df of all
Seeds_200down <- map2(Seeds_200down, seed_names_list, ~mutate(.x, seed = .y) ) %>% map_dfr(~.x)


# extract wobble positions
Seeds_1_per_BS <- Seeds_200down %>% 
  mutate(wobble = grepl("wobble", seed),
         seed = case_when(wobble  ~ substr(seed, 1, nchar(seed)-7), T ~ seed))


# order seeds by importance
Seeds_1_per_BS$seed <- factor(Seeds_1_per_BS$seed, levels = seed_importance_order )

# select 1 seed per BS --> closest seed with highest importance
Seeds_1_per_BS <- Seeds_1_per_BS %>%
   group_by(mir181BS_ID) %>%
  arrange(start, seed ) %>%
  dplyr::slice(1) %>%
  ungroup(.) 

########################
# combine the closest seed, and all found seeds to the Binding site data.frame
#######################

# add all as list column

colnames(Seeds_200down) <- c("Seeds_200down.start",
                             "Seeds_200down.end",
                             "Seeds_200down.width",
                             "mir181BS_ID",
                             "Seeds_200down.type")


mir181_bs <- left_join(mir181_bs, Seeds_200down, by = "mir181BS_ID") %>%
  tidyr::nest(all_seeds_200down = c("Seeds_200down.start",
                             "Seeds_200down.end",
                             "Seeds_200down.width",
                             "Seeds_200down.type"))

# add closest mir
colnames(Seeds_1_per_BS) <- c("first_seed_200down.start",
                             "first_seed_200down.end",
                             "first_seed_200down.width",
                             "mir181BS_ID",
                             "first_seed_200down.type",
                             "first_seed_200down.wobble")

mir181_bs <- left_join(mir181_bs, Seeds_1_per_BS, by = "mir181BS_ID") 

mir181_bs <- mir181_bs %>% 
  rowwise() %>% 
  mutate(seed_repetitions.200down = sum(all_seeds_200down$Seeds_200down.type == "seed_6mer"),
        seed_repetitions.200down.wobble = sum(all_seeds_200down$Seeds_200down.type == "seed_6mer_wobble") )



```

```{r}
###################
# plots
###################

# plot seed variations
p <- ggplot(mir181_bs %>% subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched")) %>% subset(!is.na(first_seed_200down.type)), aes(x = first_seed_200down.type, fill = first_seed_200down.wobble))+
  geom_bar(color = "black")+
  theme_paper()+
  scale_fill_manual(values = c(farbe4, darken(farbe4)))+
  theme(legend.position = "None") +
  scale_x_discrete(labels=c(seed_8mer = "U*GAA/UUGU*A",
                            seed_7mer_m8 = "U*GAA/UUGU",
                            seed_7mer_a1 = "GAA/UUGU*A",
                            seed_6mer = "GAA/UUGU",
                            seed_alt_38 = "UGAA/UUG"),
                   guide = guide_axis(angle = 25))
  
p+
  ggtitle("mir181 seed variations in 200nt after the binding site", 
          subtitle = "in case of mutiple seeds the seed nearest to the bindingsite in used")

p1 <- ggplot(mir181_bs %>% subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched")) %>% subset(!is.na(first_seed_200down.type))%>%
               subset(first_seed_200down.wobble == F), aes(x = first_seed_200down.type))+
  geom_bar(color = "black")+
  theme_paper()+
  theme(legend.position = "None") +
  scale_x_discrete(labels=c(seed_8mer = "U*GAA/UUGU*A",
                            seed_7mer_m8 = "U*GAA/UUGU",
                            seed_7mer_a1 = "GAA/UUGU*A",
                            seed_6mer = "GAA/UUGU",
                            seed_alt_38 = "UGAA/UUG"),
                   guide = guide_axis(angle = 25))
  
p1+
  ggtitle("mir181 seed variations in 200nt after the binding site", 
          subtitle = "in case of mutiple seeds the seed nearest to the bindingsite in used")

# plot seed distributions
p2 <- ggplot(mir181_bs %>% subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched")), aes(x = first_seed_200down.start))+
  geom_histogram(color = "black", binwidth = 1)+
  theme_paper()+
  ggtitle("mir181 seed positions", 
          subtitle = "in case of mutiple seeds the seed nearest to the bindingsite in used")
p2


p3 <- ggplot(mir181_bs %>% subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched")), aes(x = first_seed_200down.start))+
  geom_bar(color = "black")+
  theme_paper()+
  facet_wrap(~first_seed_200down.type)+
  ggtitle("mir181 seed positions", 
          subtitle = "in case of mutiple seeds the seed nearest to the bindingsite in used")
p3

# all seeds per BS

p4 <- ggplot(unnest(mir181_bs) %>% 
               subset(Seeds_200down.type %in% c("seed_6mer", "seed_6mer_wobble")) %>%
               subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched")), aes(x = Seeds_200down.start))+
  geom_histogram(color = "black", binwidth = 1)+
  theme_paper()

p4+
  ggtitle("mir181 seed positions", 
          subtitle = "in case of mutiple seeds all are used")



p5 <- ggplot(unnest(mir181_bs) %>% subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched")), aes(x = Seeds_200down.start))+
  geom_bar(color = "black")+
  theme_paper()+
  facet_wrap(~Seeds_200down.type, scales = "free_y")+
  ggtitle("mir181 seed positions", 
          subtitle = "in case of mutiple seeds all are used")
p5

p6 <- ggplot(unnest(mir181_bs) %>% subset(Seeds_200down.type == "seed_6mer_wobble") %>% subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched")), aes(x = Seeds_200down.start))+
  geom_histogram(color = "black", binwidth = 1)+
  theme_paper()
p6+
  ggtitle("mir181 wobble seed positions", 
          subtitle = "in case of mutiple seeds all are used")

p7 <- ggplot(unnest(mir181_bs) %>% subset(Seeds_200down.type == "seed_6mer_wobble") %>% subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched")), aes(x = Seeds_200down.start))+
  geom_histogram(color = "black", binwidth = 1)+
  theme_paper()+
  ylim(c(0,3200))

p7+
  ggtitle("mir181 wobble seed positions", 
          subtitle = "in case of mutiple seeds all are used")

ggplot(mir181_bs %>% subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched")) %>% subset(!is.na(first_seed_200down.type)), aes( x = 1, fill = first_seed_200down.wobble))+
  geom_bar(color = "black")+
  theme_paper()+
  scale_fill_manual(values = c(farbe4, darken(farbe4)))

t <- mir181_bs %>% subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched")) %>%
  pull(first_seed_200down.wobble) %>%
  table()
 
t

t/sum(t)


b <- nrow(mir181_bs %>% subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched")))

a <- mir181_bs %>% subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched")) %>%
  subset(first_seed_200down.wobble == F) 

nrow(a)

nrow(a)/b






ggsave(p, filename = paste0(out, "seed_versions_SupFigureX.pdf"), width = 6, height = 6, units = "cm"  )
ggsave(p1, filename = paste0(out, "seed_versions_Figure5B.pdf"), width = 6, height = 6, units = "cm"  )
ggsave(p4, filename = paste0(out, "seed_position_after_BS.pdf"), width = 6, height = 6, units = "cm"  )
ggsave(p6, filename = paste0(out, "wobbleseed_position_after_BS.pdf"), width = 6, height = 4, units = "cm"  )

ggsave(p7, filename = paste0(out, "wobbleseed_position_after_BS_achse_angepasst.pdf"), width = 6, height = 4, units = "cm"  )

```

## Control: mir142 seed

```{r}
seed_142 <- "CAUCAC"

seed142_downstream_occurence <- vmatchPattern(pattern = seed_142, mir181_bs_200down_seq) %>% lapply(., function(x) as.data.frame(x))

seed142_downstream_occurence <- map2(seed142_downstream_occurence, BS_ID_list, 
                                     ~mutate(.x, mir181BS_ID = .y) ) %>% 
               map_dfr(~.x)


plot_regions <- ( mir181_bs %>% subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched")) %>% pull(mir181BS_ID))

seed142_downstream_occurence <- subset(seed142_downstream_occurence, seed142_downstream_occurence$mir181BS_ID %in% plot_regions)


p <- ggplot(seed142_downstream_occurence, aes(x = start))+
  geom_histogram(color = "black", binwidth = 1)+
  theme_paper()

p+
  ggtitle("mir142 seed positions")

ggsave(p, filename = paste0(out, "mir142_seed_after_mir181_binding_sites.pdf"), width = 6, height = 4, units = "cm"  )

```



### percent binding sites with a seed downstream
```{r}
nrow(mir181_bs %>% subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched")) %>% subset(!is.na(first_seed_200down.start)))/ nrow(mir181_bs %>% subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched")))
```


# Make table of 6mers on transcripts
```{r}
# ! shift transcript positions 200nt back!!!
mir181_bs <- makeGRangesFromDataFrame(mir181_bs, keep.extra.columns = T) %>%
  shift(., shift = -200) %>% as.data.frame(.)

# get 6mers per bs
seeds_tx <- mir181_bs %>% unnest(all_seeds_200down) %>% 
  subset(!is.na(Seeds_200down.start)) %>%
  makeGRangesFromDataFrame(., keep.extra.columns = T)

seeds_tx <- shift(seeds_tx, seeds_tx$Seeds_200down.start -1)

seeds_tx <- resize(seeds_tx, width = seeds_tx$Seeds_200down.width, fix = "start")

```

## 12nt logo around 6mers
```{r}
# filter for 6mers
###################

seeds_6mer <- seeds_tx %>% 
  plyranges::filter(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched")) %>%
  plyranges::filter(Seeds_200down.type == "seed_6mer") %>%
  shift(., 200) 

seeds_6mer_12nt <- as.data.frame(seeds_6mer + 4) %>%
  left_join(transcript_fasta_df, by= c(seqnames = "tx_name"), suffix = c(".bs", ".tx")) %>%
  mutate(strand = "+") %>%
  rowwise(.) %>%
  dplyr::filter(., (end <  width.tx)  & (start > 0)) %>%
  makeGRangesFromDataFrame(., keep.extra.columns =T) 


seeds_6mer_12nt_seq <- getSeq(seeds_6mer_12nt, x = transcript_fasta) %>% 
  RNAStringSet()


seeds_6mer_12nt_seq <- as.data.frame(seeds_6mer_12nt_seq)

logo_6mer_surrounding <- ggseqlogo::ggseqlogo(seeds_6mer_12nt_seq$x, seq_type = "rna")

logo_6mer_surrounding

ggsave(logo_6mer_surrounding, filename = paste0(out, "logo_6mer_surounding.pdf"), height = 6, width = 8, units = "cm")

```



# Save tables

```{r}
saveRDS(mir181_bs, file = paste0(out, "mir181_bs_with_seeds_transcripts.rds"))
saveRDS(seeds_tx, file = paste0(out, "seeds_transcripts.rds"))

t <- mir181_bs %>% as.data.frame() %>%
   subset(set %in% c("mir181_enriched", "ago_bs_mir181_chi&mir181_enriched"))

write_csv(t, paste0(out, "STable6_MREs_transcripts_seeds.csv"))

```

# Session Info

```{r}
sessionInfo()

```

