---
title: "find_MMsat4"
author: "Nikita Verheyden"
date: "2023-05-02"
output: pdf_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Setup

```{r dir}
setwd("D:/Krueger_Lab/Publications/miR181_paper/Figure2/MMsat4")
```

## packages

```{r packages}
library(AnnotationHub)
library(dplyr)
library(ggplot2)
library(ggpubr)
library(rtracklayer)
```


## data

```{r data}
RNA <- read.csv("D:/Krueger_Lab/Publications/miR181_paper/Figure3/RNA_masterframe.csv")
RPF <- read.csv("D:/Krueger_Lab/Publications/miR181_paper/Figure3/RPF_masterframe.csv")

#load the gtf file to compare genes
gff23 <- import.gff3("D:/Krueger_Lab/Ribo_Profiling/run15112022M23/ref_genome/gencode.vM23.annotation.gff3.gz")

#target data
tject <- readRDS("D:/Krueger_Lab/Publications/miR181_paper/Figure1/mir181_binding_sites__venn_types/mir181_bs.rds")
names(tject) <- 1:length(tject$geneName)
tframe <- as.data.frame(tject)
head(tframe)
```


```{r colours}
#colours
farbeneg <- "#b4b4b4"
farbe1 <- "#0073C2FF"
farbe2 <- "#EFC000FF"
farbe3 <- "#CD534CFF"
farbe4 <- "#7AA6DCFF"
farbe5 <- "#868686FF"
farbe6 <- "#003C67FF"
farbe7 <- "#8F7700FF"
farbe8 <- "#3B3B3BFF"
farbe9 <- "#A73030FF"
farbe10 <- "#4A6990FF"
farbe11 <- "#FF6F00FF"
farbe12 <- "#C71000FF"
farbe13 <- "#008EA0FF"
farbe14 <- "#8A4198FF"
farbe15 <- "#5A9599FF"
farbe16 <- "#FF6348FF"
  
RNApcol <- "#b56504"
RNAncol <- "#027d73"
RPFpcol <- "#c4c404"
RPFncol <- "#8d0391"
```


# Get annotation

```{r annotation}
# ah = AnnotationHub()
# query(ah, c("RepeatMasker", "Mus musculus"))
# repeat_masker <- ah[["AH99012"]]

# load the downloaded data
repeat_masker <- readRDS("D:/Krueger_Lab/Publications/miR181_paper/Figure2/MMsat4/repeat_masker.rds")
repeat_masker[repeat_masker$repName == "MMSAT4"]
```

# find overlaps

```{r findingenome}
# same strand
MMSAT4 <- repeat_masker[repeat_masker$repName == "MMSAT4"]

OLgenes <- as.data.frame(subsetByOverlaps(gff23, MMSAT4))


# opposite strand
antiMMSAT4 <- MMSAT4
strand(antiMMSAT4) <- ifelse(strand(MMSAT4) == '+', '-', '+')

antiOLgenes <- as.data.frame(subsetByOverlaps(gff23, antiMMSAT4))

dim(OLgenes)
dim(antiOLgenes)


table(OLgenes$gene_name)
```
```{r splitbyoverlaps}
RNA$MMsat4 <- "No_MMsat4"
RNA$MMsat4[RNA$gene_symbol %in% OLgenes$gene_name] <- "MMsat4"
RNA$MMsat4[RNA$gene_symbol %in% antiOLgenes$gene_name] <- "anti-MMsat4"
table(RNA$MMsat4)

RPF$MMsat4 <- "No_MMsat4"
RPF$MMsat4[RPF$gene_symbol %in% OLgenes$gene_name] <- "MMsat4"
RPF$MMsat4[RPF$gene_symbol %in% antiOLgenes$gene_name] <- "anti-MMsat4"
table(RPF$MMsat4)

```
# plot into MA plots

```{r MAplots}
#RNA
RNAMA <- ggplot(RNA, aes(x=log10(baseMean), y=log2FoldChange, fill=factor(MMsat4, levels = c("No_MMsat4", "MMsat4", "anti-MMsat4")))) + 
  geom_point(shape=21, size=2, colour=farbeneg) +
  scale_fill_manual(values = c(farbeneg, farbe1, farbe3))+
  coord_cartesian(ylim = c(-3,3), xlim = c(0,6))+
  geom_point(data=RNA[RNA$gene_symbol %in% OLgenes$gene_name,], aes(x=log10(baseMean), y=log2FoldChange), colour="black", shape=21, fill=farbe1, size=3)+
  geom_point(data=RNA[RNA$gene_symbol %in% antiOLgenes$gene_name,], aes(x=log10(baseMean), y=log2FoldChange), colour="black", shape=21, fill=farbe3, size=3) + 
  geom_hline(yintercept = 0, colour= "black") +
  theme_bw() +
  theme(legend.title = element_blank()) +
  ggtitle("RNA")
  
#RPF
RPFMA <- ggplot(RPF, aes(x=log10(baseMean), y=log2FoldChange, fill=factor(MMsat4, levels = c("No_MMsat4", "MMsat4", "anti-MMsat4")))) + 
  geom_point(shape=21, size=2, colour=farbeneg) +
  scale_fill_manual(values = c(farbeneg, farbe1, farbe3))+
  coord_cartesian(ylim = c(-3,3), xlim = c(0,6))+
  geom_point(data=RPF[RPF$gene_symbol %in% OLgenes$gene_name,], aes(x=log10(baseMean), y=log2FoldChange), colour="black", shape=21, fill=farbe1, size=3)+
  geom_point(data=RPF[RPF$gene_symbol %in% antiOLgenes$gene_name,], aes(x=log10(baseMean), y=log2FoldChange), colour="black", shape=21, fill=farbe3, size=3) + 
  geom_hline(yintercept = 0, colour= "black") +
  theme_bw() +
  theme(legend.title = element_blank()) +
  ggtitle("RPF")

RNAMA
RPFMA
```


```{r boxplot, fig.width=2, fig.height=4}
my_comparisons <- list( c("No_MMsat4", "MMsat4"), c("No_MMsat4", "anti-MMsat4"), c("MMsat4", "anti-MMsat4") )

RNAbox <- ggplot(RNA, aes(x=MMsat4, y=log2FoldChange)) + 
  geom_boxplot() + 
  stat_compare_means(comparisons = my_comparisons) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +
  ggtitle("RNA")

RPFbox <- ggplot(RPF, aes(x=MMsat4, y=log2FoldChange)) + 
  geom_boxplot() + 
  stat_compare_means(comparisons = my_comparisons) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +
  ggtitle("RPF")

RNAbox

RPFbox
```

# ecdf plot with targets vs MMsat4

```{r targets}
#RNA
RNA$tvsmmsat4 <- "Non-target"
RNA$tvsmmsat4[RNA$gene_symbol %in% OLgenes$gene_name] <- "MMsat4"
RNA$tvsmmsat4[RNA$gene_symbol %in% tframe$geneName] <- "miR-181 target"
RNA$tvsmmsat4[RNA$gene_symbol %in% tframe$geneName & RNA$gene_symbol %in% OLgenes$gene_name] <- "both"
table(RNA$tvsmmsat4)

#RPF
RPF$tvsmmsat4 <- "Non-target"
RPF$tvsmmsat4[RPF$gene_symbol %in% OLgenes$gene_name] <- "MMsat4"
RPF$tvsmmsat4[RPF$gene_symbol %in% tframe$geneName] <- "miR-181 target"
RPF$tvsmmsat4[RPF$gene_symbol %in% tframe$geneName & RPF$gene_symbol %in% OLgenes$gene_name] <- "both"
table(RPF$tvsmmsat4)

#RNA
tolECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(tvsmmsat4, levels = c("Non-target", "MMsat4", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", size=1) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + theme_bw() +
  theme(legend.position = c(0.8, 0.35), legend.title = element_blank(), 
        legend.background = element_rect(colour = "transparent", fill="transparent"), 
        axis.title=element_text(size=16),plot.title = element_text(size=16, face = "bold")) +
  ggtitle("RNA")

tolECDFRNA


#RPF
tolECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(tvsmmsat4, levels = c("Non-target", "MMsat4", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", size=1) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + theme_bw() +
  theme(legend.position = c(0.8, 0.35), legend.title = element_blank(), 
        legend.background = element_rect(colour = "transparent", fill="transparent"), 
        axis.title=element_text(size=16),plot.title = element_text(size=16, face = "bold")) + 
  ggtitle("RPF")

tolECDFRPF
```

# split by region

## 3UTR and CDS
Here we took all targets (genes) that also contain a MMsat4 element and split them by 3'UTR or UTR.
The location of the MMsat4 within the gene is not considered in this analysis.
```{r region}
rframe <- tframe[tframe$geneName %in% OLgenes$gene_name,]

#RNA
RNA$regMMsat4 <- "Non-target"
RNA$regMMsat4[RNA$gene_symbol %in% rframe[rframe$region == "cds","geneName"]] <- "cds"
RNA$regMMsat4[RNA$gene_symbol %in% rframe[rframe$region == "utr3","geneName"]] <- "utr3"
RNA$regMMsat4[RNA$gene_symbol %in% rframe[rframe$region == "utr3","geneName"] & 
                RNA$gene_symbol %in% rframe[rframe$region == "cds","geneName"]] <- "both"

table(RNA$regMMsat4)

#RPF
RPF$regMMsat4 <- "Non-target"
RPF$regMMsat4[RPF$gene_symbol %in% rframe[rframe$region == "cds","geneName"]] <- "cds"
RPF$regMMsat4[RPF$gene_symbol %in% rframe[rframe$region == "utr3","geneName"]] <- "utr3"
RPF$regMMsat4[RPF$gene_symbol %in% rframe[rframe$region == "utr3","geneName"] & 
                RPF$gene_symbol %in% rframe[rframe$region == "cds","geneName"]] <- "both"

table(RPF$regMMsat4)
```
## make ecdf plots with the position data

```{r posecdf}
#RNA
posECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(regMMsat4, levels = c("Non-target", "utr3", "cds", "both")))) + 
  stat_ecdf(geom="step", size=1) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + theme_bw() +
  theme(legend.position = c(0.8, 0.35), legend.title = element_blank(), 
        legend.background = element_rect(colour = "transparent", fill="transparent"), 
        axis.title=element_text(size=16),plot.title = element_text(size=16, face = "bold")) +
  ggtitle("RNA MMsat4 containing target genes")

posECDFRNA

#RPF
posECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(regMMsat4, levels = c("Non-target", "utr3", "cds", "both")))) + 
  stat_ecdf(geom="step", size=1) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + theme_bw() +
  theme(legend.position = c(0.8, 0.35), legend.title = element_blank(), 
        legend.background = element_rect(colour = "transparent", fill="transparent"), 
        axis.title=element_text(size=16),plot.title = element_text(size=16, face = "bold")) +
  ggtitle("RPF MMsat4 containing target genes")

posECDFRPF
```



# session info

```{r session_info}
sessionInfo()
```
