---
title: "fig3 v2"
author: "Nikita Verheyden"
date: "2023-05-28"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```



# Setup

dir
```{r dir}
setwd("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2")
```

## packages
```{r packages}
source("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Supporting_scripts/themes/theme_paper.R")
library(ggplot2)
library(rtracklayer)
library(dplyr)
library(tibble)
library(Rtsne)
library(cliProfiler)
library(GenomicRanges)
library(readxl)
library(writexl)
```

## data
```{r data}
#Ribo profiling
RNA <- read.csv("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/RNA_masterframe.csv")
RPF <- read.csv("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/RPF_masterframe.csv")

#load the gtf file to compare genes
gff23 <- import.gff3("C:/Users/nikit/Documents/Krueger_Lab/Ribo_Profiling/run15112022M23/ref_genome/gencode.vM23.annotation.gff3.gz")
gff23path <- file.path("C:/Users/nikit/Documents/Krueger_Lab/Ribo_Profiling/run15112022M23/ref_genome", "gencode.vM23.annotation.gff3.gz")

#targets
larget <- readRDS("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/mir181_bs_afterFigure2B.rds")
largetframe <- as.data.frame(larget)
table(largetframe$set)
largetframe <- largetframe[!(largetframe$set == "ago_bs_mir181_chi"),]


#ms
MS <- read.csv("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure2/MS_data_MMU_02082023.csv" )


#Translational efficiency
TE <- read.csv("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Supporting_scripts/deltaTE/TE_m23_genesInRNAandRPF.csv")



# modify dataset to only include working targetsets

#only use miR181_enriched und both
# largetframe <- largetframe[largetframe$set %in% c("mir181_enriched", "ago_bs_mir181_chi&mir181_enriched"),]



#MMsat4
# repeat_masker <- readRDS("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure2/MMsat4/repeat_masker.rds")
# MMSAT4 <- repeat_masker[repeat_masker$repName == "MMSAT4"]

```

## colours
```{r colours}
#colours
farbeneg <- "#b4b4b4"
farbe1 <- "#0073C2FF"
farbe2 <- "#EFC000FF"
farbe3 <- "#CD534CFF"
farbe4 <- "#7AA6DCFF"
farbe5 <- "#868686FF"
farbe6 <- "#003C67FF"
farbe7 <- "#8F7700FF"
farbe8 <- "#3B3B3BFF"
farbe9 <- "#A73030FF"
farbe10 <- "#4A6990FF"
farbe11 <- "#FF6F00FF"
farbe12 <- "#C71000FF"
farbe13 <- "#008EA0FF"
farbe14 <- "#8A4198FF"
farbe15 <- "#5A9599FF"
farbe16 <- "#FF6348FF"
  
RNApcol <- "#b56504"
RNAncol <- "#027d73"
RPFpcol <- "#c4c404"
RPFncol <- "#8d0391"
```


## axis limits
```{r axislim}
xlim <- 0.5
zoomfac <- 0.45
```

# export for andreas
```{r exportandreas}
#enriched eclip transcripts
atargetframe <- as.data.frame(table(largetframe$geneName))
colnames(atargetframe) <- c("Target gene names", "Number of MREs")

write_xlsx(atargetframe, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/enriched_eCLIP_transcripts.xlsx")

# upregulated RF
aRFframe <- na.omit(RPF[RPF$log2FoldChange > 0 & RPF$padj <= 0.05 , c("gene_symbol", "log2FoldChange", "padj")])
write_xlsx(aRFframe, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Significantly_upregulated_genes_RF.xlsx")

```

# venn diagram RPF vs targetsites
```{r venntarget}

```


# Target vs non target ecdf
```{r targetecdf}
#RNA
RNA$target <- "Non-target"
RNA$target[RNA$gene_symbol %in% largetframe$geneName] <- "Target"
#RPF
RPF$target <- "Non-target"
RPF$target[RPF$gene_symbol %in% largetframe$geneName] <- "Target"

#MS
MS$target <- "Non-target"
MS$target[MS$gene_symbol %in% largetframe$geneName] <- "Target"
#TE
TE$target <- "Non-target"
TE$target[TE$gene_symbol %in% largetframe$geneName] <- "Target"


#RNA
targetecdfRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(target, levels = c("Non-target", "Target")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1)) +
  coord_cartesian(xlim = c(-0.5, 0.5)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Target genes RNAseq")

targetecdfRNA

# zoom
targetecdfRNAZ <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(target, levels = c("Non-target", "Target")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(), 
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")

targetecdfRNAZ

#RPF
targetecdfRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(target, levels = c("Non-target", "Target")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Target genes ribosome footprint")

targetecdfRPF

# zoom
targetecdfRPFZ <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(target, levels = c("Non-target", "Target")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(), 
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")

targetecdfRPFZ


#MS
targetecdfMS <- ggplot(MS, aes(as.numeric(log2FoldChange), colour=factor(target, levels = c("Non-target", "Target")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Target genes mass spectrometry")

targetecdfMS

# zoom
targetecdfMSZ <- ggplot(MS, aes(as.numeric(log2FoldChange), colour=factor(target, levels = c("Non-target", "Target")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(), 
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")

targetecdfMSZ

#TE
targetecdfTE <- ggplot(TE, aes(as.numeric(log2FoldChange), colour=factor(target, levels = c("Non-target", "Target")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Target genes translational efficiency")

targetecdfTE

# zoom
targetecdfTEZ <- ggplot(TE, aes(as.numeric(log2FoldChange), colour=factor(target, levels = c("Non-target", "Target")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(), 
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")

targetecdfTEZ


#export 
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/TargetECDF_RNA.pdf", width=2, height = 2)
targetecdfRNA
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/TargetECDF_RNA_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
targetecdfRNAZ
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/TargetECDF_RPF.pdf", width=2, height = 2)
targetecdfRPF
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/TargetECDF_RPF_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
targetecdfRPFZ
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/TargetECDF_MS.pdf", width=2, height = 2)
targetecdfMS
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/TargetECDF_MS_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
targetecdfMSZ
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/TargetECDF_TE.pdf", width=2, height = 2)
targetecdfTE
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/TargetECDF_TE_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
targetecdfTEZ
dev.off()

table(RNA$target)
table(RPF$target) 
table(TE$target) 

length(RPF$target)
```

# Target density TE
```{r tedensityTarget}
TEdenstarget <- ggplot(TE, aes(x=log2FoldChange, colour=factor(target, levels = c("Target", "Non-target")))) + 
  geom_density() +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c(farbe1, "black")) +
  theme_paper() +
  coord_cartesian(xlim = c(-1,1)) +
  ggtitle("TE") 

TEdenstarget

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/TargetDensity_TE.pdf", width=2, height = 2)
TEdenstarget
dev.off()
```

## Significance
```{r significanceTarget}
RNATargetTTest <- t.test(RNA[RNA$target=="Target", "log2FoldChange"], RNA[RNA$target=="Non-target", "log2FoldChange"])

RPFTargetTTest <- t.test(RPF[RPF$target=="Target", "log2FoldChange"], RPF[RPF$target=="Non-target", "log2FoldChange"])

TETargetTTest <- t.test(TE[TE$target=="Target", "log2FoldChange"], TE[TE$target=="Non-target", "log2FoldChange"])

MSTargetTTest <- t.test(MS[MS$target=="Target", "log2FoldChange"], MS[MS$target=="Non-target", "log2FoldChange"])

#kolmogorov-smirnov
RNATargetksTest <- ks.test(RNA[RNA$target=="Target", "log2FoldChange"], RNA[RNA$target=="Non-target", "log2FoldChange"])

RPFTargetksTest <- ks.test(RPF[RPF$target=="Target", "log2FoldChange"], RPF[RPF$target=="Non-target", "log2FoldChange"])

TETargetksTest <- ks.test(TE[TE$target=="Target", "log2FoldChange"], TE[TE$target=="Non-target", "log2FoldChange"])

MSTargetksTest <- ks.test(MS[MS$target=="Target", "log2FoldChange"], MS[MS$target=="Non-target", "log2FoldChange"])


Targetsig <- data.frame(c("RNA", "RPF", "TE", "MS"),
                        c("Target", "Target", "Target", "Target"),
                        c(RNATargetTTest$p.value, RPFTargetTTest$p.value, TETargetTTest$p.value, MSTargetTTest$p.value),
                        c(RNATargetksTest$p.value, RPFTargetksTest$p.value, TETargetksTest$p.value, MSTargetksTest$p.value))
colnames(Targetsig) <- c("Experiment", "Subset", "Welch Two Sample t-test", "Asymptotic two-sample Kolmogorov-Smirnov test")
Targetsig

#export
write_xlsx(Targetsig, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/Targetsignificance.xlsx")
```


# number of binding sites histogram
```{r numHist}
bsnum <- as.data.frame(table(largetframe$geneName))
colnames(bsnum) <- c("gene_symbol", "BS_number")

#RNA
RNAnum <- left_join(RNA, bsnum, by="gene_symbol")

RNAnumT <- RNAnum[!is.na(RNAnum$BS_number),]
RNAnumT$BS_number[RNAnumT$BS_number > 10] <- 10

# histogram
numHist_RNA <- ggplot(RNAnumT, aes(x=BS_number)) + 
  geom_histogram(fill=farbeneg, binwidth = 1, colour="black", linewidth = 0.1) +
  theme_paper() +
  scale_y_continuous(expand = c(0,0)) +
  scale_x_continuous("Number of binding sites") +
  coord_cartesian(ylim = c(0, 1250), xlim = c(1, max(RNAnumT$BS_number))) +
  ggtitle("Number of binding sites per gene")

numHist_RNA

#RPF
RPFnum <- left_join(RPF, bsnum, by="gene_symbol")

RPFnumT <- RPFnum[!is.na(RPFnum$BS_number),]
RPFnumT$BS_number[RPFnumT$BS_number > 10] <- 10
# histogram
numHist_RPF <- ggplot(RPFnumT, aes(x=BS_number)) + 
  geom_histogram(fill=farbeneg, binwidth = 1, colour="black", linewidth = 0.1) +
  theme_paper() +
  scale_y_continuous(expand = c(0,0)) +
  scale_x_continuous("Number of binding sites") +
  coord_cartesian(ylim = c(0, 1250), xlim = c(1, max(RPFnumT$BS_number))) +
  ggtitle("Number of binding sites per gene")

numHist_RPF

#MS
MSnum <- left_join(MS, bsnum, by="gene_symbol")

MSnumT <- MSnum[!is.na(MSnum$BS_number),]
MSnumT$BS_number[MSnumT$BS_number > 10] <- 10
# histogram
numHist_MS <- ggplot(MSnumT, aes(x=BS_number)) + 
  geom_histogram(fill=farbeneg, binwidth = 1, colour="black", linewidth = 0.1) +
  theme_paper() +
  scale_y_continuous(expand = c(0,0)) +
  scale_x_continuous("Number of binding sites") +
  coord_cartesian(ylim = c(0, 1250), xlim = c(1, max(MSnumT$BS_number))) +
  ggtitle("Number of binding sites per gene")

numHist_MS

#TE
TEnum <- left_join(TE, bsnum, by="gene_symbol")

TEnumT <- TEnum[!is.na(TEnum$BS_number),]
TEnumT$BS_number[TEnumT$BS_number > 10] <- 10
# histogram
numHist_TE <- ggplot(TEnumT, aes(x=BS_number)) + 
  geom_histogram(fill=farbeneg, binwidth = 1, colour="black", linewidth = 0.1) +
  theme_paper() +
  scale_y_continuous(expand = c(0,0)) +
  scale_x_continuous("Number of binding sites") +
  coord_cartesian(ylim = c(0, 1250), xlim = c(1, max(TEnumT$BS_number))) +
  ggtitle("Number of binding sites per gene")

numHist_TE

#export RPF
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/NumHist_RPF.pdf", width=1.5, height = 2)
numHist_RPF
dev.off()

#export RNA
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/NumHist_RNA.pdf", width=1.5, height = 2)
numHist_RNA
dev.off()

#export MS
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/NumHist_MS.pdf", width=1.5, height = 2)
numHist_MS
dev.off()

#export TE
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/NumHist_TE.pdf", width=1.5, height = 2)
numHist_TE
dev.off()

```

# number of bs ecdf
```{r numecdf}
RNAnum$BS_number[is.na(RNAnum$BS_number)] <- "Non-target"
RNAnum$BS_num_plot <- ifelse(RNAnum$BS_number == "Non-target", "Non-target",
                             ifelse(RNAnum$BS_number == 1, "One MRE",
                                    ifelse(RNAnum$BS_number == 2, "Two MREs", "More")))
                                    
RPFnum$BS_number[is.na(RPFnum$BS_number)] <- "Non-target"
RPFnum$BS_num_plot <- ifelse(RPFnum$BS_number == "Non-target", "Non-target",
                             ifelse(RPFnum$BS_number == 1, "One MRE",
                                    ifelse(RPFnum$BS_number == 2, "Two MREs", "More")))

MSnum$BS_number[is.na(MSnum$BS_number)] <- "Non-target"
MSnum$BS_num_plot <- ifelse(MSnum$BS_number == "Non-target", "Non-target",
                             ifelse(MSnum$BS_number == 1, "One MRE",
                                    ifelse(MSnum$BS_number == 2, "Two MREs", "More")))

TEnum$BS_number[is.na(TEnum$BS_number)] <- "Non-target"
TEnum$BS_num_plot <- ifelse(TEnum$BS_number == "Non-target", "Non-target",
                             ifelse(TEnum$BS_number == 1, "One MRE",
                                    ifelse(TEnum$BS_number == 2, "Two MREs", "More")))


#ecdf plot
#RNA
numecdfRNA <- ggplot(RNAnum, aes(as.numeric(log2FoldChange), colour=factor(BS_num_plot, levels = c("Non-target", "One MRE", "Two MREs", "More")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Number of binding sites RNAseq")

numecdfRNA

# zoom
numecdfRNAZ <- ggplot(RNAnum, aes(as.numeric(log2FoldChange), colour=factor(BS_num_plot, levels = c("Non-target", "One MRE", "Two MREs", "More")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(), 
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")


numecdfRNAZ

#RPF
numecdfRPF <- ggplot(RPFnum, aes(as.numeric(log2FoldChange), colour=factor(BS_num_plot, levels = c("Non-target", "One MRE", "Two MREs", "More")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Number of binding sites ribosome footprint")

numecdfRPF

# zoom
numecdfRPFZ <- ggplot(RPFnum, aes(as.numeric(log2FoldChange), colour=factor(BS_num_plot, levels = c("Non-target", "One MRE", "Two MREs", "More")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(), 
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")


numecdfRPFZ

#MS
numecdfMS <- ggplot(MSnum, aes(as.numeric(log2FoldChange), colour=factor(BS_num_plot, levels = c("Non-target", "One MRE", "Two MREs", "More")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Number of binding sites mass spectrometry")

numecdfMS

# zoom
numecdfMSZ <- ggplot(MSnum, aes(as.numeric(log2FoldChange), colour=factor(BS_num_plot, levels = c("Non-target", "One MRE", "Two MREs", "More")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(), 
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")


numecdfMSZ

#TE
numecdfTE <- ggplot(TEnum, aes(as.numeric(log2FoldChange), colour=factor(BS_num_plot, levels = c("Non-target", "One MRE", "Two MREs", "More")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Number of binding sites translational efficiency")

numecdfTE

# zoom
numecdfTEZ <- ggplot(TEnum, aes(as.numeric(log2FoldChange), colour=factor(BS_num_plot, levels = c("Non-target", "One MRE", "Two MREs", "More")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(), 
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")


numecdfTEZ


# export
#RNA
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/NumECDF_RNA.pdf", width=2, height = 2)
numecdfRNA
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/NumECDF_RNA_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
numecdfRNAZ
dev.off()

#RPF
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/NumECDF_RPF.pdf", width=2, height = 2)
numecdfRPF
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/NumECDF_RPF_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
numecdfRPFZ
dev.off()

#MS
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/NumECDF_MS.pdf", width=2, height = 2)
numecdfMS
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/NumECDF_MS_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
numecdfMSZ
dev.off()

#TE
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/NumECDF_TE.pdf", width=2, height = 2)
numecdfTE
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/NumECDF_TE_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
numecdfTEZ
dev.off()


#numplot 2 samples

RPFnum$BS_num_plot2 <- ifelse(RPFnum$BS_number == "Non-target", "Non-target",
                             ifelse(RPFnum$BS_number %in% c(1,2), "1-2 MREs", "More"))


#ecdf plot
numecdfRPF2 <- ggplot(RPFnum, aes(as.numeric(log2FoldChange), colour=factor(BS_num_plot2, levels = c("Non-target", "1-2 MREs", "More")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Number of binding sites ribosome footprint")

numecdfRPF2

#zoom
numecdfRPF2Z <- ggplot(RPFnum, aes(as.numeric(log2FoldChange), colour=factor(BS_num_plot2, levels = c("Non-target", "1-2 MREs", "More")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(), 
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")

numecdfRPF2Z


pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/NumECDF_RPF2.pdf", width=2, height = 2)
numecdfRPF2
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/NumECDF_RPF2_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
numecdfRPF2Z
dev.off()

table(RPFnum$BS_num_plot)
```

# TE density num
```{R TEdensityNum}
TEdensNum <- ggplot(TEnum, aes(x=log2FoldChange, colour=factor(BS_num_plot, levels = c("One MRE", "Two MREs", "More", "Non-target")))) + 
  geom_density() +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c(farbe1, farbe2, farbe3, "black")) +
  theme_paper() +
  coord_cartesian(xlim = c(-1,1)) +
  ggtitle("TE")

TEdensNum

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/NumDensity_TE.pdf", width=2, height = 2)
TEdensNum
dev.off()

```
## Significance
```{r significanceNum}
RNANumTTest1 <- t.test(RNAnum[RNAnum$BS_num_plot=="One MRE", "log2FoldChange"], RNAnum[RNAnum$BS_num_plot=="Non-target", "log2FoldChange"])
RNANumTTest2 <- t.test(RNAnum[RNAnum$BS_num_plot=="Two MREs", "log2FoldChange"], RNAnum[RNAnum$BS_num_plot=="Non-target", "log2FoldChange"])
RNANumTTest3 <- t.test(RNAnum[RNAnum$BS_num_plot=="More", "log2FoldChange"], RNAnum[RNAnum$BS_num_plot=="Non-target", "log2FoldChange"])

RPFNumTTest1 <- t.test(RPFnum[RPFnum$BS_num_plot=="One MRE", "log2FoldChange"], RPFnum[RPFnum$BS_num_plot=="Non-target", "log2FoldChange"])
RPFNumTTest2 <- t.test(RPFnum[RPFnum$BS_num_plot=="Two MREs", "log2FoldChange"], RPFnum[RPFnum$BS_num_plot=="Non-target", "log2FoldChange"])
RPFNumTTest3 <- t.test(RPFnum[RPFnum$BS_num_plot=="More", "log2FoldChange"], RPFnum[RPFnum$BS_num_plot=="Non-target", "log2FoldChange"])

TENumTTest1 <- t.test(TEnum[TEnum$BS_num_plot=="One MRE", "log2FoldChange"], TEnum[TEnum$BS_num_plot=="Non-target", "log2FoldChange"])
TENumTTest2 <- t.test(TEnum[TEnum$BS_num_plot=="Two MREs", "log2FoldChange"], TEnum[TEnum$BS_num_plot=="Non-target", "log2FoldChange"])
TENumTTest3 <- t.test(TEnum[TEnum$BS_num_plot=="More", "log2FoldChange"], TEnum[TEnum$BS_num_plot=="Non-target", "log2FoldChange"])

MSNumTTest1 <- t.test(MSnum[MSnum$BS_num_plot=="One MRE", "log2FoldChange"], MSnum[MSnum$BS_num_plot=="Non-target", "log2FoldChange"])
MSNumTTest2 <- t.test(MSnum[MSnum$BS_num_plot=="Two MREs", "log2FoldChange"], MSnum[MSnum$BS_num_plot=="Non-target", "log2FoldChange"])
MSNumTTest3 <- t.test(MSnum[MSnum$BS_num_plot=="More", "log2FoldChange"], MSnum[MSnum$BS_num_plot=="Non-target", "log2FoldChange"])

#kolmogorov-smirnov
RNANumksTest1 <- ks.test(RNAnum[RNAnum$BS_num_plot=="One MRE", "log2FoldChange"], RNAnum[RNAnum$BS_num_plot=="Non-target", "log2FoldChange"])
RNANumksTest2 <- ks.test(RNAnum[RNAnum$BS_num_plot=="Two MREs", "log2FoldChange"], RNAnum[RNAnum$BS_num_plot=="Non-target", "log2FoldChange"])
RNANumksTest3 <- ks.test(RNAnum[RNAnum$BS_num_plot=="More", "log2FoldChange"], RNAnum[RNAnum$BS_num_plot=="Non-target", "log2FoldChange"])

RPFNumksTest1 <- ks.test(RPFnum[RPFnum$BS_num_plot=="One MRE", "log2FoldChange"], RPFnum[RPFnum$BS_num_plot=="Non-target", "log2FoldChange"])
RPFNumksTest2 <- ks.test(RPFnum[RPFnum$BS_num_plot=="Two MREs", "log2FoldChange"], RPFnum[RPFnum$BS_num_plot=="Non-target", "log2FoldChange"])
RPFNumksTest3 <- ks.test(RPFnum[RPFnum$BS_num_plot=="More", "log2FoldChange"], RPFnum[RPFnum$BS_num_plot=="Non-target", "log2FoldChange"])

TENumksTest1 <- ks.test(TEnum[TEnum$BS_num_plot=="One MRE", "log2FoldChange"], TEnum[TEnum$BS_num_plot=="Non-target", "log2FoldChange"])
TENumksTest2 <- ks.test(TEnum[TEnum$BS_num_plot=="Two MREs", "log2FoldChange"], TEnum[TEnum$BS_num_plot=="Non-target", "log2FoldChange"])
TENumksTest3 <- ks.test(TEnum[TEnum$BS_num_plot=="More", "log2FoldChange"], TEnum[TEnum$BS_num_plot=="Non-target", "log2FoldChange"])

MSNumksTest1 <- ks.test(MSnum[MSnum$BS_num_plot=="One MRE", "log2FoldChange"], MSnum[MSnum$BS_num_plot=="Non-target", "log2FoldChange"])
MSNumksTest2 <- ks.test(MSnum[MSnum$BS_num_plot=="Two MREs", "log2FoldChange"], MSnum[MSnum$BS_num_plot=="Non-target", "log2FoldChange"])
MSNumksTest3 <- ks.test(MSnum[MSnum$BS_num_plot=="More", "log2FoldChange"], MSnum[MSnum$BS_num_plot=="Non-target", "log2FoldChange"])


Numsig <- data.frame(c("RNA", "RNA", "RNA", "RPF", "RPF", "RPF", "TE", "TE", "TE", "MS", "MS", "MS"),
                        c("One MRE", "Two MREs", "More", "One MRE", "Two MREs", "More",
                          "One MRE", "Two MREs", "More","One MRE", "Two MREs", "More"),
                        c(RNANumTTest1$p.value, RNANumTTest2$p.value, RNANumTTest3$p.value,
                          RPFNumTTest1$p.value, RPFNumTTest2$p.value, RPFNumTTest3$p.value,
                          TENumTTest1$p.value, TENumTTest2$p.value, TENumTTest3$p.value, 
                          MSNumTTest1$p.value, MSNumTTest2$p.value, MSNumTTest3$p.value),
                        c(RNANumksTest1$p.value, RNANumksTest2$p.value, RNANumksTest3$p.value,
                          RPFNumksTest1$p.value, RPFNumksTest2$p.value, RPFNumksTest3$p.value, 
                          TENumksTest1$p.value, TENumksTest2$p.value, TENumksTest3$p.value,
                          MSNumksTest1$p.value, MSNumksTest2$p.value, MSNumksTest3$p.value))
colnames(Numsig) <- c("Experiment", "Subset", "Welch Two Sample t-test", "Asymptotic two-sample Kolmogorov-Smirnov test")
Numsig

#export
write_xlsx(Numsig, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/Numsignificance.xlsx")
```

#  metaGeneProfile

```{r metaGeneProfile}
glarget <- makeGRangesFromDataFrame(largetframe, keep.extra.columns = F)
meta <- metaGeneProfile(object = glarget, annotation = gff23path)

meta


pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/metagene_profile.pdf", width=2, height = 2)
meta
dev.off()
```


# position in gene ECDF
```{r posecdf}
# majority votes region

# add data to RNA/RPF frames
utr5 <- as.data.frame(table(largetframe[largetframe$region == "utr5","geneName"]))
colnames(utr5) <- c("gene_symbol", "num_5UTR")
cds <- as.data.frame(table(largetframe[largetframe$region == "cds","geneName"]))
colnames(cds) <- c("gene_symbol", "num_CDS")
utr3 <- as.data.frame(table(largetframe[largetframe$region == "utr3","geneName"]))
colnames(utr3) <- c("gene_symbol", "num_3UTR")

#RNA
RNAnumr <- left_join(RNAnum, utr5, by="gene_symbol")
RNAnumr <- left_join(RNAnumr, cds, by="gene_symbol")
RNAnumr <- left_join(RNAnumr, utr3, by="gene_symbol")

RNAnumr[is.na(RNAnumr$num_5UTR), "num_5UTR"] <- 0
RNAnumr[is.na(RNAnumr$num_CDS), "num_CDS"] <- 0
RNAnumr[is.na(RNAnumr$num_3UTR), "num_3UTR"] <- 0

# make majority id
RNAnumr$region_majority <- "Equal"
RNAnumr$region_majority[RNAnumr$num_5UTR > RNAnumr$num_CDS & RNAnumr$num_5UTR > RNAnumr$num_3UTR] <- "5'UTR" 
RNAnumr$region_majority[RNAnumr$num_CDS > RNAnumr$num_5UTR & RNAnumr$num_CDS > RNAnumr$num_3UTR] <- "CDS" 
RNAnumr$region_majority[RNAnumr$num_3UTR > RNAnumr$num_5UTR & RNAnumr$num_3UTR > RNAnumr$num_CDS] <- "3'UTR" 
RNAnumr$region_majority[RNAnumr$num_5UTR == 0 & RNAnumr$num_CDS == 0 & RNAnumr$num_3UTR == 0] <- "Non-target"

# make single id
RNAnumr$region_single <- RNAnumr$region_majority
RNAnumr$region_single[as.numeric(RNAnumr$BS_number) > 1] <- "Multiple"

#RPF
RPFnumr <- left_join(RPFnum, utr5, by="gene_symbol")
RPFnumr <- left_join(RPFnumr, cds, by="gene_symbol")
RPFnumr <- left_join(RPFnumr, utr3, by="gene_symbol")

RPFnumr[is.na(RPFnumr$num_5UTR), "num_5UTR"] <- 0
RPFnumr[is.na(RPFnumr$num_CDS), "num_CDS"] <- 0
RPFnumr[is.na(RPFnumr$num_3UTR), "num_3UTR"] <- 0

# make majority id
RPFnumr$region_majority <- "Equal"
RPFnumr$region_majority[RPFnumr$num_5UTR > RPFnumr$num_CDS & RPFnumr$num_5UTR > RPFnumr$num_3UTR] <- "5'UTR" 
RPFnumr$region_majority[RPFnumr$num_CDS > RPFnumr$num_5UTR & RPFnumr$num_CDS > RPFnumr$num_3UTR] <- "CDS" 
RPFnumr$region_majority[RPFnumr$num_3UTR > RPFnumr$num_5UTR & RPFnumr$num_3UTR > RPFnumr$num_CDS] <- "3'UTR" 
RPFnumr$region_majority[RPFnumr$num_5UTR == 0 & RPFnumr$num_CDS == 0 & RPFnumr$num_3UTR == 0] <- "Non-target"

# make single id
RPFnumr$region_single <- RPFnumr$region_majority
RPFnumr$region_single[as.numeric(RPFnumr$BS_number) > 1] <- "Multiple"


#MS
MSnumr <- left_join(MSnum, utr5, by="gene_symbol")
MSnumr <- left_join(MSnumr, cds, by="gene_symbol")
MSnumr <- left_join(MSnumr, utr3, by="gene_symbol")

MSnumr[is.na(MSnumr$num_5UTR), "num_5UTR"] <- 0
MSnumr[is.na(MSnumr$num_CDS), "num_CDS"] <- 0
MSnumr[is.na(MSnumr$num_3UTR), "num_3UTR"] <- 0

# make majority id
MSnumr$region_majority <- "Equal"
MSnumr$region_majority[MSnumr$num_5UTR > MSnumr$num_CDS & MSnumr$num_5UTR > MSnumr$num_3UTR] <- "5'UTR" 
MSnumr$region_majority[MSnumr$num_CDS > MSnumr$num_5UTR & MSnumr$num_CDS > MSnumr$num_3UTR] <- "CDS" 
MSnumr$region_majority[MSnumr$num_3UTR > MSnumr$num_5UTR & MSnumr$num_3UTR > MSnumr$num_CDS] <- "3'UTR" 
MSnumr$region_majority[MSnumr$num_5UTR == 0 & MSnumr$num_CDS == 0 & MSnumr$num_3UTR == 0] <- "Non-target"

# make single id
MSnumr$region_single <- MSnumr$region_majority
MSnumr$region_single[as.numeric(MSnumr$BS_number) > 1] <- "Multiple"


#TE
TEnumr <- left_join(TEnum, utr5, by="gene_symbol")
TEnumr <- left_join(TEnumr, cds, by="gene_symbol")
TEnumr <- left_join(TEnumr, utr3, by="gene_symbol")

TEnumr[is.na(TEnumr$num_5UTR), "num_5UTR"] <- 0
TEnumr[is.na(TEnumr$num_CDS), "num_CDS"] <- 0
TEnumr[is.na(TEnumr$num_3UTR), "num_3UTR"] <- 0

# make majority id
TEnumr$region_majority <- "Equal"
TEnumr$region_majority[TEnumr$num_5UTR > TEnumr$num_CDS & TEnumr$num_5UTR > TEnumr$num_3UTR] <- "5'UTR" 
TEnumr$region_majority[TEnumr$num_CDS > TEnumr$num_5UTR & TEnumr$num_CDS > TEnumr$num_3UTR] <- "CDS" 
TEnumr$region_majority[TEnumr$num_3UTR > TEnumr$num_5UTR & TEnumr$num_3UTR > TEnumr$num_CDS] <- "3'UTR" 
TEnumr$region_majority[TEnumr$num_5UTR == 0 & TEnumr$num_CDS == 0 & TEnumr$num_3UTR == 0] <- "Non-target"

# make single id
TEnumr$region_single <- TEnumr$region_majority
TEnumr$region_single[as.numeric(TEnumr$BS_number) > 1] <- "Multiple"



#ECDF

#RNA
#majority
regMAJecdfRNA <- ggplot(RNAnumr, aes(as.numeric(log2FoldChange), colour=factor(region_majority, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "Equal")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Region by majority vote RNAseq")

regMAJecdfRNA


regMAJecdfRNAZ <- ggplot(RNAnumr, aes(as.numeric(log2FoldChange), colour=factor(region_majority, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "Equal")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(), 
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")
regMAJecdfRNAZ


pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/region_majorityECDF_RNA.pdf", width=2, height = 2)
regMAJecdfRNA
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/region_majorityECDF_RNA_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
regMAJecdfRNAZ
dev.off()

table(RNAnumr$region_majority)



#single
regMAJecdfRNAS <- ggplot(RNAnumr, aes(as.numeric(log2FoldChange), colour=factor(region_single, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Region single RNAseq")

regMAJecdfRNAS


regMAJecdfRNASZ <- ggplot(RNAnumr, aes(as.numeric(log2FoldChange), colour=factor(region_single, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(), 
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")
regMAJecdfRNASZ


pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/region_singleECDF_RNA.pdf", width=2, height = 2)
regMAJecdfRNAS
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/region_singleECDF_RNA_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
regMAJecdfRNASZ
dev.off()

table(RNAnumr$region_single)

#RPF
#majority
regMAJecdfRPF <- ggplot(RPFnumr, aes(as.numeric(log2FoldChange), colour=factor(region_majority, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "Equal")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Region by majority vote ribosome footprint")

regMAJecdfRPF


regMAJecdfRPFZ <- ggplot(RPFnumr, aes(as.numeric(log2FoldChange), colour=factor(region_majority, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "Equal")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(), 
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")
regMAJecdfRPFZ


pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/region_majorityECDF_RPF.pdf", width=2, height = 2)
regMAJecdfRPF
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/region_majorityECDF_RPF_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
regMAJecdfRPFZ
dev.off()

table(RPFnumr$region_majority)



#single
regMAJecdfRPFS <- ggplot(RPFnumr, aes(as.numeric(log2FoldChange), colour=factor(region_single, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Region single ribosome footprint")

regMAJecdfRPFS


regMAJecdfRPFSZ <- ggplot(RPFnumr, aes(as.numeric(log2FoldChange), colour=factor(region_single, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(), 
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")
regMAJecdfRPFSZ


pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/region_singleECDF_RPF.pdf", width=2, height = 2)
regMAJecdfRPFS
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/region_singleECDF_RPF_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
regMAJecdfRPFSZ
dev.off()

table(RPFnumr$region_single)


#MS
#majority
regMAJecdfMS <- ggplot(MSnumr, aes(as.numeric(log2FoldChange), colour=factor(region_majority, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "Equal")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Region by majority vote mass spectrometry")

regMAJecdfMS


regMAJecdfMSZ <- ggplot(MSnumr, aes(as.numeric(log2FoldChange), colour=factor(region_majority, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "Equal")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(), 
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")
regMAJecdfMSZ


pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/region_majorityECDF_MS.pdf", width=2, height = 2)
regMAJecdfMS
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/region_majorityECDF_MS_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
regMAJecdfMSZ
dev.off()

table(MSnumr$region_majority)



#single
regMAJecdfMSS <- ggplot(MSnumr, aes(as.numeric(log2FoldChange), colour=factor(region_single, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Region single mass spectrometry")

regMAJecdfMSS


regMAJecdfMSSZ <- ggplot(MSnumr, aes(as.numeric(log2FoldChange), colour=factor(region_single, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(), 
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")
regMAJecdfMSSZ


pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/region_singleECDF_MS.pdf", width=2, height = 2)
regMAJecdfMSS
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/region_singleECDF_MS_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
regMAJecdfMSSZ
dev.off()

table(MSnumr$region_single)


#TE
#majority
regMAJecdfTE <- ggplot(TEnumr, aes(as.numeric(log2FoldChange), colour=factor(region_majority, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "Equal")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Region by majority vote translational efficiency")

regMAJecdfTE


regMAJecdfTEZ <- ggplot(TEnumr, aes(as.numeric(log2FoldChange), colour=factor(region_majority, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "Equal")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(), 
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")
regMAJecdfTEZ


pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/region_majorityECDF_TE.pdf", width=2, height = 2)
regMAJecdfTE
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/region_majorityECDF_TE_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
regMAJecdfTEZ
dev.off()

table(TEnumr$region_majority)



#single
regMAJecdfTES <- ggplot(TEnumr, aes(as.numeric(log2FoldChange), colour=factor(region_single, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Region single translational efficiency")

regMAJecdfTES


regMAJecdfTESZ <- ggplot(TEnumr, aes(as.numeric(log2FoldChange), colour=factor(region_single, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(), 
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")
regMAJecdfTESZ


pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/region_singleECDF_TE.pdf", width=2, height = 2)
regMAJecdfTES
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/region_singleECDF_TE_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
regMAJecdfTESZ
dev.off()

table(TEnumr$region_single)

```


# TE density position
```{R TEdensityregion}
TEdensPos <- ggplot(TEnumr, aes(x=log2FoldChange, colour=factor(region_single, levels = c( "5'UTR", "CDS", "3'UTR", "Multiple", "Non-target")))) +
  geom_density() +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c(farbe1, farbe2, farbe3, farbe4, "black")) +
  theme_paper() +
  coord_cartesian(xlim = c(-1,1)) +
  ggtitle("TE")

TEdensPos

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/PosDensity_TE.pdf", width=2, height = 2)
TEdensPos
dev.off()

```


## Significance
```{r significanceRegion}
RNARegTTest1 <- t.test(RNAnumr[RNAnumr$region_single=="5'UTR", "log2FoldChange"], RNAnumr[RNAnumr$region_single=="Non-target", "log2FoldChange"])
RNARegTTest2 <- t.test(RNAnumr[RNAnumr$region_single=="CDS", "log2FoldChange"], RNAnumr[RNAnumr$region_single=="Non-target", "log2FoldChange"])
RNARegTTest3 <- t.test(RNAnumr[RNAnumr$region_single=="3'UTR", "log2FoldChange"], RNAnumr[RNAnumr$region_single=="Non-target", "log2FoldChange"])

RPFRegTTest1 <- t.test(RPFnumr[RPFnumr$region_single=="5'UTR", "log2FoldChange"], RPFnumr[RPFnumr$region_single=="Non-target", "log2FoldChange"])
RPFRegTTest2 <- t.test(RPFnumr[RPFnumr$region_single=="CDS", "log2FoldChange"], RPFnumr[RPFnumr$region_single=="Non-target", "log2FoldChange"])
RPFRegTTest3 <- t.test(RPFnumr[RPFnumr$region_single=="3'UTR", "log2FoldChange"], RPFnumr[RPFnumr$region_single=="Non-target", "log2FoldChange"])

TERegTTest1 <- t.test(TEnumr[TEnumr$region_single=="5'UTR", "log2FoldChange"], TEnumr[TEnumr$region_single=="Non-target", "log2FoldChange"])
TERegTTest2 <- t.test(TEnumr[TEnumr$region_single=="CDS", "log2FoldChange"], TEnumr[TEnumr$region_single=="Non-target", "log2FoldChange"])
TERegTTest3 <- t.test(TEnumr[TEnumr$region_single=="3'UTR", "log2FoldChange"], TEnumr[TEnumr$region_single=="Non-target", "log2FoldChange"])

MSRegTTest1 <- t.test(MSnumr[MSnumr$region_single=="5'UTR", "log2FoldChange"], MSnumr[MSnumr$region_single=="Non-target", "log2FoldChange"])
MSRegTTest2 <- t.test(MSnumr[MSnumr$region_single=="CDS", "log2FoldChange"], MSnumr[MSnumr$region_single=="Non-target", "log2FoldChange"])
MSRegTTest3 <- t.test(MSnumr[MSnumr$region_single=="3'UTR", "log2FoldChange"], MSnumr[MSnumr$region_single=="Non-target", "log2FoldChange"])

#kolmogorov-smirnov
RNARegksTest1 <- ks.test(RNAnumr[RNAnumr$region_single=="5'UTR", "log2FoldChange"], RNAnumr[RNAnumr$region_single=="Non-target", "log2FoldChange"])
RNARegksTest2 <- ks.test(RNAnumr[RNAnumr$region_single=="CDS", "log2FoldChange"], RNAnumr[RNAnumr$region_single=="Non-target", "log2FoldChange"])
RNARegksTest3 <- ks.test(RNAnumr[RNAnumr$region_single=="3'UTR", "log2FoldChange"], RNAnumr[RNAnumr$region_single=="Non-target", "log2FoldChange"])

RPFRegksTest1 <- ks.test(RPFnumr[RPFnumr$region_single=="5'UTR", "log2FoldChange"], RPFnumr[RPFnumr$region_single=="Non-target", "log2FoldChange"])
RPFRegksTest2 <- ks.test(RPFnumr[RPFnumr$region_single=="CDS", "log2FoldChange"], RPFnumr[RPFnumr$region_single=="Non-target", "log2FoldChange"])
RPFRegksTest3 <- ks.test(RPFnumr[RPFnumr$region_single=="3'UTR", "log2FoldChange"], RPFnumr[RPFnumr$region_single=="Non-target", "log2FoldChange"])

TERegksTest1 <- ks.test(TEnumr[TEnumr$region_single=="5'UTR", "log2FoldChange"], TEnumr[TEnumr$region_single=="Non-target", "log2FoldChange"])
TERegksTest2 <- ks.test(TEnumr[TEnumr$region_single=="CDS", "log2FoldChange"], TEnumr[TEnumr$region_single=="Non-target", "log2FoldChange"])
TERegksTest3 <- ks.test(TEnumr[TEnumr$region_single=="3'UTR", "log2FoldChange"], TEnumr[TEnumr$region_single=="Non-target", "log2FoldChange"])

MSRegksTest1 <- ks.test(MSnumr[MSnumr$region_single=="5'UTR", "log2FoldChange"], MSnumr[MSnumr$region_single=="Non-target", "log2FoldChange"])
MSRegksTest2 <- ks.test(MSnumr[MSnumr$region_single=="CDS", "log2FoldChange"], MSnumr[MSnumr$region_single=="Non-target", "log2FoldChange"])
MSRegksTest3 <- ks.test(MSnumr[MSnumr$region_single=="3'UTR", "log2FoldChange"], MSnumr[MSnumr$region_single=="Non-target", "log2FoldChange"])


Regsig <- data.frame(c("RNA", "RNA", "RNA", "RPF", "RPF", "RPF", "TE", "TE", "TE", "MS", "MS", "MS"),
                        c("5'UTR", "CDS", "3'UTR", "5'UTR", "CDS", "3'UTR",
                          "5'UTR", "CDS", "3'UTR","5'UTR", "CDS", "3'UTR"),
                        c(RNARegTTest1$p.value, RNARegTTest2$p.value, RNARegTTest3$p.value,
                          RPFRegTTest1$p.value, RPFRegTTest2$p.value, RPFRegTTest3$p.value,
                          TERegTTest1$p.value, TERegTTest2$p.value, TERegTTest3$p.value, 
                          MSRegTTest1$p.value, MSRegTTest2$p.value, MSRegTTest3$p.value),
                        c(RNARegksTest1$p.value, RNARegksTest2$p.value, RNARegksTest3$p.value,
                          RPFRegksTest1$p.value, RPFRegksTest2$p.value, RPFRegksTest3$p.value, 
                          TERegksTest1$p.value, TERegksTest2$p.value, TERegksTest3$p.value,
                          MSRegksTest1$p.value, MSRegksTest2$p.value, MSRegksTest3$p.value))
colnames(Regsig) <- c("Experiment", "Subset", "Welch Two Sample t-test", "Asymptotic two-sample Kolmogorov-Smirnov test")
Regsig

#export
write_xlsx(Regsig, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/Regsignificance.xlsx")
```

# only until here is in the paper------------------------------------------------------------------



# analysis pipeline including altering of basis date along the line


## dotplot region vs num vs expression
```{r dotplot}
utr5tab <- as.data.frame(table(largetframe[largetframe$region == "utr5","geneName"]))
colnames(utr5tab) <- c("gene_symbol", "utr5_sites")
cdstab <- as.data.frame(table(largetframe[largetframe$region == "cds","geneName"]))
colnames(cdstab) <- c("gene_symbol", "cds_sites")
utr3tab <- as.data.frame(table(largetframe[largetframe$region == "utr3","geneName"]))
colnames(utr3tab) <- c("gene_symbol", "utr3_sites")

#RNA
#regions seperately
RNArnum <- left_join(RNA, utr5tab, by="gene_symbol")
RNArnum[is.na(RNArnum$utr5_sites), "utr5_sites"] <- 0
RNArnum <- left_join(RNArnum, cdstab, by="gene_symbol")
RNArnum[is.na(RNArnum$cds_sites), "cds_sites"] <- 0
RNArnum <- left_join(RNArnum, utr3tab, by="gene_symbol")
RNArnum[is.na(RNArnum$utr3_sites), "utr3_sites"] <- 0
#regions sum
RNArnum$total_sites <- rowSums(RNArnum[, c("utr5_sites", "cds_sites", "utr3_sites")])
#regions writing
RNArnum$bsregions <- "Non-target"
RNArnum$bsregions[RNArnum$utr5_sites > 0] <- "5'UTR"
RNArnum$bsregions[RNArnum$cds_sites > 0] <- "CDS"
RNArnum$bsregions[RNArnum$utr3_sites > 0] <- "3'UTR"
RNArnum$bsregions[RNArnum$utr5_sites > 0 & RNArnum$cds_sites > 0] <- "5'UTR and CDS"
RNArnum$bsregions[RNArnum$utr3_sites > 0 & RNArnum$cds_sites > 0] <- "CDS and 3'UTR"
RNArnum$bsregions[RNArnum$utr5_sites > 0 & RNArnum$utr3_sites > 0] <- "5'UTR and 3'UTR"
RNArnum$bsregions[RNArnum$utr5_sites > 0 & RNArnum$cds_sites > 0 & RNArnum$utr3_sites > 0] <- "5'UTR, CDS and 3'UTR"


head(RNArnum)

#RPF
#regions seperately
RPFrnum <- left_join(RPF, utr5tab, by="gene_symbol")
RPFrnum[is.na(RPFrnum$utr5_sites), "utr5_sites"] <- 0
RPFrnum <- left_join(RPFrnum, cdstab, by="gene_symbol")
RPFrnum[is.na(RPFrnum$cds_sites), "cds_sites"] <- 0
RPFrnum <- left_join(RPFrnum, utr3tab, by="gene_symbol")
RPFrnum[is.na(RPFrnum$utr3_sites), "utr3_sites"] <- 0
#regions sum
RPFrnum$total_sites <- rowSums(RPFrnum[, c("utr5_sites", "cds_sites", "utr3_sites")])
#regions writing
RPFrnum$bsregions <- "Non-target"
RPFrnum$bsregions[RPFrnum$utr5_sites > 0] <- "5'UTR"
RPFrnum$bsregions[RPFrnum$cds_sites > 0] <- "CDS"
RPFrnum$bsregions[RPFrnum$utr3_sites > 0] <- "3'UTR"
RPFrnum$bsregions[RPFrnum$utr5_sites > 0 & RPFrnum$cds_sites > 0] <- "5'UTR and CDS"
RPFrnum$bsregions[RPFrnum$utr3_sites > 0 & RPFrnum$cds_sites > 0] <- "CDS and 3'UTR"
RPFrnum$bsregions[RPFrnum$utr5_sites > 0 & RPFrnum$utr3_sites > 0] <- "5'UTR and 3'UTR"
RPFrnum$bsregions[RPFrnum$utr5_sites > 0 & RPFrnum$cds_sites > 0 & RPFrnum$utr3_sites > 0] <- "5'UTR, CDS and 3'UTR"


head(RPFrnum)
```

### Dotplot for number and type of MRE

```{r nrdotplot}
#colour limits
clim <- 0.5
#RNA
RNArnumlim <- RNArnum
RNArnumlim[RNArnumlim$log2FoldChange > clim, "log2FoldChange"] <- clim
RNArnumlim[RNArnumlim$log2FoldChange < -clim, "log2FoldChange"] <- -clim


rndplotRNA <- ggplot(RNArnumlim[RNArnumlim$padj <= 0.05 ,], aes(x=factor(bsregions, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "5'UTR and CDS", "CDS and 3'UTR", "5'UTR and 3'UTR", "5'UTR, CDS and 3'UTR")),
                                  y=total_sites, 
                                  colour=log2FoldChange)) + 
  geom_point(position = "jitter") +
  scale_colour_gradient2(low = "#05fae6", mid="black", high="#fa8c05", limits=c(-clim, clim)) + 
  theme_paper() + theme(legend.position = "right", axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  ggtitle("RNA")

rndplotRNA

#RPF
RPFrnumlim <- RPFrnum
RPFrnumlim[RPFrnumlim$log2FoldChange > clim, "log2FoldChange"] <- clim
RPFrnumlim[RPFrnumlim$log2FoldChange < -clim, "log2FoldChange"] <- -clim


rndplotRPF <- ggplot(RPFrnumlim[RPFrnumlim$padj <= 0.05 ,], aes(x=factor(bsregions, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "5'UTR and CDS", "CDS and 3'UTR", "5'UTR and 3'UTR", "5'UTR, CDS and 3'UTR")),
                                  y=total_sites, 
                                  colour=log2FoldChange)) + 
  geom_point(position = "jitter") +
  scale_colour_gradient2(low = "#05fae6", mid="black", high="#fa8c05", limits=c(-clim, clim)) + 
  theme_paper() + theme(legend.position = "right", axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  ggtitle("RPF")

rndplotRPF

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Dotplot_region_number-RNA.pdf", height = 4, width = 5)
rndplotRNA 
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Dotplot_region_number-RPF.pdf", height = 4, width = 5)
rndplotRPF 
dev.off()
```

## mir181majoritys

analyze the chimeric binders for mir181 majoritys
(apparently not included in this dataframe anymore)
```{r maj}
larget
table(larget$mir_IP)


# Subset the columns you want to include in the t-SNE plot
subset_cols <- c("scoreSum", "scoreMean", "scoreMax", "region", "n_mir181", "n_mir181a", "n_mir181b", "n_mir181c", "n_mir181d", "set", "counts.bs.1_KO", 
                 "counts.bs.2_KO", "counts.bs.3_KO", "counts.bs.4_WT", "counts.bs.5_WT", "counts.bs.6_WT")

# Subset the tibble based on the selected columns
subset_data <- larget[, subset_cols]

# Perform t-SNE dimensionality reduction
tsne_result <- Rtsne(subset_data)

# Create a data frame from the t-SNE result
tsne_df <- data.frame(X = tsne_result$Y[, 1], Y = tsne_result$Y[, 2])

# Plot the t-SNE result
ggplot(tsne_df, aes(X, Y)) +
  geom_point() +
  labs(title = "t-SNE Plot")



```





## remove non seeds

```{r seed}
largetframe$first_seed_200down.wobble <- as.character(largetframe$first_seed_200down.wobble)
largetframe$first_seed_200down.wobble[is.na(largetframe$first_seed_200down.wobble)] <- "No seed"
table(largetframe$first_seed_200down.wobble)

#remove non seeds
largetframeS <- largetframe[!(largetframe$first_seed_200down.wobble =="No seed"),]
table(largetframeS$first_seed_200down.wobble)
```









### ECDF plot seed yes or no (ignored wobble)

```{r wobbleedcf}
#adjust data
#RNA
RNA$wobble <- "Non-target"
RNA$wobble[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble=="No seed","geneName"]] <- "No seed"
RNA$wobble[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble=="TRUE","geneName"]] <- "Canonical"
RNA$wobble[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble=="FALSE","geneName"]] <- "Non-canonical"




#ecdf
#RNA
wobbleECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(wobble, levels = c("Non-target", "No seed", "Seed")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA wobble")

wobbleECDFRNA


```






# inspect targetdata

We're keeping all of those targets for now but will analyze the in ecdf plots
```{r set}
table(largetframe$set)

colnames(largetframe)

#remove duplicates but kep NAs (and ask melina where the NAs come from)
largetframe <- largetframe[!(duplicated(largetframe$BS_ID) & !is.na(largetframe$BS_ID)), ]

## adjust type column
# Add "No seed" as a level to the factor column
largetframe$first_seed_200down.type <- factor(largetframe$first_seed_200down.type, levels = c(levels(largetframe$first_seed_200down.type), "No seed"))

# Replace NAs with "No seed"
largetframe$first_seed_200down.type[is.na(largetframe$first_seed_200down.type)] <- "No seed"
table(largetframe$first_seed_200down.type)

## adjust wobble column
largetframe$first_seed_200down.wobble[is.na(largetframe$first_seed_200down.wobble)] <- "No seed"


``` 


# ECDF plots

each code chunk is a split of the main target table that is then used for a specific ecdf plot


## targetfiles
```{r targetsets, fig.show="hold", out.width="50%"}
#old target with introns
#RNA
RNA$oldtarg <- "Non-target"
RNA$oldtarg[RNA$gene_symbol %in% tframe$geneName] <- "Target"
#RPF
RPF$oldtarg <- "Non-target"
RPF$oldtarg[RPF$gene_symbol %in% tframe$geneName] <- "Target"

#giant frame
#RNA
RNA$target <- "Non-target"
RNA$target[RNA$gene_symbol %in% largetframe$geneName] <- "Target"
#RPF
RPF$target <- "Non-target"
RPF$target[RPF$gene_symbol %in% largetframe$geneName] <- "Target"


#ECDF
#old targets with introns
#RNA
targetoldECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(oldtarg, levels = c("Non-target", "Target")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA targetset with introns")

targetoldECDFRNA

#RPF
targetoldECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(oldtarg, levels = c("Non-target", "Target")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF targetset with introns")

targetoldECDFRPF

#targets
#RNA
targetECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(target, levels = c("Non-target", "Target")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA target")

targetECDFRNA

#RPF
targetECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(target, levels = c("Non-target", "Target")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF target")

targetECDFRPF

```


## datasets

```{r datasets, fig.show="hold", out.width="50%"}
#RNA
RNA$targetset <- "Non-target"
RNA$targetset[RNA$gene_symbol %in% largetframe[largetframe$set == "ago_bs_mir181_chi", "geneName"]] <- "ago_bs_mir181_chi"
RNA$targetset[RNA$gene_symbol %in% largetframe[largetframe$set == "mir181_enriched", "geneName"]] <- "mir181_enriched"
RNA$targetset[RNA$gene_symbol %in% largetframe[largetframe$set == "ago_bs_mir181_chi&mir181_enriched", "geneName"]] <- "both"

table(RNA$targetset)

#RPF
RPF$targetset <- "Non-target"
RPF$targetset[RPF$gene_symbol %in% largetframe[largetframe$set == "ago_bs_mir181_chi", "geneName"]] <- "ago_bs_mir181_chi"
RPF$targetset[RPF$gene_symbol %in% largetframe[largetframe$set == "mir181_enriched", "geneName"]] <- "mir181_enriched"
RPF$targetset[RPF$gene_symbol %in% largetframe[largetframe$set == "ago_bs_mir181_chi&mir181_enriched", "geneName"]] <- "both"

table(RPF$targetset)


# ecdf plots
#RNA
setECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(targetset, levels = c("Non-target", "ago_bs_mir181_chi", "mir181_enriched", "both")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA by targetset")

setECDFRNA

#RPF
setECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(targetset, levels = c("Non-target", "ago_bs_mir181_chi", "mir181_enriched", "both")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF by targetset")

setECDFRPF

```






## region (single targets)
```{r region, fig.show="hold", out.width="50%"}
#get number of binding sites per gene to be able to sort for singles
bsnum <- as.data.frame(table(largetframe$geneName))
colnames(bsnum) <- c("geneName", "BS_number")

#RNA
RNA$region_single <- "Non-target"
RNA$region_single[RNA$gene_symbol %in% largetframe[largetframe$region == "utr5", "geneName"]] <- "5'UTR"
RNA$region_single[RNA$gene_symbol %in% largetframe[largetframe$region == "cds", "geneName"]] <- "CDS"
RNA$region_single[RNA$gene_symbol %in% largetframe[largetframe$region == "utr3", "geneName"]] <- "3'UTR"
RNA$region_single[RNA$gene_symbol %in% bsnum[bsnum$BS_number > 1, "geneName"]] <- "multiple"

table(RNA$region_single)

#RPF
RPF$region_single <- "Non-target"
RPF$region_single[RPF$gene_symbol %in% largetframe[largetframe$region == "utr5", "geneName"]] <- "5'UTR"
RPF$region_single[RPF$gene_symbol %in% largetframe[largetframe$region == "cds", "geneName"]] <- "CDS"
RPF$region_single[RPF$gene_symbol %in% largetframe[largetframe$region == "utr3", "geneName"]] <- "3'UTR"
RPF$region_single[RPF$gene_symbol %in% bsnum[bsnum$BS_number > 1, "geneName"]] <- "multiple"

table(RPF$region_single)


# ECDF plots
#RNA
regsingECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(region_single, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbeneg)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA region only single BS")

regsingECDFRNA

#RPF
regsingECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(region_single, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbeneg)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF region only single BS")

regsingECDFRPF 

```

## number of target sites

```{r targetnum, fig.show="hold", out.width="50%"}
colnames(bsnum) <- c("gene_symbol", "BS_number")

#RNA
RNAnum <- left_join(RNA, bsnum, by="gene_symbol")
RNAnum$BS_number[is.na(RNAnum$BS_number)] <- "Non-target"
RNAnum$BS_num_plot <- ifelse(RNAnum$BS_number == "Non-target", "Non-target",
                             ifelse(RNAnum$BS_number == 1, "One bs",
                                    ifelse(RNAnum$BS_number == 2, "Two bs", "More")))


#RPF
RPFnum <- left_join(RPF, bsnum, by="gene_symbol")
RPFnum$BS_number[is.na(RPFnum$BS_number)] <- "Non-target"
RPFnum$BS_num_plot <- ifelse(RPFnum$BS_number == "Non-target", "Non-target",
                             ifelse(RPFnum$BS_number == 1, "One bs",
                                    ifelse(RPFnum$BS_number == 2, "Two bs", "More")))

#ecdf plots
#RNA
numECDFRNA <- ggplot(RNAnum, aes(as.numeric(log2FoldChange), colour=factor(BS_num_plot, levels = c("Non-target", "One bs", "Two bs", "More")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA number of BS")

numECDFRNA

#RPF
numECDFRPF <- ggplot(RPFnum, aes(as.numeric(log2FoldChange), colour=factor(BS_num_plot, levels = c("Non-target", "One bs", "Two bs", "More")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF number of BS")

numECDFRPF


pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/NumECDF_RPF2.pdf", width=2, height = 2)
numECDFRPF
dev.off()
```

## MMsat4
```{r mmsat4, fig.show="hold", out.width="50%"}
mmsat4frame <- as.data.frame(subsetByOverlaps(gff23, MMSAT4))

#RNA
RNA$tvsmmsat4 <- "Non-target"
RNA$tvsmmsat4[RNA$gene_symbol %in% mmsat4frame$gene_name] <- "MMsat4"
RNA$tvsmmsat4[RNA$gene_symbol %in% tframe$geneName] <- "miR-181 target"
RNA$tvsmmsat4[RNA$gene_symbol %in% tframe$geneName & RNA$gene_symbol %in% mmsat4frame$gene_name] <- "both"
table(RNA$tvsmmsat4)

#RPF
RPF$tvsmmsat4 <- "Non-target"
RPF$tvsmmsat4[RPF$gene_symbol %in% mmsat4frame$gene_name] <- "MMsat4"
RPF$tvsmmsat4[RPF$gene_symbol %in% tframe$geneName] <- "miR-181 target"
RPF$tvsmmsat4[RPF$gene_symbol %in% tframe$geneName & RPF$gene_symbol %in% mmsat4frame$gene_name] <- "both"
table(RPF$tvsmmsat4)

#RNA
tolECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(tvsmmsat4, levels = c("Non-target", "MMsat4", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA Mmsat4 vs target")

tolECDFRNA


#RPF
tolECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(tvsmmsat4, levels = c("Non-target", "MMsat4", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF Mmsat4 vs target")

tolECDFRPF

```

## Mmsat4 in 3UTR and CDS
Here we took all targets (genes) that also contain a MMsat4 element and split them by 3'UTR or UTR.
The location of the MMsat4 within the gene is not considered in this analysis.
```{r mmsat4region, fig.show="hold", out.width="50%"}
rframe <- tframe[tframe$geneName %in% mmsat4frame$gene_name,]

#RNA
RNA$regMMsat4 <- "Non-target"
RNA$regMMsat4[RNA$gene_symbol %in% rframe[rframe$region == "cds","geneName"]] <- "cds"
RNA$regMMsat4[RNA$gene_symbol %in% rframe[rframe$region == "utr3","geneName"]] <- "utr3"
RNA$regMMsat4[RNA$gene_symbol %in% rframe[rframe$region == "utr3","geneName"] & 
                RNA$gene_symbol %in% rframe[rframe$region == "cds","geneName"]] <- "both"

table(RNA$regMMsat4)

#RPF
RPF$regMMsat4 <- "Non-target"
RPF$regMMsat4[RPF$gene_symbol %in% rframe[rframe$region == "cds","geneName"]] <- "cds"
RPF$regMMsat4[RPF$gene_symbol %in% rframe[rframe$region == "utr3","geneName"]] <- "utr3"
RPF$regMMsat4[RPF$gene_symbol %in% rframe[rframe$region == "utr3","geneName"] & 
                RPF$gene_symbol %in% rframe[rframe$region == "cds","geneName"]] <- "both"

table(RPF$regMMsat4)

#ecdf plots
#RNA
posECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(regMMsat4, levels = c("Non-target", "utr3", "cds", "both")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA MMsat4 by region")

posECDFRNA

#RPF
posECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(regMMsat4, levels = c("Non-target", "utr3", "cds", "both")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) +
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF MMsat4 by region")

posECDFRPF
```

## by type of MRE (wobble and non wobble combined)
```{r MREtype}
#RNA
RNA$MREtype <- "Non-target"
RNA$MREtype[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "No seed","geneName"]] <- "No seed"
RNA$MREtype[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_6mer","geneName"]] <- "seed_6mer"
RNA$MREtype[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_a1","geneName"]] <- "seed_7mer_a1"
RNA$MREtype[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_m8","geneName"]] <- "seed_7mer_m8"
RNA$MREtype[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_8mer","geneName"]] <- "seed_8mer"

#RPF
RPF$MREtype <- "Non-target"
RPF$MREtype[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "No seed","geneName"]] <- "No seed"
RPF$MREtype[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_6mer","geneName"]] <- "seed_6mer"
RPF$MREtype[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_a1","geneName"]] <- "seed_7mer_a1"
RPF$MREtype[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_m8","geneName"]] <- "seed_7mer_m8"
RPF$MREtype[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_8mer","geneName"]] <- "seed_8mer"

# ecdf plots
#RNA
typeECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA type of MRE")

typeECDFRNA

#RPF
typeECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPFtype of MRE")

typeECDFRPF


```

## divide by wobble and non wobble in seed
```{r wobble}
#RNA
RNA$wobble <- "Non-target"
RNA$wobble[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == "No seed","geneName"]] <- "No seed"
RNA$wobble[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == FALSE,"geneName"]] <- "Canonical"
RNA$wobble[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == TRUE,"geneName"]] <- "Wobble"

#RPF
RPF$wobble <- "Non-target"
RPF$wobble[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == "No seed","geneName"]] <- "No seed"
RPF$wobble[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == FALSE,"geneName"]] <- "Canonical"
RPF$wobble[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == TRUE,"geneName"]] <- "Wobble"


#ECDF plots
#RNA
wobbleECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), 
                              colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Wobble")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA Wobble")

wobbleECDFRNA

#RPF
wobbleECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), 
                              colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Wobble")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF Wobble")

wobbleECDFRPF

```

## mRNA target expression levels (10% bins or quartiles)

```{r}
#RNA
tRNA <- RNA[RNA$target == "Target",]
tRNA <- tRNA[order(tRNA$baseMean, decreasing = T),]
RNA$expressionlvl <- "Non-target"
RNA$expressionlvl[RNA$gene_symbol %in% tRNA[1:(0.25*length(tRNA$gene_symbol)), "gene_symbol"]] <- "High"
RNA$expressionlvl[RNA$gene_symbol %in% tRNA[(0.25*length(tRNA$gene_symbol)):(0.5*length(tRNA$gene_symbol)), "gene_symbol"]] <- "Mid-high"
RNA$expressionlvl[RNA$gene_symbol %in% tRNA[(0.5*length(tRNA$gene_symbol)):(0.75*length(tRNA$gene_symbol)), "gene_symbol"]] <- "Mid-low"
RNA$expressionlvl[RNA$gene_symbol %in% tRNA[(0.75*length(tRNA$gene_symbol)):length(tRNA$gene_symbol), "gene_symbol"]] <- "Low"

#RPF
tRPF <- RPF[RPF$target == "Target",]
tRPF <- tRPF[order(tRPF$baseMean, decreasing = T),]
RPF$expressionlvl <- "Non-target"
RPF$expressionlvl[RPF$gene_symbol %in% tRPF[1:(0.25*length(tRPF$gene_symbol)), "gene_symbol"]] <- "High"
RPF$expressionlvl[RPF$gene_symbol %in% tRPF[(0.25*length(tRPF$gene_symbol)):(0.5*length(tRPF$gene_symbol)), "gene_symbol"]] <- "Mid-high"
RPF$expressionlvl[RPF$gene_symbol %in% tRPF[(0.5*length(tRPF$gene_symbol)):(0.75*length(tRPF$gene_symbol)), "gene_symbol"]] <- "Mid-low"
RPF$expressionlvl[RPF$gene_symbol %in% tRPF[(0.75*length(tRPF$gene_symbol)):length(tRPF$gene_symbol), "gene_symbol"]] <- "Low"


#ECDF plots
#RNA
expECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), 
                              colour=factor(expressionlvl, levels = c("Non-target", "Low", "Mid-low", "Mid-high", "High")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA transcript expression")

expECDFRNA

#RPF
expECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), 
                              colour=factor(expressionlvl, levels = c("Non-target", "Low", "Mid-low", "Mid-high", "High")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF transcript expression")

expECDFRPF



```

## Region with multiple sites ordered by "importance"
they were sorted in the following order: 5utr cds 3UTR
```{r}
RNA$region_multiple <- "Non-target"
RNA$region_multiple[RNA$gene_symbol %in% largetframe[largetframe$region == "utr5", "geneName"]] <- "utr5"
RNA$region_multiple[RNA$gene_symbol %in% largetframe[largetframe$region == "cds", "geneName"]] <- "cds"
RNA$region_multiple[RNA$gene_symbol %in% largetframe[largetframe$region == "utr3", "geneName"]] <- "utr3"

RPF$region_multiple <- "Non-target"
RPF$region_multiple[RPF$gene_symbol %in% largetframe[largetframe$region == "utr5", "geneName"]] <- "utr5"
RPF$region_multiple[RPF$gene_symbol %in% largetframe[largetframe$region == "cds", "geneName"]] <- "cds"
RPF$region_multiple[RPF$gene_symbol %in% largetframe[largetframe$region == "utr3", "geneName"]] <- "utr3"

# ECDF plots

#RNA
regionmultiECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), 
                              colour=factor(region_multiple, levels = c("Non-target", "utr5", "cds", "utr3")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA region including multiples priority:5utr,cds,3utr")

regionmultiECDFRNA

#RPF
regionmultiECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), 
                              colour=factor(region_multiple, levels = c("Non-target", "utr5", "cds", "utr3")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF region including multiples priority:5utr,cds,3utr")

regionmultiECDFRPF

```


```{r inspect}
head(largetframe)

```


# Export
"C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4"
```{r export}
#target

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/targetoldintronECDFRNA.pdf", width=2, height = 2)
targetoldECDFRNA
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/targetoldintronECDFRPF.pdf", width=2, height = 2)
targetoldECDFRPF
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/targetECDF_RNA.pdf", width=2, height = 2)
targetECDFRNA
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/targetECDF_RPF.pdf", width=2, height = 2)
targetECDFRPF
dev.off()

#set

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/setECDF_RNA.pdf", width=2, height = 2)
setECDFRNA
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/setECDF_RPF.pdf", width=2, height = 2)
setECDFRPF
dev.off()

#region single targets

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/regsingECDF_RNA.pdf", width=2, height = 2)
regsingECDFRNA
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/regsingECDF_RPF.pdf", width=2, height = 2)
regsingECDFRPF
dev.off()

#number

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/numECDF_RNA.pdf", width=2, height = 2)
numECDFRNA
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/numECDF_RPF.pdf", width=2, height = 2)
numECDFRPF
dev.off()

# MMsat4 vs target

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/MMsat4vsTargetECDF_RNA.pdf", width=2, height = 2)
tolECDFRNA
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/MMsat4vsTargetECDF_RPF.pdf", width=2, height = 2)
tolECDFRPF
dev.off()

# region with MMsat4 cds and 3'utr

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/MMsat4byRegionECDF_RNA.pdf", width=2, height = 2)
posECDFRNA
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/MMsat4byRegionECDF_RPF.pdf", width=2, height = 2)
posECDFRPF
dev.off()

# MRE type

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/MRtypeECDF_RNA.pdf", width=2, height = 2)
typeECDFRNA
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/MRtypeECDF_RPF.pdf", width=2, height = 2)
typeECDFRPF
dev.off()

# Wobble

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/wobbleECDF_RNA.pdf", width=2, height = 2)
wobbleECDFRNA
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/wobbleECDF_RPF.pdf", width=2, height = 2)
wobbleECDFRPF
dev.off()

# transcript expression

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/expresssionECDF_RNA.pdf", width=2, height = 2)
expECDFRNA
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/expresssionECDF_RPF.pdf", width=2, height = 2)
expECDFRPF
dev.off()

# transcript expression

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/regiomultiECDF_RNA.pdf", width=2, height = 2)
regionmultiECDFRNA
dev.off()

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/Fig3_2/regiomultiECDF_RPF.pdf", width=2, height = 2)
regionmultiECDFRPF
dev.off()

```


# GO plots and enrichment analysis

```{r goplots}

```


# session info
```{r sessioninfo}
sessionInfo()
```
