---
title: "Figure4_v2"
author: "Nikita Verheyden"
date: "2023-06-13"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```



# Setup

dir
```{r dir}
setwd("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4")
```


## Packages
```{r packages}
source("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Supporting_scripts/themes/theme_paper.R")
library(ggplot2)
library(rtracklayer)
library(dplyr)
library(tibble)
library(Rtsne)
library(cliProfiler)
library(GenomicRanges)
library(cowplot)
library(eulerr)
library(readxl)

```

## Data
```{r data}
#Ribo profiling
RNA <- read.csv("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/RNA_masterframe.csv")
RPF <- read.csv("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/RPF_masterframe.csv")

#load the gtf file to compare genes
gff23 <- import.gff3("C:/Users/nikit/Documents/Krueger_Lab/Ribo_Profiling/run15112022M23/ref_genome/gencode.vM23.annotation.gff3.gz")
gff23path <- file.path("C:/Users/nikit/Documents/Krueger_Lab/Ribo_Profiling/run15112022M23/ref_genome", "gencode.vM23.annotation.gff3.gz")

#targets
larget <- readRDS("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/mir181_bs_with_seeds_transcripts.rds")
largetframe <- as.data.frame(larget)
table(largetframe$set)
# largetframe <- largetframe[!(largetframe$set == "ago_bs_mir181_chi"),]

largetframe <- largetframe %>%
  subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched"))


# modify dataset to only include working targetsets

#only use miR181_enriched und both
# largetframe <- largetframe[largetframe$set %in% c("mir181_enriched", "ago_bs_mir181_chi&mir181_enriched"),]


#ms
MS <- as.data.frame(read_xlsx("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/MS_analysis.xlsx", 
                              sheet = 2))
#adjust upper case lower case of gene names of MS data:
#function
capFirst <- function(s) {
  paste(toupper(substring(s, 1, 1)), substring(s, 2), sep = "")
}
MS$`Gene Symbol` <- tolower(MS$`Gene Symbol`)
MS$`Gene Symbol` <- capFirst(MS$`Gene Symbol`)
names(MS)[names(MS) == 'Gene Symbol'] <- 'gene_symbol'
names(MS)[names(MS) == 'P-value'] <- 'pvalue'
names(MS)[names(MS) == 'Log2(FC)'] <- 'log2FoldChange'

head(MS$gene_symbol)

#Translational efficiency
TE <- read.csv("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Supporting_scripts/deltaTE/TE_m23_genesInRNAandRPF.csv")




#MMsat4
# repeat_masker <- readRDS("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure2/MMsat4/repeat_masker.rds")
# MMSAT4 <- repeat_masker[repeat_masker$repName == "MMSAT4"]
# MurSatRep1 <- repeat_masker[repeat_masker$repName == "MurSatRep1"]


#melina repeats
melrep <- readRDS("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/rep_on_transcripts.rds")
MMSAT4 <- melrep[melrep$repName == "MMSAT4"]
MurSatRep1 <- melrep[melrep$repName == "MurSatRep1"]
```


## colours
```{r colours}
#colours
farbeneg <- "#b4b4b4"
farbe1 <- "#0073C2FF"
farbe2 <- "#EFC000FF"
farbe3 <- "#CD534CFF"
farbe4 <- "#7AA6DCFF"
farbe5 <- "#868686FF"
farbe6 <- "#003C67FF"
farbe7 <- "#8F7700FF"
farbe8 <- "#3B3B3BFF"
farbe9 <- "#A73030FF"
farbe10 <- "#4A6990FF"
farbe11 <- "#FF6F00FF"
farbe12 <- "#C71000FF"
farbe13 <- "#008EA0FF"
farbe14 <- "#8A4198FF"
farbe15 <- "#5A9599FF"
farbe16 <- "#FF6348FF"
  
RNApcol <- "#b56504"
RNAncol <- "#027d73"
RPFpcol <- "#c4c404"
RPFncol <- "#8d0391"
```

## axis limits
```{r axislim}
xlim <- 0.5
zoomfac <- 0.45
```


## number of binding sites 
```{r num}
bsnum <- as.data.frame(table(largetframe$geneName))
colnames(bsnum) <- c("gene_symbol", "BS_number")
```


# wobble plot

## ecdf
```{r wobbleecdf}
#RNA
RNA$wobble <- "Non-target"
RNA$wobble[RNA$gene_symbol %in% largetframe[is.na(largetframe$first_seed_200down.wobble) ,"geneName"]] <- "No seed"
RNA$wobble[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == FALSE ,"geneName"]] <- "Canonical"
RNA$wobble[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == TRUE ,"geneName"]] <- "Non-Canonical"
RNA$wobble[RNA$gene_symbol %in% bsnum[bsnum$BS_number > 1, "gene_symbol"]] <- "Multiple"
#RPF
RPF$wobble <- "Non-target"
RPF$wobble[RPF$gene_symbol %in% largetframe[is.na(largetframe$first_seed_200down.wobble) ,"geneName"]] <- "No seed"
RPF$wobble[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == FALSE ,"geneName"]] <- "Canonical"
RPF$wobble[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == TRUE ,"geneName"]] <- "Non-Canonical"
RPF$wobble[RPF$gene_symbol %in% bsnum[bsnum$BS_number > 1, "gene_symbol"]] <- "Multiple"

#MS
MS$wobble <- "Non-target"
MS$wobble[MS$gene_symbol %in% largetframe[is.na(largetframe$first_seed_200down.wobble) ,"geneName"]] <- "No seed"
MS$wobble[MS$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == FALSE ,"geneName"]] <- "Canonical"
MS$wobble[MS$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == TRUE ,"geneName"]] <- "Non-Canonical"
MS$wobble[MS$gene_symbol %in% bsnum[bsnum$BS_number > 1, "gene_symbol"]] <- "Multiple"

#TE
TE$wobble <- "Non-target"
TE$wobble[TE$gene_symbol %in% largetframe[is.na(largetframe$first_seed_200down.wobble) ,"geneName"]] <- "No seed"
TE$wobble[TE$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == FALSE ,"geneName"]] <- "Canonical"
TE$wobble[TE$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == TRUE ,"geneName"]] <- "Non-Canonical"
TE$wobble[TE$gene_symbol %in% bsnum[bsnum$BS_number > 1, "gene_symbol"]] <- "Multiple"

#ECDF

#RNA
wobbleecdfRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Non-Canonical", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbeneg, farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Target genes RNAseq")

wobbleecdfRNA

#zoom
wobbleecdfRNAZ <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Non-Canonical", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbeneg, farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(),
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")


wobbleecdfRNAZ

#RPF
wobbleecdfRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Non-Canonical", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbeneg, farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Target genes ribosome profiling")

wobbleecdfRPF

#zoom
wobbleecdfRPFZ <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Non-Canonical", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbeneg, farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(),
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")


wobbleecdfRPFZ

#MS
wobbleecdfMS <- ggplot(MS, aes(as.numeric(log2FoldChange), colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Non-Canonical", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbeneg, farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Target genes mass spectrometry")

wobbleecdfMS

#zoom
wobbleecdfMSZ <- ggplot(MS, aes(as.numeric(log2FoldChange), colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Non-Canonical", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbeneg, farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(),
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")


wobbleecdfMSZ

#TE
wobbleecdfTE <- ggplot(TE, aes(as.numeric(log2FoldChange), colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Non-Canonical", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbeneg, farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Target genes translational efficiency")

wobbleecdfTE

#zoom
wobbleecdfTEZ <- ggplot(TE, aes(as.numeric(log2FoldChange), colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Non-Canonical", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbeneg, farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(),
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")


wobbleecdfTEZ


#export RNA
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/wobbleECDF_RNA.pdf", width=2, height = 2)
wobbleecdfRNA
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/wobbleECDF_RNA_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
wobbleecdfRNAZ
dev.off()

#export RPF
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/wobbleECDF_RPF.pdf", width=2, height = 2)
wobbleecdfRPF
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/wobbleECDF_RPF_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
wobbleecdfRPFZ
dev.off()

#export MS
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/wobbleECDF_MS.pdf", width=2, height = 2)
wobbleecdfMS
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/wobbleECDF_MS_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
wobbleecdfMSZ
dev.off()

#export TE
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/wobbleECDF_TE.pdf", width=2, height = 2)
wobbleecdfTE
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/wobbleECDF_TE_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
wobbleecdfTEZ
dev.off()


table(RPF$wobble)
```
## by type of MRE (wobble and non wobble combined)
```{r MREtype}
#RNA
RNA$MREtype <- "Non-target"
RNA$MREtype[RNA$gene_symbol %in% largetframe[is.na(largetframe$first_seed_200down.type),"geneName"]] <- "No seed"
RNA$MREtype[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_6mer","geneName"]] <- "seed_6mer"
RNA$MREtype[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_a1","geneName"]] <- "seed_7mer_a1"
RNA$MREtype[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_m8","geneName"]] <- "seed_7mer_m8"
RNA$MREtype[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_8mer","geneName"]] <- "seed_8mer"
RNA$MREtype[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_alt_38","geneName"]] <- "seed_alt_38"
RNA$MREtype[RNA$gene_symbol %in% bsnum[bsnum$BS_number > 1, "gene_symbol"]] <- "Multiple"


#RPF
RPF$MREtype <- "Non-target"
RPF$MREtype[RPF$gene_symbol %in% largetframe[is.na(largetframe$first_seed_200down.type),"geneName"]] <- "No seed"
RPF$MREtype[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_6mer","geneName"]] <- "seed_6mer"
RPF$MREtype[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_a1","geneName"]] <- "seed_7mer_a1"
RPF$MREtype[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_m8","geneName"]] <- "seed_7mer_m8"
RPF$MREtype[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_8mer","geneName"]] <- "seed_8mer"
RPF$MREtype[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_alt_38","geneName"]] <- "seed_alt_38"
RPF$MREtype[RPF$gene_symbol %in% bsnum[bsnum$BS_number > 1, "gene_symbol"]] <- "Multiple"


#MS
MS$MREtype <- "Non-target"
MS$MREtype[MS$gene_symbol %in% largetframe[is.na(largetframe$first_seed_200down.type),"geneName"]] <- "No seed"
MS$MREtype[MS$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_6mer","geneName"]] <- "seed_6mer"
MS$MREtype[MS$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_a1","geneName"]] <- "seed_7mer_a1"
MS$MREtype[MS$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_m8","geneName"]] <- "seed_7mer_m8"
MS$MREtype[MS$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_8mer","geneName"]] <- "seed_8mer"
MS$MREtype[MS$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_alt_38","geneName"]] <- "seed_alt_38"
MS$MREtype[MS$gene_symbol %in% bsnum[bsnum$BS_number > 1, "gene_symbol"]] <- "Multiple"


#TE
TE$MREtype <- "Non-target"
TE$MREtype[TE$gene_symbol %in% largetframe[is.na(largetframe$first_seed_200down.type),"geneName"]] <- "No seed"
TE$MREtype[TE$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_6mer","geneName"]] <- "seed_6mer"
TE$MREtype[TE$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_a1","geneName"]] <- "seed_7mer_a1"
TE$MREtype[TE$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_m8","geneName"]] <- "seed_7mer_m8"
TE$MREtype[TE$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_8mer","geneName"]] <- "seed_8mer"
TE$MREtype[TE$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_alt_38","geneName"]] <- "seed_alt_38"
TE$MREtype[TE$gene_symbol %in% bsnum[bsnum$BS_number > 1, "gene_symbol"]] <- "Multiple"



# ecdf plots
#RNA
typeECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA type of MRE")

typeECDFRNA


# zoom
typeECDFRNAZ <- ggplot(RNA, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(),
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")

typeECDFRNAZ

#RPF
typeECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF type of MRE")

typeECDFRPF

# zoom
typeECDFRPFZ <- ggplot(RPF, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(),
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")

typeECDFRPFZ


#MS
typeECDFMS <- ggplot(MS, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("MS type of MRE")

typeECDFMS

# zoom
typeECDFMSZ <- ggplot(MS, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(),
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")

typeECDFMSZ


#TE
typeECDFTE <- ggplot(TE, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("TE type of MRE")

typeECDFTE

# zoom
typeECDFTEZ <- ggplot(TE, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(),
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")

typeECDFTEZ


#export RNA
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/typeECDF_RNA.pdf", width=2, height = 2)
typeECDFRNA
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/typeECDF_RNA_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
typeECDFRNAZ
dev.off()

#export RPF
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/typeECDF_RPF.pdf", width=2, height = 2)
typeECDFRPF
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/typeECDF_RPF_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
typeECDFRPFZ
dev.off()

#export MS
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/typeECDF_MS.pdf", width=2, height = 2)
typeECDFMS
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/typeECDF_MS_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
typeECDFMSZ
dev.off()

#export TE
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/typeECDF_TE.pdf", width=2, height = 2)
typeECDFTE
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/typeECDF_TE_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
typeECDFTEZ
dev.off()


table(RPF$MREtype)
```



# mmsat4
```{r mmsat4}
mmsat4frame <- as.data.frame(MMSAT4)

#RNA
RNA$tvsmmsat4 <- "Non-target"
RNA$tvsmmsat4[sub("\\..*", "", RNA$Gene) %in% mmsat4frame$gene_id.x] <- "MMsat4"
RNA$tvsmmsat4[sub("\\..*", "", RNA$Gene) %in% largetframe$geneID] <- "miR-181 target"
RNA$tvsmmsat4[sub("\\..*", "", RNA$Gene) %in% largetframe$geneID & sub("\\..*", "", RNA$Gene) %in% mmsat4frame$gene_id.x] <- "both"
table(RNA$tvsmmsat4)

#RPF
RPF$tvsmmsat4 <- "Non-target"
RPF$tvsmmsat4[sub("\\..*", "", RPF$Gene) %in% mmsat4frame$gene_id.x] <- "MMsat4"
RPF$tvsmmsat4[sub("\\..*", "", RPF$Gene) %in% largetframe$geneID] <- "miR-181 target"
RPF$tvsmmsat4[sub("\\..*", "", RPF$Gene) %in% largetframe$geneID & sub("\\..*", "", RPF$Gene) %in% mmsat4frame$gene_id.x] <- "both"
table(RPF$tvsmmsat4)

#MS
MS$tvsmmsat4 <- "Non-target"
MS$tvsmmsat4[MS$gene_symbol %in% mmsat4frame$gene_name] <- "MMsat4"
MS$tvsmmsat4[MS$gene_symbol %in% largetframe$geneID] <- "miR-181 target"
MS$tvsmmsat4[MS$gene_symbol %in% largetframe$geneID & MS$gene_symbol %in% mmsat4frame$gene_name] <- "both"
table(MS$tvsmmsat4)

#TE
TE$tvsmmsat4 <- "Non-target"
TE$tvsmmsat4[sub("\\..*", "", TE$Gene) %in% mmsat4frame$gene_id.x] <- "MMsat4"
TE$tvsmmsat4[sub("\\..*", "", TE$Gene) %in% largetframe$geneID] <- "miR-181 target"
TE$tvsmmsat4[sub("\\..*", "", TE$Gene) %in% largetframe$geneID & sub("\\..*", "", TE$Gene) %in% mmsat4frame$gene_id.x] <- "both"
table(TE$tvsmmsat4)



#RNA
tolECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(tvsmmsat4, levels = c("Non-target", "MMsat4", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA Mmsat4 vs target")

tolECDFRNA


#RPF
tolECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(tvsmmsat4, levels = c("Non-target", "MMsat4", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF Mmsat4 vs target")

tolECDFRPF

#MS
tolECDFMS <- ggplot(MS, aes(as.numeric(log2FoldChange), colour=factor(tvsmmsat4, levels = c("Non-target", "MMsat4", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("MS Mmsat4 vs target")

tolECDFMS

#TE
tolECDFTE <- ggplot(TE, aes(as.numeric(log2FoldChange), colour=factor(tvsmmsat4, levels = c("Non-target", "MMsat4", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("TE Mmsat4 vs target")

tolECDFTE

#export RNA
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MMSAT4ECDF_RNA.pdf", width=2, height = 2)
tolECDFRNA
dev.off()


#export RPF
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MMSAT4ECDF_RPF.pdf", width=2, height = 2)
tolECDFRPF
dev.off()

#export MS
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MMSAT4ECDF_MS.pdf", width=2, height = 2)
tolECDFMS
dev.off()

#export TE
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MMSAT4ECDF_TE.pdf", width=2, height = 2)
tolECDFTE
dev.off()


table(RPF$tvsmmsat4)
```

# MurSatRep1
```{r MurSatRep1}
MurSatRep1frame <- as.data.frame(MurSatRep1)

#RNA
RNA$tvsMurSatRep1 <- "Non-target"
RNA$tvsMurSatRep1[sub("\\..*", "", RNA$Gene) %in% MurSatRep1frame$gene_id.x] <- "MurSatRep1"
RNA$tvsMurSatRep1[sub("\\..*", "", RNA$Gene) %in% largetframe$geneID] <- "miR-181 target"
RNA$tvsMurSatRep1[sub("\\..*", "", RNA$Gene) %in% largetframe$geneID & sub("\\..*", "", RNA$Gene) %in% MurSatRep1frame$gene_id.x] <- "both"
table(RNA$tvsMurSatRep1)

#RPF
RPF$tvsMurSatRep1 <- "Non-target"
RPF$tvsMurSatRep1[sub("\\..*", "", RPF$Gene) %in% MurSatRep1frame$gene_id.x] <- "MurSatRep1"
RPF$tvsMurSatRep1[sub("\\..*", "", RPF$Gene) %in% largetframe$geneID] <- "miR-181 target"
RPF$tvsMurSatRep1[sub("\\..*", "", RPF$Gene) %in% largetframe$geneID & sub("\\..*", "", RPF$Gene) %in% MurSatRep1frame$gene_id.x] <- "both"
table(RPF$tvsMurSatRep1)

#MS
MS$tvsMurSatRep1 <- "Non-target"
MS$tvsMurSatRep1[MS$gene_symbol %in% MurSatRep1frame$gene_name] <- "MurSatRep1"
MS$tvsMurSatRep1[MS$gene_symbol %in% largetframe$geneID] <- "miR-181 target"
MS$tvsMurSatRep1[MS$gene_symbol %in% largetframe$geneID & MS$gene_symbol %in% MurSatRep1frame$gene_name] <- "both"
table(MS$tvsMurSatRep1)

#TE
TE$tvsMurSatRep1 <- "Non-target"
TE$tvsMurSatRep1[sub("\\..*", "", TE$Gene) %in% MurSatRep1frame$gene_id.x] <- "MurSatRep1"
TE$tvsMurSatRep1[sub("\\..*", "", TE$Gene) %in% largetframe$geneID] <- "miR-181 target"
TE$tvsMurSatRep1[sub("\\..*", "", TE$Gene) %in% largetframe$geneID & sub("\\..*", "", TE$Gene) %in% MurSatRep1frame$gene_id.x] <- "both"
table(TE$tvsMurSatRep1)


#RNA
murECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(tvsMurSatRep1, levels = c("Non-target", "MurSatRep1", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA MurSatRep1 vs target")

murECDFRNA


#RPF
murECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(tvsMurSatRep1, levels = c("Non-target", "MurSatRep1", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF MurSatRep1 vs target")

murECDFRPF

#MS
murECDFMS <- ggplot(MS, aes(as.numeric(log2FoldChange), colour=factor(tvsMurSatRep1, levels = c("Non-target", "MurSatRep1", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("MS MurSatRep1 vs target")

murECDFMS

#TE
murECDFTE <- ggplot(TE, aes(as.numeric(log2FoldChange), colour=factor(tvsMurSatRep1, levels = c("Non-target", "MurSatRep1", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("TE MurSatRep1 vs target")

murECDFTE



#export RNA
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/mursatrep1ECDF_RNA.pdf", width=2, height = 2)
murECDFRNA
dev.off()

#export RPF
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/mursatrep1ECDF_RPF.pdf", width=2, height = 2)
murECDFRPF
dev.off()

#export MS
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/mursatrep1ECDF_MS.pdf", width=2, height = 2)
murECDFMS
dev.off()

#export TE
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/mursatrep1ECDF_TE.pdf", width=2, height = 2)
murECDFTE
dev.off()


table(RPF$tvsMurSatRep1)
```


## metagene plot
```{r metageneplot}
MMSAT4$id <- "MMsat4"
MurSatRep1$id <- "MurSatRep1"

metaob <- c(MMSAT4, MurSatRep1)

metael <- metaGeneProfile(object = metaob, annotation = gff23path, group = "id")
metasat4 <- metaGeneProfile(object = MMSAT4, annotation = gff23path)
metamur <- metaGeneProfile(object = MurSatRep1, annotation = gff23path)

metasat4
metamur

metael


pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/metagene_profile_both.pdf", width=2, height = 2)
metael
dev.off()


# metagene plot that only contains genes that also have Mres 

mmsat4MRE <- subsetByOverlaps(MMSAT4 , makeGRangesFromDataFrame(mmsat4frame))
MurSatRep1MRE <- subsetByOverlaps(MurSatRep1 , makeGRangesFromDataFrame(MurSatRep1frame))

mmsat4MRE$id <- "MMsat4"
MurSatRep1MRE$id <- "MurSatRep1"

metaob2 <- c(mmsat4MRE, MurSatRep1MRE)

metael2 <- metaGeneProfile(object = metaob2, annotation = gff23path, group = "id")
metael2

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/metagene_profile_both_onlywithMRE.pdf", width=2, height = 2)
metael2
dev.off()

```



## venn diagram
```{r vennmmsat4}

vset <- list(mmsat4frame[!duplicated(mmsat4frame$gene_id.x) & mmsat4frame$gene_id.x %in% largetframe$geneID,"gene_id.x"], MurSatRep1frame[!duplicated(MurSatRep1frame$gene_id.x) & MurSatRep1frame$gene_id.x %in% largetframe$geneID,"gene_id.x"])
names(vset) <- c("MMSAT4", "MurSatRep1")

vplot <- plot(euler(vset, shape = "ellipse"), quantities = TRUE, main="Genes by contained motifes" , fills=c(farbe1, farbe2)) 


pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/Venn_motifes.pdf", width=2, height = 2)
vplot
dev.off()


vsetECDF <- list(mmsat4frame[!duplicated(mmsat4frame$gene_id.x) & mmsat4frame$gene_id.x %in% largetframe$geneID & mmsat4frame$gene_id.x %in% sub("\\..*", "", RNA$Gene) & mmsat4frame$gene_id.x %in% sub("\\..*", "", RPF$Gene) ,"gene_id.x"], MurSatRep1frame[!duplicated(MurSatRep1frame$gene_id.x) & MurSatRep1frame$gene_id.x %in% largetframe$geneID & MurSatRep1frame$gene_id.x %in% sub("\\..*", "", RNA$Gene) & MurSatRep1frame$gene_id.x %in% sub("\\..*", "", RPF$Gene),"gene_id.x"])
names(vsetECDF) <- c("MMSAT4", "MurSatRep1")

vplotecdf <- plot(euler(vsetECDF, shape = "ellipse"), quantities = TRUE, main="Genes by contained motifes" , fills=c(farbe1, farbe2)) 


pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/Venn_motifes_likeECDF.pdf", width=2, height = 2)
vplotecdf
dev.off()

```


# histogram MREs per gene with repeat
```{r mresPerGeneRep}
histmremmsat4 <- as.data.frame(table(largetframe[largetframe$geneID %in% mmsat4frame$gene_id.x, "geneID"]))
dim(histmremmsat4)

mmsat4GMRElist <- as.data.frame(table(largetframe[largetframe$geneID %in% mmsat4frame$gene_id.x, "geneName"]))
colnames(mmsat4GMRElist) = c("Gene", "Number of MREs")
write.csv(mmsat4GMRElist, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/genelist_MMsat4withMRE.csv")

histmremmsat4P <- ggplot(histmremmsat4, aes(x=Freq)) + 
  geom_histogram(binwidth = 1, colour="black", fill="#B4B4B4") +
  theme_paper() +
  scale_y_continuous(expand = c(0,0))

histmremmsat4P

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MRE_histogram_mres_per_MMSAT4.pdf", width=2, height = 2)
histmremmsat4P
dev.off()

histmremursat <- as.data.frame(table(largetframe[largetframe$geneID %in% MurSatRep1frame$gene_id.x, "geneID"]))
dim(histmremursat)

mursatrep1GMRElist <- as.data.frame(table(largetframe[largetframe$geneID %in% MurSatRep1frame$gene_id.x, "geneName"]))
colnames(mursatrep1GMRElist) = c("Gene", "Number of MREs")
write.csv(mursatrep1GMRElist, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/genelist_MursatRep1withMRE.csv")

histmremursatP<- ggplot(histmremursat, aes(x=Freq)) + 
  geom_histogram(binwidth = 1, colour="black", fill="#B4B4B4") +
  theme_paper() +
  scale_y_continuous(expand = c(0,0))

histmremursatP

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MRE_histogram_mres_per_MurSatRep1.pdf", width=2, height = 2)
histmremursatP
dev.off()

#like ecdf (occurs in RNA and RPF)
histmremmsat4ECDF <- as.data.frame(table(largetframe[largetframe$geneID %in% mmsat4frame$gene_id.x & largetframe$geneID %in% sub("\\..*", "", RNA$Gene) & largetframe$geneID %in% sub("\\..*", "", RPF$Gene), "geneID"]))
dim(histmremmsat4ECDF)

histmremmsat4ECDFP <- ggplot(histmremmsat4ECDF, aes(x=Freq)) + 
  geom_histogram(binwidth = 1, colour="black", fill="#B4B4B4") +
  theme_paper() +
  scale_y_continuous(expand = c(0,0))

histmremmsat4ECDFP

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MRE_histogram_mres_per_MMSAT4_likeECDF.pdf", width=2, height = 2)
histmremmsat4ECDFP
dev.off()

histmremursatECDF <- as.data.frame(table(largetframe[largetframe$geneID %in% MurSatRep1frame$gene_id.x & largetframe$geneID %in% sub("\\..*", "", RNA$Gene) & largetframe$geneID %in% sub("\\..*", "", RPF$Gene), "geneID"]))
dim(histmremursatECDF)


histmremursatECDFP<- ggplot(histmremursatECDF, aes(x=Freq)) + 
  geom_histogram(binwidth = 1, colour="black", fill="#B4B4B4") +
  theme_paper() +
  scale_y_continuous(expand = c(0,0))

histmremursatECDFP

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MRE_histogram_mres_per_MurSatRep1_likeECDF.pdf", width=2, height = 2)
histmremursatECDFP
dev.off()

```





## Sequence overlap mmsat4 mursatrep1
Just check to make sure these are not the same
```{r seqol}
olmmsat4 <- subsetByOverlaps(MMSAT4, MurSatRep1)
olmursatrep1 <- subsetByOverlaps(MurSatRep1, MMSAT4)


#make pie charts
pimmsat <- data.frame(c("MMsat4", "Overlap"),
                      c(length(MMSAT4)-length(olmmsat4), length(olmmsat4)))
colnames(pimmsat) <- c("Group", "Count")

pimursatrep1 <- data.frame(c("MurSatRep1", "Overlap"),
                      c(length(MurSatRep1)-length(olmursatrep1), length(olmursatrep1)))
colnames(pimursatrep1) <- c("Group", "Count")

pimmsatP <- ggplot(pimmsat, aes(x="", y=Count, fill=Group)) + 
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  geom_text(aes(label = Count),
            position = position_stack(vjust = 0.5)) 

pimmsatP

pimursatrep1P <- ggplot(pimursatrep1, aes(x="", y=Count, fill=Group)) + 
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  geom_text(aes(label = Count),
            position = position_stack(vjust = 0.5)) 

pimursatrep1P

```


# MAplots of targets with satellites
```{r masat}
#mmsat4 RNA
MAmmsat4RNA <- ggplot(RNA, aes(x=log10(baseMean), y=log2FoldChange, fill=factor(tvsmmsat4, levels = c("Non-target", "miR-181 target", "MMsat4", "both"))), colour="black") + 
  geom_point(shape=21, size=1) + 
  scale_fill_manual(values = c(farbeneg, farbe1, farbe2, farbe3)) +
  geom_point(data = RNA[RNA$tvsmmsat4=="Non-target",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour=farbeneg, fill=farbeneg, size=1) +
  geom_point(data = RNA[RNA$tvsmmsat4=="miR-181 target",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe1, size=2) +
  geom_point(data = RNA[RNA$tvsmmsat4=="MMsat4",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe2, size=2) +
  geom_point(data = RNA[RNA$tvsmmsat4=="both",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe3, size=3) +
  geom_hline(yintercept = 0, linetype="dashed", colour="black")+
  theme_paper() +
  coord_cartesian(ylim = c(-2,2), xlim = c(1,5))
MAmmsat4RNA

##mmsat4 RPF
MAmmsat4RPF <- ggplot(RPF, aes(x=log10(baseMean), y=log2FoldChange, fill=factor(tvsmmsat4, levels = c("Non-target", "miR-181 target", "MMsat4", "both"))), colour="black") + 
  geom_point(shape=21, size=1) + 
  scale_fill_manual(values = c(farbeneg, farbe1, farbe2, farbe3)) +
  geom_point(data = RPF[RPF$tvsmmsat4=="Non-target",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour=farbeneg, fill=farbeneg, size=1) +
  geom_point(data = RPF[RPF$tvsmmsat4=="miR-181 target",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe1, size=2) +
  geom_point(data = RPF[RPF$tvsmmsat4=="MMsat4",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe2, size=2) +
  geom_point(data = RPF[RPF$tvsmmsat4=="both",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe3, size=3) +
  geom_hline(yintercept = 0, linetype="dashed", colour="black")+
  theme_paper() +
  coord_cartesian(ylim = c(-2,2), xlim = c(1,5))
MAmmsat4RPF

#Mursatrep1 RNA
MAmursatrep1RNA <- ggplot(RNA, aes(x=log10(baseMean), y=log2FoldChange, fill=factor(tvsMurSatRep1, levels = c("Non-target", "miR-181 target", "MurSatRep1", "both"))), colour="black") + 
  geom_point(shape=21, size=1) + 
  scale_fill_manual(values = c(farbeneg, farbe1, farbe2, farbe3)) +
  geom_point(data = RNA[RNA$tvsMurSatRep1=="Non-target",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour=farbeneg, fill=farbeneg, size=1) +
  geom_point(data = RNA[RNA$tvsMurSatRep1=="miR-181 target",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe1, size=2) +
  geom_point(data = RNA[RNA$tvsMurSatRep1=="MurSatRep1",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe2, size=2) +
  geom_point(data = RNA[RNA$tvsMurSatRep1=="both",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe3, size=3) +
  geom_hline(yintercept = 0, linetype="dashed", colour="black")+
  theme_paper() +
  coord_cartesian(ylim = c(-2,2), xlim = c(1,5))
MAmursatrep1RNA

#Mursatrep1 RPF
MAmursatrep1RPF <- ggplot(RPF, aes(x=log10(baseMean), y=log2FoldChange, fill=factor(tvsMurSatRep1, levels = c("Non-target", "miR-181 target", "MurSatRep1", "both"))), colour="black") + 
  geom_point(shape=21, size=1) + 
  scale_fill_manual(values = c(farbeneg, farbe1, farbe2, farbe3)) +
  geom_point(data = RPF[RPF$tvsMurSatRep1=="Non-target",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour=farbeneg, fill=farbeneg, size=1) +
  geom_point(data = RPF[RPF$tvsMurSatRep1=="miR-181 target",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe1, size=2) +
  geom_point(data = RPF[RPF$tvsMurSatRep1=="MurSatRep1",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe2, size=2) +
  geom_point(data = RPF[RPF$tvsMurSatRep1=="both",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe3, size=3) +
  geom_hline(yintercept = 0, linetype="dashed", colour="black")+
  theme_paper() +
  coord_cartesian(ylim = c(-2,2), xlim = c(1,5))
MAmursatrep1RPF


```


# Session info
```{r sessioninfo}
sessionInfo()
```






