---
title: "Figure4_v2"
author:
- name: Nikita Verheyden
  affiliation: JLU Giessen
  date: "`r Sys.Date()`"
output:
  BiocStyle::html_document:
      toc: TRUE
      toc_float: TRUE
      code_folding: hide

header-includes:
- \makeatletter\renewcommand*{\fps@figure}{h}\makeatother
- \usepackage{placeins}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```



# Setup

dir
```{r dir}
setwd("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4")
```


## Packages
```{r packages}
source("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Supporting_scripts/themes/theme_paper.R")
library(ggplot2)
library(rtracklayer)
library(dplyr)
library(tibble)
library(Rtsne)
library(cliProfiler)
library(GenomicRanges)
library(cowplot)
library(eulerr)
library(readxl)
library(biomaRt)
library(writexl)

```

## Data
```{r data}
#Ribo profiling
RNA <- read.csv("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/RNA_masterframe.csv")
RPF <- read.csv("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/RPF_masterframe.csv")

#load the gtf file to compare genes
gff23 <- import.gff3("C:/Users/nikit/Documents/Krueger_Lab/Ribo_Profiling/run15112022M23/ref_genome/gencode.vM23.annotation.gff3.gz")
gff23path <- file.path("C:/Users/nikit/Documents/Krueger_Lab/Ribo_Profiling/run15112022M23/ref_genome", "gencode.vM23.annotation.gff3.gz")

#targets
larget <- readRDS("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/mir181_bs_with_seeds_transcripts.rds")
largetframe <- as.data.frame(larget)
table(largetframe$set)
# largetframe <- largetframe[!(largetframe$set == "ago_bs_mir181_chi"),]

largetframe <- largetframe %>%
  subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched"))


# modify dataset to only include working targetsets

#only use miR181_enriched und both
# largetframe <- largetframe[largetframe$set %in% c("mir181_enriched", "ago_bs_mir181_chi&mir181_enriched"),]


#ms
#ms
MS <- read.csv("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure2/MS_data_MMU_02082023.csv" )

#Translational efficiency
TE <- read.csv("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Supporting_scripts/deltaTE/TE_m23_genesInRNAandRPF.csv")




#MMsat4
# repeat_masker <- readRDS("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure2/MMsat4/repeat_masker.rds")
# MMSAT4 <- repeat_masker[repeat_masker$repName == "MMSAT4"]
# MurSatRep1 <- repeat_masker[repeat_masker$repName == "MurSatRep1"]


#melina repeats
melrep <- readRDS("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/rep_on_transcripts.rds")
MMSAT4 <- melrep[melrep$repName == "MMSAT4"]
MurSatRep1 <- melrep[melrep$repName == "MurSatRep1"]
```


## colours
```{r colours}
#colours
farbeneg <- "#b4b4b4"
farbe1 <- "#0073C2FF"
farbe2 <- "#EFC000FF"
farbe3 <- "#CD534CFF"
farbe4 <- "#7AA6DCFF"
farbe5 <- "#868686FF"
farbe6 <- "#003C67FF"
farbe7 <- "#8F7700FF"
farbe8 <- "#3B3B3BFF"
farbe9 <- "#A73030FF"
farbe10 <- "#4A6990FF"
farbe11 <- "#FF6F00FF"
farbe12 <- "#C71000FF"
farbe13 <- "#008EA0FF"
farbe14 <- "#8A4198FF"
farbe15 <- "#5A9599FF"
farbe16 <- "#FF6348FF"
  
RNApcol <- "#b56504"
RNAncol <- "#027d73"
RPFpcol <- "#c4c404"
RPFncol <- "#8d0391"
```

## axis limits
```{r axislim}
xlim <- 0.5
zoomfac <- 0.45
```


## number of binding sites 
```{r num}
bsnum <- as.data.frame(table(largetframe$geneName))
colnames(bsnum) <- c("gene_symbol", "BS_number")
```


# wobble plot

## ecdf
```{r wobbleecdf}
#RNA
RNA$wobble <- "Non-target"
RNA$wobble[RNA$gene_symbol %in% largetframe[is.na(largetframe$first_seed_200down.wobble) ,"geneName"]] <- "No seed"
RNA$wobble[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == FALSE ,"geneName"]] <- "Canonical"
RNA$wobble[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == TRUE ,"geneName"]] <- "Non-Canonical"
RNA$wobble[RNA$gene_symbol %in% bsnum[bsnum$BS_number > 1, "gene_symbol"]] <- "Multiple"
#RPF
RPF$wobble <- "Non-target"
RPF$wobble[RPF$gene_symbol %in% largetframe[is.na(largetframe$first_seed_200down.wobble) ,"geneName"]] <- "No seed"
RPF$wobble[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == FALSE ,"geneName"]] <- "Canonical"
RPF$wobble[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == TRUE ,"geneName"]] <- "Non-Canonical"
RPF$wobble[RPF$gene_symbol %in% bsnum[bsnum$BS_number > 1, "gene_symbol"]] <- "Multiple"

#MS
MS$wobble <- "Non-target"
MS$wobble[MS$gene_symbol %in% largetframe[is.na(largetframe$first_seed_200down.wobble) ,"geneName"]] <- "No seed"
MS$wobble[MS$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == FALSE ,"geneName"]] <- "Canonical"
MS$wobble[MS$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == TRUE ,"geneName"]] <- "Non-Canonical"
MS$wobble[MS$gene_symbol %in% bsnum[bsnum$BS_number > 1, "gene_symbol"]] <- "Multiple"

#TE
TE$wobble <- "Non-target"
TE$wobble[TE$gene_symbol %in% largetframe[is.na(largetframe$first_seed_200down.wobble) ,"geneName"]] <- "No seed"
TE$wobble[TE$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == FALSE ,"geneName"]] <- "Canonical"
TE$wobble[TE$gene_symbol %in% largetframe[largetframe$first_seed_200down.wobble == TRUE ,"geneName"]] <- "Non-Canonical"
TE$wobble[TE$gene_symbol %in% bsnum[bsnum$BS_number > 1, "gene_symbol"]] <- "Multiple"

#ECDF

#RNA
wobbleecdfRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Non-Canonical", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbeneg, farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Target genes RNAseq")

wobbleecdfRNA

#zoom
wobbleecdfRNAZ <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Non-Canonical", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbeneg, farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(),
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")


wobbleecdfRNAZ

#RPF
wobbleecdfRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Non-Canonical", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbeneg, farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Target genes ribosome profiling")

wobbleecdfRPF

#zoom
wobbleecdfRPFZ <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Non-Canonical", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbeneg, farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(),
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")


wobbleecdfRPFZ

#MS
wobbleecdfMS <- ggplot(MS, aes(as.numeric(log2FoldChange), colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Non-Canonical", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbeneg, farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Target genes mass spectrometry")

wobbleecdfMS

#zoom
wobbleecdfMSZ <- ggplot(MS, aes(as.numeric(log2FoldChange), colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Non-Canonical", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbeneg, farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(),
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")


wobbleecdfMSZ

#TE
wobbleecdfTE <- ggplot(TE, aes(as.numeric(log2FoldChange), colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Non-Canonical", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbeneg, farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange KO vs WT") +
  ggtitle("Target genes translational efficiency")

wobbleecdfTE

#zoom
wobbleecdfTEZ <- ggplot(TE, aes(as.numeric(log2FoldChange), colour=factor(wobble, levels = c("Non-target", "No seed", "Canonical", "Non-Canonical", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbeneg, farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(),
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")


wobbleecdfTEZ


#export RNA
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/wobbleECDF_RNA.pdf", width=2, height = 2)
wobbleecdfRNA
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/wobbleECDF_RNA_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
wobbleecdfRNAZ
dev.off()

#export RPF
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/wobbleECDF_RPF.pdf", width=2, height = 2)
wobbleecdfRPF
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/wobbleECDF_RPF_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
wobbleecdfRPFZ
dev.off()

#export MS
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/wobbleECDF_MS.pdf", width=2, height = 2)
wobbleecdfMS
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/wobbleECDF_MS_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
wobbleecdfMSZ
dev.off()

#export TE
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/wobbleECDF_TE.pdf", width=2, height = 2)
wobbleecdfTE
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/wobbleECDF_TE_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
wobbleecdfTEZ
dev.off()


table(RPF$wobble)
```
# TE density wobble
```{R TEdensitywobble}
TEdenswob <- ggplot(TE, aes(x=log2FoldChange, colour=factor(wobble, levels = c("No seed", "Canonical", "Non-Canonical", "Multiple", "Non-target")))) +
  geom_density() +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c( farbeneg, farbe1, farbe2, farbe3, "black")) +
  theme_paper() +
  coord_cartesian(xlim = c(-1,1)) +
  ggtitle("TE")

TEdenswob

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/WobbleDensity_TE.pdf", width=2, height = 2)
TEdenswob
dev.off()

```

## Significance
```{r significanceWobble}
RNAWobbleTTest1 <- t.test(RNA[RNA$wobble=="No seed", "log2FoldChange"], RNA[RNA$wobble=="Non-target", "log2FoldChange"])
RNAWobbleTTest2 <- t.test(RNA[RNA$wobble=="Canonical", "log2FoldChange"], RNA[RNA$wobble=="Non-target", "log2FoldChange"])
RNAWobbleTTest3 <- t.test(RNA[RNA$wobble=="Non-Canonical", "log2FoldChange"], RNA[RNA$wobble=="Non-target", "log2FoldChange"])

RPFWobbleTTest1 <- t.test(RPF[RPF$wobble=="No seed", "log2FoldChange"], RPF[RPF$wobble=="Non-target", "log2FoldChange"])
RPFWobbleTTest2 <- t.test(RPF[RPF$wobble=="Canonical", "log2FoldChange"], RPF[RPF$wobble=="Non-target", "log2FoldChange"])
RPFWobbleTTest3 <- t.test(RPF[RPF$wobble=="Non-Canonical", "log2FoldChange"], RPF[RPF$wobble=="Non-target", "log2FoldChange"])

TEWobbleTTest1 <- t.test(TE[TE$wobble=="No seed", "log2FoldChange"], TE[TE$wobble=="Non-target", "log2FoldChange"])
TEWobbleTTest2 <- t.test(TE[TE$wobble=="Canonical", "log2FoldChange"], TE[TE$wobble=="Non-target", "log2FoldChange"])
TEWobbleTTest3 <- t.test(TE[TE$wobble=="Non-Canonical", "log2FoldChange"], TE[TE$wobble=="Non-target", "log2FoldChange"])

MSWobbleTTest1 <- t.test(MS[MS$wobble=="No seed", "log2FoldChange"], MS[MS$wobble=="Non-target", "log2FoldChange"])
MSWobbleTTest2 <- t.test(MS[MS$wobble=="Canonical", "log2FoldChange"], MS[MS$wobble=="Non-target", "log2FoldChange"])
MSWobbleTTest3 <- t.test(MS[MS$wobble=="Non-Canonical", "log2FoldChange"], MS[MS$wobble=="Non-target", "log2FoldChange"])

#kolmogorov-smirnov
RNAWobbleksTest1 <- ks.test(RNA[RNA$wobble=="No seed", "log2FoldChange"], RNA[RNA$wobble=="Non-target", "log2FoldChange"])
RNAWobbleksTest2 <- ks.test(RNA[RNA$wobble=="Canonical", "log2FoldChange"], RNA[RNA$wobble=="Non-target", "log2FoldChange"])
RNAWobbleksTest3 <- ks.test(RNA[RNA$wobble=="Non-Canonical", "log2FoldChange"], RNA[RNA$wobble=="Non-target", "log2FoldChange"])

RPFWobbleksTest1 <- ks.test(RPF[RPF$wobble=="No seed", "log2FoldChange"], RPF[RPF$wobble=="Non-target", "log2FoldChange"])
RPFWobbleksTest2 <- ks.test(RPF[RPF$wobble=="Canonical", "log2FoldChange"], RPF[RPF$wobble=="Non-target", "log2FoldChange"])
RPFWobbleksTest3 <- ks.test(RPF[RPF$wobble=="Non-Canonical", "log2FoldChange"], RPF[RPF$wobble=="Non-target", "log2FoldChange"])

TEWobbleksTest1 <- ks.test(TE[TE$wobble=="No seed", "log2FoldChange"], TE[TE$wobble=="Non-target", "log2FoldChange"])
TEWobbleksTest2 <- ks.test(TE[TE$wobble=="Canonical", "log2FoldChange"], TE[TE$wobble=="Non-target", "log2FoldChange"])
TEWobbleksTest3 <- ks.test(TE[TE$wobble=="Non-Canonical", "log2FoldChange"], TE[TE$wobble=="Non-target", "log2FoldChange"])

MSWobbleksTest1 <- ks.test(MS[MS$wobble=="No seed", "log2FoldChange"], MS[MS$wobble=="Non-target", "log2FoldChange"])
MSWobbleksTest2 <- ks.test(MS[MS$wobble=="Canonical", "log2FoldChange"], MS[MS$wobble=="Non-target", "log2FoldChange"])
MSWobbleksTest3 <- ks.test(MS[MS$wobble=="Non-Canonical", "log2FoldChange"], MS[MS$wobble=="Non-target", "log2FoldChange"])


Wobblesig <- data.frame(c("RNA", "RNA", "RNA", "RPF", "RPF", "RPF", "TE", "TE", "TE", "MS", "MS", "MS"),
                        c("No seed", "Canonical", "Non-Canonical", "No seed", "Canonical", "Non-Canonical",
                          "No seed", "Canonical", "Non-Canonical","No seed", "Canonical", "Non-Canonical"),
                        c(RNAWobbleTTest1$p.value, RNAWobbleTTest2$p.value, RNAWobbleTTest3$p.value,
                          RPFWobbleTTest1$p.value, RPFWobbleTTest2$p.value, RPFWobbleTTest3$p.value,
                          TEWobbleTTest1$p.value, TEWobbleTTest2$p.value, TEWobbleTTest3$p.value, 
                          MSWobbleTTest1$p.value, MSWobbleTTest2$p.value, MSWobbleTTest3$p.value),
                        c(RNAWobbleksTest1$p.value, RNAWobbleksTest2$p.value, RNAWobbleksTest3$p.value,
                          RPFWobbleksTest1$p.value, RPFWobbleksTest2$p.value, RPFWobbleksTest3$p.value, 
                          TEWobbleksTest1$p.value, TEWobbleksTest2$p.value, TEWobbleksTest3$p.value,
                          MSWobbleksTest1$p.value, MSWobbleksTest2$p.value, MSWobbleksTest3$p.value))
colnames(Wobblesig) <- c("Experiment", "Subset", "Welch Two Sample t-test", "Asymptotic two-sample Kolmogorov-Smirnov test")
Wobblesig

#export
write_xlsx(Wobblesig, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/Wobblesignificance.xlsx")
```


## wobble bar
```{r wobblebar}
wobar <- as.data.frame(table(largetframe$first_seed_200down.wobble))
wobar$y <- "MREs"
wobarp <- ggplot(wobar, aes(y=y, x=Freq, fill=Var1)) + geom_bar(stat = "identity", position = "stack") +
  theme_paper()

wobarp

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/Wobblebar.pdf", width = 2, height = 1)
wobarp
dev.off()
```


## by type of MRE (wobble and non wobble combined)
```{r MREtype}
#RNA
RNA$MREtype <- "Non-target"
RNA$MREtype[RNA$gene_symbol %in% largetframe[is.na(largetframe$first_seed_200down.type),"geneName"]] <- "No seed"
RNA$MREtype[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_6mer","geneName"]] <- "seed_6mer"
RNA$MREtype[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_a1","geneName"]] <- "seed_7mer_a1"
RNA$MREtype[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_m8","geneName"]] <- "seed_7mer_m8"
RNA$MREtype[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_8mer","geneName"]] <- "seed_8mer"
RNA$MREtype[RNA$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_alt_38","geneName"]] <- "seed_alt_38"
RNA$MREtype[RNA$gene_symbol %in% bsnum[bsnum$BS_number > 1, "gene_symbol"]] <- "Multiple"


#RPF
RPF$MREtype <- "Non-target"
RPF$MREtype[RPF$gene_symbol %in% largetframe[is.na(largetframe$first_seed_200down.type),"geneName"]] <- "No seed"
RPF$MREtype[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_6mer","geneName"]] <- "seed_6mer"
RPF$MREtype[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_a1","geneName"]] <- "seed_7mer_a1"
RPF$MREtype[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_m8","geneName"]] <- "seed_7mer_m8"
RPF$MREtype[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_8mer","geneName"]] <- "seed_8mer"
RPF$MREtype[RPF$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_alt_38","geneName"]] <- "seed_alt_38"
RPF$MREtype[RPF$gene_symbol %in% bsnum[bsnum$BS_number > 1, "gene_symbol"]] <- "Multiple"


#MS
MS$MREtype <- "Non-target"
MS$MREtype[MS$gene_symbol %in% largetframe[is.na(largetframe$first_seed_200down.type),"geneName"]] <- "No seed"
MS$MREtype[MS$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_6mer","geneName"]] <- "seed_6mer"
MS$MREtype[MS$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_a1","geneName"]] <- "seed_7mer_a1"
MS$MREtype[MS$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_m8","geneName"]] <- "seed_7mer_m8"
MS$MREtype[MS$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_8mer","geneName"]] <- "seed_8mer"
MS$MREtype[MS$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_alt_38","geneName"]] <- "seed_alt_38"
MS$MREtype[MS$gene_symbol %in% bsnum[bsnum$BS_number > 1, "gene_symbol"]] <- "Multiple"


#TE
TE$MREtype <- "Non-target"
TE$MREtype[TE$gene_symbol %in% largetframe[is.na(largetframe$first_seed_200down.type),"geneName"]] <- "No seed"
TE$MREtype[TE$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_6mer","geneName"]] <- "seed_6mer"
TE$MREtype[TE$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_a1","geneName"]] <- "seed_7mer_a1"
TE$MREtype[TE$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_7mer_m8","geneName"]] <- "seed_7mer_m8"
TE$MREtype[TE$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_8mer","geneName"]] <- "seed_8mer"
TE$MREtype[TE$gene_symbol %in% largetframe[largetframe$first_seed_200down.type == "seed_alt_38","geneName"]] <- "seed_alt_38"
TE$MREtype[TE$gene_symbol %in% bsnum[bsnum$BS_number > 1, "gene_symbol"]] <- "Multiple"



# ecdf plots
#RNA
typeECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA type of MRE")

typeECDFRNA


# zoom
typeECDFRNAZ <- ggplot(RNA, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(),
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")

typeECDFRNAZ

#RPF
typeECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF type of MRE")

typeECDFRPF

# zoom
typeECDFRPFZ <- ggplot(RPF, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(),
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")

typeECDFRPFZ


#MS
typeECDFMS <- ggplot(MS, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("MS type of MRE")

typeECDFMS

# zoom
typeECDFMSZ <- ggplot(MS, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(),
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")

typeECDFMSZ


#TE
typeECDFTE <- ggplot(TE, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("TE type of MRE")

typeECDFTE

# zoom
typeECDFTEZ <- ggplot(TE, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7)) +
  coord_cartesian(xlim = c(-(xlim/8), (xlim/8)), ylim = c(0.5-1/16, 0.5+1/16)) + 
  theme_bw() +
  theme(legend.position = "none", axis.text = element_blank(), panel.grid = element_blank(),
        panel.background = element_rect(fill='transparent'), #transparent panel bg
        plot.background = element_rect(fill='transparent', color=NA)) +
  scale_y_continuous("") + scale_x_continuous("")

typeECDFTEZ


#export RNA
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/typeECDF_RNA.pdf", width=2, height = 2)
typeECDFRNA
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/typeECDF_RNA_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
typeECDFRNAZ
dev.off()

#export RPF
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/typeECDF_RPF.pdf", width=2, height = 2)
typeECDFRPF
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/typeECDF_RPF_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
typeECDFRPFZ
dev.off()

#export MS
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/typeECDF_MS.pdf", width=2, height = 2)
typeECDFMS
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/typeECDF_MS_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
typeECDFMSZ
dev.off()

#export TE
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/typeECDF_TE.pdf", width=2, height = 2)
typeECDFTE
dev.off()
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/typeECDF_TE_zoom.pdf", width=2*zoomfac, height = 2*zoomfac)
typeECDFTEZ
dev.off()


table(RPF$MREtype)
```
# TE density type
```{R TEdensitytype}
TEdenstyp <- ggplot(TE, aes(x=log2FoldChange, colour=factor(MREtype, levels = c("No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", "Multiple", "Non-target")))) +
  geom_density() +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c( farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, "black")) +
  theme_paper() +
  coord_cartesian(xlim = c(-1,1)) +
  ggtitle("TE")

TEdenstyp

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/TypeDensity_TE.pdf", width=2, height = 2)
TEdenstyp
dev.off()

```

## Significance
```{r significanceType}
RNATypeTTest1 <- t.test(RNA[RNA$MREtype=="seed_6mer", "log2FoldChange"], RNA[RNA$MREtype=="Non-target", "log2FoldChange"])
RNATypeTTest2 <- t.test(RNA[RNA$MREtype=="seed_7mer_a1", "log2FoldChange"], RNA[RNA$MREtype=="Non-target", "log2FoldChange"])
RNATypeTTest3 <- t.test(RNA[RNA$MREtype=="seed_7mer_m8", "log2FoldChange"], RNA[RNA$MREtype=="Non-target", "log2FoldChange"])
RNATypeTTest4 <- t.test(RNA[RNA$MREtype=="seed_8mer", "log2FoldChange"], RNA[RNA$MREtype=="Non-target", "log2FoldChange"])
RNATypeTTest5 <- t.test(RNA[RNA$MREtype=="seed_alt_38", "log2FoldChange"], RNA[RNA$MREtype=="Non-target", "log2FoldChange"])

RPFTypeTTest1 <- t.test(RPF[RPF$MREtype=="seed_6mer", "log2FoldChange"], RPF[RPF$MREtype=="Non-target", "log2FoldChange"])
RPFTypeTTest2 <- t.test(RPF[RPF$MREtype=="seed_7mer_a1", "log2FoldChange"], RPF[RPF$MREtype=="Non-target", "log2FoldChange"])
RPFTypeTTest3 <- t.test(RPF[RPF$MREtype=="seed_7mer_m8", "log2FoldChange"], RPF[RPF$MREtype=="Non-target", "log2FoldChange"])
RPFTypeTTest4 <- t.test(RPF[RPF$MREtype=="seed_8mer", "log2FoldChange"], RPF[RPF$MREtype=="Non-target", "log2FoldChange"])
RPFTypeTTest5 <- t.test(RPF[RPF$MREtype=="seed_alt_38", "log2FoldChange"], RPF[RPF$MREtype=="Non-target", "log2FoldChange"])

TETypeTTest1 <- t.test(TE[TE$MREtype=="seed_6mer", "log2FoldChange"], TE[TE$MREtype=="Non-target", "log2FoldChange"])
TETypeTTest2 <- t.test(TE[TE$MREtype=="seed_7mer_a1", "log2FoldChange"], TE[TE$MREtype=="Non-target", "log2FoldChange"])
TETypeTTest3 <- t.test(TE[TE$MREtype=="seed_7mer_m8", "log2FoldChange"], TE[TE$MREtype=="Non-target", "log2FoldChange"])
TETypeTTest4 <- t.test(TE[TE$MREtype=="seed_8mer", "log2FoldChange"], TE[TE$MREtype=="Non-target", "log2FoldChange"])
TETypeTTest5 <- t.test(TE[TE$MREtype=="seed_alt_38", "log2FoldChange"], TE[TE$MREtype=="Non-target", "log2FoldChange"])

MSTypeTTest1 <- t.test(MS[MS$MREtype=="seed_6mer", "log2FoldChange"], MS[MS$MREtype=="Non-target", "log2FoldChange"])
MSTypeTTest2 <- t.test(MS[MS$MREtype=="seed_7mer_a1", "log2FoldChange"], MS[MS$MREtype=="Non-target", "log2FoldChange"])
MSTypeTTest3 <- t.test(MS[MS$MREtype=="seed_7mer_m8", "log2FoldChange"], MS[MS$MREtype=="Non-target", "log2FoldChange"])
MSTypeTTest4 <- t.test(MS[MS$MREtype=="seed_8mer", "log2FoldChange"], MS[MS$MREtype=="Non-target", "log2FoldChange"])
MSTypeTTest5 <- t.test(MS[MS$MREtype=="seed_alt_38", "log2FoldChange"], MS[MS$MREtype=="Non-target", "log2FoldChange"])

#kolmogorov-smirnov
RNATypeksTest1 <- ks.test(RNA[RNA$MREtype=="seed_6mer", "log2FoldChange"], RNA[RNA$MREtype=="Non-target", "log2FoldChange"])
RNATypeksTest2 <- ks.test(RNA[RNA$MREtype=="seed_7mer_a1", "log2FoldChange"], RNA[RNA$MREtype=="Non-target", "log2FoldChange"])
RNATypeksTest3 <- ks.test(RNA[RNA$MREtype=="seed_7mer_m8", "log2FoldChange"], RNA[RNA$MREtype=="Non-target", "log2FoldChange"])
RNATypeksTest4 <- ks.test(RNA[RNA$MREtype=="seed_8mer", "log2FoldChange"], RNA[RNA$MREtype=="Non-target", "log2FoldChange"])
RNATypeksTest5 <- ks.test(RNA[RNA$MREtype=="seed_alt_38", "log2FoldChange"], RNA[RNA$MREtype=="Non-target", "log2FoldChange"])

RPFTypeksTest1 <- ks.test(RPF[RPF$MREtype=="seed_6mer", "log2FoldChange"], RPF[RPF$MREtype=="Non-target", "log2FoldChange"])
RPFTypeksTest2 <- ks.test(RPF[RPF$MREtype=="seed_7mer_a1", "log2FoldChange"], RPF[RPF$MREtype=="Non-target", "log2FoldChange"])
RPFTypeksTest3 <- ks.test(RPF[RPF$MREtype=="seed_7mer_m8", "log2FoldChange"], RPF[RPF$MREtype=="Non-target", "log2FoldChange"])
RPFTypeksTest4 <- ks.test(RPF[RPF$MREtype=="seed_8mer", "log2FoldChange"], RPF[RPF$MREtype=="Non-target", "log2FoldChange"])
RPFTypeksTest5 <- ks.test(RPF[RPF$MREtype=="seed_alt_38", "log2FoldChange"], RPF[RPF$MREtype=="Non-target", "log2FoldChange"])

TETypeksTest1 <- ks.test(TE[TE$MREtype=="seed_6mer", "log2FoldChange"], TE[TE$MREtype=="Non-target", "log2FoldChange"])
TETypeksTest2 <- ks.test(TE[TE$MREtype=="seed_7mer_a1", "log2FoldChange"], TE[TE$MREtype=="Non-target", "log2FoldChange"])
TETypeksTest3 <- ks.test(TE[TE$MREtype=="seed_7mer_m8", "log2FoldChange"], TE[TE$MREtype=="Non-target", "log2FoldChange"])
TETypeksTest4 <- ks.test(TE[TE$MREtype=="seed_8mer", "log2FoldChange"], TE[TE$MREtype=="Non-target", "log2FoldChange"])
TETypeksTest5 <- ks.test(TE[TE$MREtype=="seed_alt_38", "log2FoldChange"], TE[TE$MREtype=="Non-target", "log2FoldChange"])

MSTypeksTest1 <- ks.test(MS[MS$MREtype=="seed_6mer", "log2FoldChange"], MS[MS$MREtype=="Non-target", "log2FoldChange"])
MSTypeksTest2 <- ks.test(MS[MS$MREtype=="seed_7mer_a1", "log2FoldChange"], MS[MS$MREtype=="Non-target", "log2FoldChange"])
MSTypeksTest3 <- ks.test(MS[MS$MREtype=="seed_7mer_m8", "log2FoldChange"], MS[MS$MREtype=="Non-target", "log2FoldChange"])
MSTypeksTest4 <- ks.test(MS[MS$MREtype=="seed_8mer", "log2FoldChange"], MS[MS$MREtype=="Non-target", "log2FoldChange"])
MSTypeksTest5 <- ks.test(MS[MS$MREtype=="seed_alt_38", "log2FoldChange"], MS[MS$MREtype=="Non-target", "log2FoldChange"])


Typesig <- data.frame(c("RNA", "RNA", "RNA", "RNA", "RNA", "RPF", "RPF", "RPF", "RPF", "RPF",
                        "TE", "TE", "TE", "TE", "TE", "MS", "MS", "MS", "MS", "MS"),
                        c("seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", 
                          "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38",
                          "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38", 
                          "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "seed_alt_38"),
                        c(RNATypeTTest1$p.value, RNATypeTTest2$p.value, RNATypeTTest3$p.value, RNATypeTTest4$p.value, RNATypeTTest5$p.value,
                          RPFTypeTTest1$p.value, RPFTypeTTest2$p.value, RPFTypeTTest3$p.value, RPFTypeTTest4$p.value, RPFTypeTTest5$p.value,
                          TETypeTTest1$p.value, TETypeTTest2$p.value, TETypeTTest3$p.value, TETypeTTest4$p.value, TETypeTTest5$p.value, 
                          MSTypeTTest1$p.value, MSTypeTTest2$p.value, MSTypeTTest3$p.value, MSTypeTTest4$p.value, MSTypeTTest5$p.value),
                        c(RNATypeksTest1$p.value, RNATypeksTest2$p.value, RNATypeksTest3$p.value, RNATypeksTest4$p.value, RNATypeksTest5$p.value,
                          RPFTypeksTest1$p.value, RPFTypeksTest2$p.value, RPFTypeksTest3$p.value, RPFTypeksTest4$p.value, RPFTypeksTest5$p.value, 
                          TETypeksTest1$p.value, TETypeksTest2$p.value, TETypeksTest3$p.value, TETypeksTest4$p.value, TETypeksTest5$p.value,
                          MSTypeksTest1$p.value, MSTypeksTest2$p.value, MSTypeksTest3$p.value, MSTypeksTest4$p.value, MSTypeksTest5$p.value))
colnames(Typesig) <- c("Experiment", "Subset", "Welch Two Sample t-test", "Asymptotic two-sample Kolmogorov-Smirnov test")
Typesig

#export
write_xlsx(Typesig, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/Typesignificance.xlsx")
```


# 8mer wobble vs non wobble ecdf
```{r 8merwooblenonwobble}
#RNA
RNA8w <- RNA
RNA8w[RNA8w$MREtype == "seed_8mer" & RNA8w$wobble == "Canonical", "MREtype"] <- "8mer AA"
RNA8w[RNA8w$MREtype == "seed_8mer" & RNA8w$wobble == "Non-Canonical", "MREtype"] <- "8mer AU"
table(RNA8w$MREtype)

typeECDFRNA8w <- ggplot(RNA8w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "8mer AA","8mer AU", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA type of 8mer")

typeECDFRNA8w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/8mertypeECDF_RNA.pdf", width=2, height = 2)
typeECDFRNA8w
dev.off()



#RPF
RPF8w <- RPF
RPF8w[RPF8w$MREtype == "seed_8mer" & RPF8w$wobble == "Canonical", "MREtype"] <- "8mer AA"
RPF8w[RPF8w$MREtype == "seed_8mer" & RPF8w$wobble == "Non-Canonical", "MREtype"] <- "8mer AU"
table(RPF8w$MREtype)

typeECDFRPF8w <- ggplot(RPF8w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "8mer AA","8mer AU", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF type of 8mer")

typeECDFRPF8w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/8mertypeECDF_RPF.pdf", width=2, height = 2)
typeECDFRPF8w
dev.off()



#TE
TE8w <- TE
TE8w[TE8w$MREtype == "seed_8mer" & TE8w$wobble == "Canonical", "MREtype"] <- "8mer AA"
TE8w[TE8w$MREtype == "seed_8mer" & TE8w$wobble == "Non-Canonical", "MREtype"] <- "8mer AU"
table(TE8w$MREtype)

typeECDFTE8w <- ggplot(TE8w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "8mer AA","8mer AU", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("TE type of 8mer")

typeECDFTE8w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/8mertypeECDF_TE.pdf", width=2, height = 2)
typeECDFTE8w
dev.off()


#MS
MS8w <- MS
MS8w[MS8w$MREtype == "seed_8mer" & MS8w$wobble == "Canonical", "MREtype"] <- "8mer AA"
MS8w[MS8w$MREtype == "seed_8mer" & MS8w$wobble == "Non-Canonical", "MREtype"] <- "8mer AU"
table(MS8w$MREtype)

typeECDFMS8w <- ggplot(MS8w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "8mer AA","8mer AU", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("MS type of 8mer")

typeECDFMS8w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/8mertypeECDF_MS.pdf", width=2, height = 2)
typeECDFMS8w
dev.off()


```

## Significance
```{r significance8wobble}
RNA8wTTest1 <- t.test(RNA8w[RNA8w$MREtype=="seed_6mer", "log2FoldChange"], RNA8w[RNA8w$MREtype=="Non-target", "log2FoldChange"])
RNA8wTTest2 <- t.test(RNA8w[RNA8w$MREtype=="seed_7mer_a1", "log2FoldChange"], RNA8w[RNA8w$MREtype=="Non-target", "log2FoldChange"])
RNA8wTTest3 <- t.test(RNA8w[RNA8w$MREtype=="seed_7mer_m8", "log2FoldChange"], RNA8w[RNA8w$MREtype=="Non-target", "log2FoldChange"])
RNA8wTTest4 <- t.test(RNA8w[RNA8w$MREtype=="seed_alt_38", "log2FoldChange"], RNA8w[RNA8w$MREtype=="Non-target", "log2FoldChange"])
RNA8wTTest5 <- t.test(RNA8w[RNA8w$MREtype=="8mer AA", "log2FoldChange"], RNA8w[RNA8w$MREtype=="Non-target", "log2FoldChange"])
RNA8wTTest6 <- t.test(RNA8w[RNA8w$MREtype=="8mer AU", "log2FoldChange"], RNA8w[RNA8w$MREtype=="Non-target", "log2FoldChange"])

RPF8wTTest1 <- t.test(RPF8w[RPF8w$MREtype=="seed_6mer", "log2FoldChange"], RPF8w[RPF8w$MREtype=="Non-target", "log2FoldChange"])
RPF8wTTest2 <- t.test(RPF8w[RPF8w$MREtype=="seed_7mer_a1", "log2FoldChange"], RPF8w[RPF8w$MREtype=="Non-target", "log2FoldChange"])
RPF8wTTest3 <- t.test(RPF8w[RPF8w$MREtype=="seed_7mer_m8", "log2FoldChange"], RPF8w[RPF8w$MREtype=="Non-target", "log2FoldChange"])
RPF8wTTest4 <- t.test(RPF8w[RPF8w$MREtype=="seed_alt_38", "log2FoldChange"], RPF8w[RPF8w$MREtype=="Non-target", "log2FoldChange"])
RPF8wTTest5 <- t.test(RPF8w[RPF8w$MREtype=="8mer AA", "log2FoldChange"], RPF8w[RPF8w$MREtype=="Non-target", "log2FoldChange"])
RPF8wTTest6 <- t.test(RPF8w[RPF8w$MREtype=="8mer AU", "log2FoldChange"], RPF8w[RPF8w$MREtype=="Non-target", "log2FoldChange"])

TE8wTTest1 <- t.test(TE8w[TE8w$MREtype=="seed_6mer", "log2FoldChange"], TE8w[TE8w$MREtype=="Non-target", "log2FoldChange"])
TE8wTTest2 <- t.test(TE8w[TE8w$MREtype=="seed_7mer_a1", "log2FoldChange"], TE8w[TE8w$MREtype=="Non-target", "log2FoldChange"])
TE8wTTest3 <- t.test(TE8w[TE8w$MREtype=="seed_7mer_m8", "log2FoldChange"], TE8w[TE8w$MREtype=="Non-target", "log2FoldChange"])
TE8wTTest4 <- t.test(TE8w[TE8w$MREtype=="seed_alt_38", "log2FoldChange"], TE8w[TE8w$MREtype=="Non-target", "log2FoldChange"])
TE8wTTest5 <- t.test(TE8w[TE8w$MREtype=="8mer AA", "log2FoldChange"], TE8w[TE8w$MREtype=="Non-target", "log2FoldChange"])
TE8wTTest6 <- t.test(TE8w[TE8w$MREtype=="8mer AU", "log2FoldChange"], TE8w[TE8w$MREtype=="Non-target", "log2FoldChange"])

MS8wTTest1 <- t.test(MS8w[MS8w$MREtype=="seed_6mer", "log2FoldChange"], MS8w[MS8w$MREtype=="Non-target", "log2FoldChange"])
MS8wTTest2 <- t.test(MS8w[MS8w$MREtype=="seed_7mer_a1", "log2FoldChange"], MS8w[MS8w$MREtype=="Non-target", "log2FoldChange"])
MS8wTTest3 <- t.test(MS8w[MS8w$MREtype=="seed_7mer_m8", "log2FoldChange"], MS8w[MS8w$MREtype=="Non-target", "log2FoldChange"])
MS8wTTest4 <- t.test(MS8w[MS8w$MREtype=="seed_alt_38", "log2FoldChange"], MS8w[MS8w$MREtype=="Non-target", "log2FoldChange"])
MS8wTTest5 <- t.test(MS8w[MS8w$MREtype=="8mer AA", "log2FoldChange"], MS8w[MS8w$MREtype=="Non-target", "log2FoldChange"])
MS8wTTest6 <- t.test(MS8w[MS8w$MREtype=="8mer AU", "log2FoldChange"], MS8w[MS8w$MREtype=="Non-target", "log2FoldChange"])

#kolmogorov-smirnov
RNA8wksTest1 <- ks.test(RNA8w[RNA8w$MREtype=="seed_6mer", "log2FoldChange"], RNA8w[RNA8w$MREtype=="Non-target", "log2FoldChange"])
RNA8wksTest2 <- ks.test(RNA8w[RNA8w$MREtype=="seed_7mer_a1", "log2FoldChange"], RNA8w[RNA8w$MREtype=="Non-target", "log2FoldChange"])
RNA8wksTest3 <- ks.test(RNA8w[RNA8w$MREtype=="seed_7mer_m8", "log2FoldChange"], RNA8w[RNA8w$MREtype=="Non-target", "log2FoldChange"])
RNA8wksTest4 <- ks.test(RNA8w[RNA8w$MREtype=="seed_alt_38", "log2FoldChange"], RNA8w[RNA8w$MREtype=="Non-target", "log2FoldChange"])
RNA8wksTest5 <- ks.test(RNA8w[RNA8w$MREtype=="8mer AA", "log2FoldChange"], RNA8w[RNA8w$MREtype=="Non-target", "log2FoldChange"])
RNA8wksTest6 <- ks.test(RNA8w[RNA8w$MREtype=="8mer AU", "log2FoldChange"], RNA8w[RNA8w$MREtype=="Non-target", "log2FoldChange"])

RPF8wksTest1 <- ks.test(RPF8w[RPF8w$MREtype=="seed_6mer", "log2FoldChange"], RPF8w[RPF8w$MREtype=="Non-target", "log2FoldChange"])
RPF8wksTest2 <- ks.test(RPF8w[RPF8w$MREtype=="seed_7mer_a1", "log2FoldChange"], RPF8w[RPF8w$MREtype=="Non-target", "log2FoldChange"])
RPF8wksTest3 <- ks.test(RPF8w[RPF8w$MREtype=="seed_7mer_m8", "log2FoldChange"], RPF8w[RPF8w$MREtype=="Non-target", "log2FoldChange"])
RPF8wksTest4 <- ks.test(RPF8w[RPF8w$MREtype=="seed_alt_38", "log2FoldChange"], RPF8w[RPF8w$MREtype=="Non-target", "log2FoldChange"])
RPF8wksTest5 <- ks.test(RPF8w[RPF8w$MREtype=="8mer AA", "log2FoldChange"], RPF8w[RPF8w$MREtype=="Non-target", "log2FoldChange"])
RPF8wksTest6 <- ks.test(RPF8w[RPF8w$MREtype=="8mer AU", "log2FoldChange"], RPF8w[RPF8w$MREtype=="Non-target", "log2FoldChange"])

TE8wksTest1 <- ks.test(TE8w[TE8w$MREtype=="seed_6mer", "log2FoldChange"], TE8w[TE8w$MREtype=="Non-target", "log2FoldChange"])
TE8wksTest2 <- ks.test(TE8w[TE8w$MREtype=="seed_7mer_a1", "log2FoldChange"], TE8w[TE8w$MREtype=="Non-target", "log2FoldChange"])
TE8wksTest3 <- ks.test(TE8w[TE8w$MREtype=="seed_7mer_m8", "log2FoldChange"], TE8w[TE8w$MREtype=="Non-target", "log2FoldChange"])
TE8wksTest4 <- ks.test(TE8w[TE8w$MREtype=="seed_alt_38", "log2FoldChange"], TE8w[TE8w$MREtype=="Non-target", "log2FoldChange"])
TE8wksTest5 <- ks.test(TE8w[TE8w$MREtype=="8mer AA", "log2FoldChange"], TE8w[TE8w$MREtype=="Non-target", "log2FoldChange"])
TE8wksTest6 <- ks.test(TE8w[TE8w$MREtype=="8mer AU", "log2FoldChange"], TE8w[TE8w$MREtype=="Non-target", "log2FoldChange"])

MS8wksTest1 <- ks.test(MS8w[MS8w$MREtype=="seed_6mer", "log2FoldChange"], MS8w[MS8w$MREtype=="Non-target", "log2FoldChange"])
MS8wksTest2 <- ks.test(MS8w[MS8w$MREtype=="seed_7mer_a1", "log2FoldChange"], MS8w[MS8w$MREtype=="Non-target", "log2FoldChange"])
MS8wksTest3 <- ks.test(MS8w[MS8w$MREtype=="seed_7mer_m8", "log2FoldChange"], MS8w[MS8w$MREtype=="Non-target", "log2FoldChange"])
MS8wksTest4 <- ks.test(MS8w[MS8w$MREtype=="seed_alt_38", "log2FoldChange"], MS8w[MS8w$MREtype=="Non-target", "log2FoldChange"])
MS8wksTest5 <- ks.test(MS8w[MS8w$MREtype=="8mer AA", "log2FoldChange"], MS8w[MS8w$MREtype=="Non-target", "log2FoldChange"])
MS8wksTest6 <- ks.test(MS8w[MS8w$MREtype=="8mer AU", "log2FoldChange"], MS8w[MS8w$MREtype=="Non-target", "log2FoldChange"])


Type8wsig <- data.frame(c("RNA", "RNA", "RNA", "RNA", "RNA", "RNA", "RPF", "RPF", "RPF", "RPF", "RPF", "RPF",
                        "TE", "TE", "TE", "TE", "TE", "TE", "MS", "MS", "MS", "MS", "MS", "MS"),
                        c("seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_alt_38", "8mer AA", "8mer AU",
                          "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_alt_38", "8mer AA", "8mer AU",
                          "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_alt_38", "8mer AA", "8mer AU", 
                          "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_alt_38", "8mer AA", "8mer AU"),
                        c(RNA8wTTest1$p.value, RNA8wTTest2$p.value, RNA8wTTest3$p.value, RNA8wTTest4$p.value, RNA8wTTest5$p.value, RNA8wTTest6$p.value,
                          RPF8wTTest1$p.value, RPF8wTTest2$p.value, RPF8wTTest3$p.value, RPF8wTTest4$p.value, RPF8wTTest5$p.value, RPF8wTTest6$p.value,
                          TE8wTTest1$p.value, TE8wTTest2$p.value, TE8wTTest3$p.value, TE8wTTest4$p.value, TE8wTTest5$p.value, TE8wTTest6$p.value,
                          MS8wTTest1$p.value, MS8wTTest2$p.value, MS8wTTest3$p.value, MS8wTTest4$p.value, MS8wTTest5$p.value, MS8wTTest6$p.value),
                        c(RNA8wksTest1$p.value, RNA8wksTest2$p.value, RNA8wksTest3$p.value, RNA8wksTest4$p.value, RNA8wksTest5$p.value, RNA8wksTest6$p.value,
                          RPF8wksTest1$p.value, RPF8wksTest2$p.value, RPF8wksTest3$p.value, RPF8wksTest4$p.value, RPF8wksTest5$p.value, RPF8wksTest6$p.value, 
                          TE8wksTest1$p.value, TE8wksTest2$p.value, TE8wksTest3$p.value, TE8wksTest4$p.value, TE8wksTest5$p.value, TE8wksTest6$p.value,
                          MS8wksTest1$p.value, MS8wksTest2$p.value, MS8wksTest3$p.value, MS8wksTest4$p.value, MS8wksTest5$p.value, MS8wksTest6$p.value))
colnames(Type8wsig) <- c("Experiment", "Subset", "Welch Two Sample t-test", "Asymptotic two-sample Kolmogorov-Smirnov test")
Type8wsig

#export
write_xlsx(Type8wsig, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/8wsignificance.xlsx")
```


# 7merA1 wobble vs non wobble ecdf
```{r 7merA1wooblenonwobble}
#RNA
RNA7A1w <- RNA
RNA7A1w[RNA7A1w$MREtype == "seed_7mer_a1" & RNA7A1w$wobble == "Canonical", "MREtype"] <- "7mer_a1 AA"
RNA7A1w[RNA7A1w$MREtype == "seed_7mer_a1" & RNA7A1w$wobble == "Non-Canonical", "MREtype"] <- "7mer_a1 AU"
table(RNA7A1w$MREtype)

typeECDFRNA7A1w <- ggplot(RNA7A1w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_8mer", "seed_7mer_m8", "7mer_a1 AA","7mer_a1 AU", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA type of 7mer_a1")

typeECDFRNA7A1w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/7mer_a1typeECDF_RNA.pdf", width=2, height = 2)
typeECDFRNA7A1w
dev.off()



#RPF
RPF7A1w <- RPF
RPF7A1w[RPF7A1w$MREtype == "seed_7mer_a1" & RPF7A1w$wobble == "Canonical", "MREtype"] <- "7mer_a1 AA"
RPF7A1w[RPF7A1w$MREtype == "seed_7mer_a1" & RPF7A1w$wobble == "Non-Canonical", "MREtype"] <- "7mer_a1 AU"
table(RPF7A1w$MREtype)

typeECDFRPF7A1w <- ggplot(RPF7A1w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_8mer", "seed_7mer_m8", "7mer_a1 AA","7mer_a1 AU", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF type of 7mer_a1")

typeECDFRPF7A1w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/7mer_a1typeECDF_RPF.pdf", width=2, height = 2)
typeECDFRPF7A1w
dev.off()



#TE
TE7A1w <- TE
TE7A1w[TE7A1w$MREtype == "seed_7mer_a1" & TE7A1w$wobble == "Canonical", "MREtype"] <- "7mer_a1 AA"
TE7A1w[TE7A1w$MREtype == "seed_7mer_a1" & TE7A1w$wobble == "Non-Canonical", "MREtype"] <- "7mer_a1 AU"
table(TE7A1w$MREtype)

typeECDFTE7A1w <- ggplot(TE7A1w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_8mer", "seed_7mer_m8", "7mer_a1 AA","7mer_a1 AU", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("TE type of 7mer_a1")

typeECDFTE7A1w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/7mer_a1typeECDF_TE.pdf", width=2, height = 2)
typeECDFTE7A1w
dev.off()


#MS
MS7A1w <- MS
MS7A1w[MS7A1w$MREtype == "seed_7mer_a1" & MS7A1w$wobble == "Canonical", "MREtype"] <- "7mer_a1 AA"
MS7A1w[MS7A1w$MREtype == "seed_7mer_a1" & MS7A1w$wobble == "Non-Canonical", "MREtype"] <- "7mer_a1 AU"
table(MS7A1w$MREtype)

typeECDFMS7A1w <- ggplot(MS7A1w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_8mer", "seed_7mer_m8", "7mer_a1 AA","7mer_a1 AU", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("MS type of 7mer_a1")

typeECDFMS7A1w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/7mer_a1typeECDF_MS.pdf", width=2, height = 2)
typeECDFMS7A1w
dev.off()


```

## Significance
```{r significance7merA1}
RNA7A1wTTest1 <- t.test(RNA7A1w[RNA7A1w$MREtype=="seed_6mer", "log2FoldChange"], RNA7A1w[RNA7A1w$MREtype=="Non-target", "log2FoldChange"])
RNA7A1wTTest2 <- t.test(RNA7A1w[RNA7A1w$MREtype=="seed_8mer", "log2FoldChange"], RNA7A1w[RNA7A1w$MREtype=="Non-target", "log2FoldChange"])
RNA7A1wTTest3 <- t.test(RNA7A1w[RNA7A1w$MREtype=="seed_7mer_m8", "log2FoldChange"], RNA7A1w[RNA7A1w$MREtype=="Non-target", "log2FoldChange"])
RNA7A1wTTest4 <- t.test(RNA7A1w[RNA7A1w$MREtype=="seed_alt_38", "log2FoldChange"], RNA7A1w[RNA7A1w$MREtype=="Non-target", "log2FoldChange"])
RNA7A1wTTest5 <- t.test(RNA7A1w[RNA7A1w$MREtype=="7mer_a1 AA", "log2FoldChange"], RNA7A1w[RNA7A1w$MREtype=="Non-target", "log2FoldChange"])
RNA7A1wTTest6 <- t.test(RNA7A1w[RNA7A1w$MREtype=="7mer_a1 AU", "log2FoldChange"], RNA7A1w[RNA7A1w$MREtype=="Non-target", "log2FoldChange"])

RPF7A1wTTest1 <- t.test(RPF7A1w[RPF7A1w$MREtype=="seed_6mer", "log2FoldChange"], RPF7A1w[RPF7A1w$MREtype=="Non-target", "log2FoldChange"])
RPF7A1wTTest2 <- t.test(RPF7A1w[RPF7A1w$MREtype=="seed_8mer", "log2FoldChange"], RPF7A1w[RPF7A1w$MREtype=="Non-target", "log2FoldChange"])
RPF7A1wTTest3 <- t.test(RPF7A1w[RPF7A1w$MREtype=="seed_7mer_m8", "log2FoldChange"], RPF7A1w[RPF7A1w$MREtype=="Non-target", "log2FoldChange"])
RPF7A1wTTest4 <- t.test(RPF7A1w[RPF7A1w$MREtype=="seed_alt_38", "log2FoldChange"], RPF7A1w[RPF7A1w$MREtype=="Non-target", "log2FoldChange"])
RPF7A1wTTest5 <- t.test(RPF7A1w[RPF7A1w$MREtype=="7mer_a1 AA", "log2FoldChange"], RPF7A1w[RPF7A1w$MREtype=="Non-target", "log2FoldChange"])
RPF7A1wTTest6 <- t.test(RPF7A1w[RPF7A1w$MREtype=="7mer_a1 AU", "log2FoldChange"], RPF7A1w[RPF7A1w$MREtype=="Non-target", "log2FoldChange"])

TE7A1wTTest1 <- t.test(TE7A1w[TE7A1w$MREtype=="seed_6mer", "log2FoldChange"], TE7A1w[TE7A1w$MREtype=="Non-target", "log2FoldChange"])
TE7A1wTTest2 <- t.test(TE7A1w[TE7A1w$MREtype=="seed_8mer", "log2FoldChange"], TE7A1w[TE7A1w$MREtype=="Non-target", "log2FoldChange"])
TE7A1wTTest3 <- t.test(TE7A1w[TE7A1w$MREtype=="seed_7mer_m8", "log2FoldChange"], TE7A1w[TE7A1w$MREtype=="Non-target", "log2FoldChange"])
TE7A1wTTest4 <- t.test(TE7A1w[TE7A1w$MREtype=="seed_alt_38", "log2FoldChange"], TE7A1w[TE7A1w$MREtype=="Non-target", "log2FoldChange"])
TE7A1wTTest5 <- t.test(TE7A1w[TE7A1w$MREtype=="7mer_a1 AA", "log2FoldChange"], TE7A1w[TE7A1w$MREtype=="Non-target", "log2FoldChange"])
TE7A1wTTest6 <- t.test(TE7A1w[TE7A1w$MREtype=="7mer_a1 AU", "log2FoldChange"], TE7A1w[TE7A1w$MREtype=="Non-target", "log2FoldChange"])

MS7A1wTTest1 <- t.test(MS7A1w[MS7A1w$MREtype=="seed_6mer", "log2FoldChange"], MS7A1w[MS7A1w$MREtype=="Non-target", "log2FoldChange"])
MS7A1wTTest2 <- t.test(MS7A1w[MS7A1w$MREtype=="seed_8mer", "log2FoldChange"], MS7A1w[MS7A1w$MREtype=="Non-target", "log2FoldChange"])
MS7A1wTTest3 <- t.test(MS7A1w[MS7A1w$MREtype=="seed_7mer_m8", "log2FoldChange"], MS7A1w[MS7A1w$MREtype=="Non-target", "log2FoldChange"])
MS7A1wTTest4 <- t.test(MS7A1w[MS7A1w$MREtype=="seed_alt_38", "log2FoldChange"], MS7A1w[MS7A1w$MREtype=="Non-target", "log2FoldChange"])
MS7A1wTTest5 <- t.test(MS7A1w[MS7A1w$MREtype=="7mer_a1 AA", "log2FoldChange"], MS7A1w[MS7A1w$MREtype=="Non-target", "log2FoldChange"])
MS7A1wTTest6 <- t.test(MS7A1w[MS7A1w$MREtype=="7mer_a1 AU", "log2FoldChange"], MS7A1w[MS7A1w$MREtype=="Non-target", "log2FoldChange"])

#kolmogorov-smirnov
RNA7A1wksTest1 <- ks.test(RNA7A1w[RNA7A1w$MREtype=="seed_6mer", "log2FoldChange"], RNA7A1w[RNA7A1w$MREtype=="Non-target", "log2FoldChange"])
RNA7A1wksTest2 <- ks.test(RNA7A1w[RNA7A1w$MREtype=="seed_8mer", "log2FoldChange"], RNA7A1w[RNA7A1w$MREtype=="Non-target", "log2FoldChange"])
RNA7A1wksTest3 <- ks.test(RNA7A1w[RNA7A1w$MREtype=="seed_7mer_m8", "log2FoldChange"], RNA7A1w[RNA7A1w$MREtype=="Non-target", "log2FoldChange"])
RNA7A1wksTest4 <- ks.test(RNA7A1w[RNA7A1w$MREtype=="seed_alt_38", "log2FoldChange"], RNA7A1w[RNA7A1w$MREtype=="Non-target", "log2FoldChange"])
RNA7A1wksTest5 <- ks.test(RNA7A1w[RNA7A1w$MREtype=="7mer_a1 AA", "log2FoldChange"], RNA7A1w[RNA7A1w$MREtype=="Non-target", "log2FoldChange"])
RNA7A1wksTest6 <- ks.test(RNA7A1w[RNA7A1w$MREtype=="7mer_a1 AU", "log2FoldChange"], RNA7A1w[RNA7A1w$MREtype=="Non-target", "log2FoldChange"])

RPF7A1wksTest1 <- ks.test(RPF7A1w[RPF7A1w$MREtype=="seed_6mer", "log2FoldChange"], RPF7A1w[RPF7A1w$MREtype=="Non-target", "log2FoldChange"])
RPF7A1wksTest2 <- ks.test(RPF7A1w[RPF7A1w$MREtype=="seed_8mer", "log2FoldChange"], RPF7A1w[RPF7A1w$MREtype=="Non-target", "log2FoldChange"])
RPF7A1wksTest3 <- ks.test(RPF7A1w[RPF7A1w$MREtype=="seed_7mer_m8", "log2FoldChange"], RPF7A1w[RPF7A1w$MREtype=="Non-target", "log2FoldChange"])
RPF7A1wksTest4 <- ks.test(RPF7A1w[RPF7A1w$MREtype=="seed_alt_38", "log2FoldChange"], RPF7A1w[RPF7A1w$MREtype=="Non-target", "log2FoldChange"])
RPF7A1wksTest5 <- ks.test(RPF7A1w[RPF7A1w$MREtype=="7mer_a1 AA", "log2FoldChange"], RPF7A1w[RPF7A1w$MREtype=="Non-target", "log2FoldChange"])
RPF7A1wksTest6 <- ks.test(RPF7A1w[RPF7A1w$MREtype=="7mer_a1 AU", "log2FoldChange"], RPF7A1w[RPF7A1w$MREtype=="Non-target", "log2FoldChange"])

TE7A1wksTest1 <- ks.test(TE7A1w[TE7A1w$MREtype=="seed_6mer", "log2FoldChange"], TE7A1w[TE7A1w$MREtype=="Non-target", "log2FoldChange"])
TE7A1wksTest2 <- ks.test(TE7A1w[TE7A1w$MREtype=="seed_8mer", "log2FoldChange"], TE7A1w[TE7A1w$MREtype=="Non-target", "log2FoldChange"])
TE7A1wksTest3 <- ks.test(TE7A1w[TE7A1w$MREtype=="seed_7mer_m8", "log2FoldChange"], TE7A1w[TE7A1w$MREtype=="Non-target", "log2FoldChange"])
TE7A1wksTest4 <- ks.test(TE7A1w[TE7A1w$MREtype=="seed_alt_38", "log2FoldChange"], TE7A1w[TE7A1w$MREtype=="Non-target", "log2FoldChange"])
TE7A1wksTest5 <- ks.test(TE7A1w[TE7A1w$MREtype=="7mer_a1 AA", "log2FoldChange"], TE7A1w[TE7A1w$MREtype=="Non-target", "log2FoldChange"])
TE7A1wksTest6 <- ks.test(TE7A1w[TE7A1w$MREtype=="7mer_a1 AU", "log2FoldChange"], TE7A1w[TE7A1w$MREtype=="Non-target", "log2FoldChange"])

MS7A1wksTest1 <- ks.test(MS7A1w[MS7A1w$MREtype=="seed_6mer", "log2FoldChange"], MS7A1w[MS7A1w$MREtype=="Non-target", "log2FoldChange"])
MS7A1wksTest2 <- ks.test(MS7A1w[MS7A1w$MREtype=="seed_8mer", "log2FoldChange"], MS7A1w[MS7A1w$MREtype=="Non-target", "log2FoldChange"])
MS7A1wksTest3 <- ks.test(MS7A1w[MS7A1w$MREtype=="seed_7mer_m8", "log2FoldChange"], MS7A1w[MS7A1w$MREtype=="Non-target", "log2FoldChange"])
MS7A1wksTest4 <- ks.test(MS7A1w[MS7A1w$MREtype=="seed_alt_38", "log2FoldChange"], MS7A1w[MS7A1w$MREtype=="Non-target", "log2FoldChange"])
MS7A1wksTest5 <- ks.test(MS7A1w[MS7A1w$MREtype=="7mer_a1 AA", "log2FoldChange"], MS7A1w[MS7A1w$MREtype=="Non-target", "log2FoldChange"])
MS7A1wksTest6 <- ks.test(MS7A1w[MS7A1w$MREtype=="7mer_a1 AU", "log2FoldChange"], MS7A1w[MS7A1w$MREtype=="Non-target", "log2FoldChange"])


Type7A1wsig <- data.frame(c("RNA", "RNA", "RNA", "RNA", "RNA", "RNA", "RPF", "RPF", "RPF", "RPF", "RPF", "RPF",
                        "TE", "TE", "TE", "TE", "TE", "TE", "MS", "MS", "MS", "MS", "MS", "MS"),
                        c("seed_6mer", "seed_8mer", "seed_7mer_m8", "seed_alt_38", "7mer_a1 AA", "7mer_a1 AU",
                          "seed_6mer", "seed_8mer", "seed_7mer_m8", "seed_alt_38", "7mer_a1 AA", "7mer_a1 AU",
                          "seed_6mer", "seed_8mer", "seed_7mer_m8", "seed_alt_38", "7mer_a1 AA", "7mer_a1 AU", 
                          "seed_6mer", "seed_8mer", "seed_7mer_m8", "seed_alt_38", "7mer_a1 AA", "7mer_a1 AU"),
                        c(RNA7A1wTTest1$p.value, RNA7A1wTTest2$p.value, RNA7A1wTTest3$p.value, RNA7A1wTTest4$p.value, RNA7A1wTTest5$p.value, RNA7A1wTTest6$p.value,
                          RPF7A1wTTest1$p.value, RPF7A1wTTest2$p.value, RPF7A1wTTest3$p.value, RPF7A1wTTest4$p.value, RPF7A1wTTest5$p.value, RPF7A1wTTest6$p.value,
                          TE7A1wTTest1$p.value, TE7A1wTTest2$p.value, TE7A1wTTest3$p.value, TE7A1wTTest4$p.value, TE7A1wTTest5$p.value, TE7A1wTTest6$p.value,
                          MS7A1wTTest1$p.value, MS7A1wTTest2$p.value, MS7A1wTTest3$p.value, MS7A1wTTest4$p.value, MS7A1wTTest5$p.value, MS7A1wTTest6$p.value),
                        c(RNA7A1wksTest1$p.value, RNA7A1wksTest2$p.value, RNA7A1wksTest3$p.value, RNA7A1wksTest4$p.value, RNA7A1wksTest5$p.value, RNA7A1wksTest6$p.value,
                          RPF7A1wksTest1$p.value, RPF7A1wksTest2$p.value, RPF7A1wksTest3$p.value, RPF7A1wksTest4$p.value, RPF7A1wksTest5$p.value, RPF7A1wksTest6$p.value, 
                          TE7A1wksTest1$p.value, TE7A1wksTest2$p.value, TE7A1wksTest3$p.value, TE7A1wksTest4$p.value, TE7A1wksTest5$p.value, TE7A1wksTest6$p.value,
                          MS7A1wksTest1$p.value, MS7A1wksTest2$p.value, MS7A1wksTest3$p.value, MS7A1wksTest4$p.value, MS7A1wksTest5$p.value, MS7A1wksTest6$p.value))
colnames(Type7A1wsig) <- c("Experiment", "Subset", "Welch Two Sample t-test", "Asymptotic two-sample Kolmogorov-Smirnov test")
Type7A1wsig

#export
write_xlsx(Type7A1wsig, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/7A1wsignificance.xlsx")
```




# 7mer_m8 wobble vs non wobble ecdf
```{r 7mer_m8wooblenonwobble}
#RNA
RNA7m8w <- RNA
RNA7m8w[RNA7m8w$MREtype == "seed_7mer_m8" & RNA7m8w$wobble == "Canonical", "MREtype"] <- "7mer_m8 AA"
RNA7m8w[RNA7m8w$MREtype == "seed_7mer_m8" & RNA7m8w$wobble == "Non-Canonical", "MREtype"] <- "7mer_m8 AU"
table(RNA7m8w$MREtype)

typeECDFRNA7m8w <- ggplot(RNA7m8w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_8mer", "7mer_m8 AA","7mer_m8 AU", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA type of 7mer_m8")

typeECDFRNA7m8w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/7mer_m8typeECDF_RNA.pdf", width=2, height = 2)
typeECDFRNA7m8w
dev.off()



#RPF
RPF7m8w <- RPF
RPF7m8w[RPF7m8w$MREtype == "seed_7mer_m8" & RPF7m8w$wobble == "Canonical", "MREtype"] <- "7mer_m8 AA"
RPF7m8w[RPF7m8w$MREtype == "seed_7mer_m8" & RPF7m8w$wobble == "Non-Canonical", "MREtype"] <- "7mer_m8 AU"
table(RPF7m8w$MREtype)

typeECDFRPF7m8w <- ggplot(RPF7m8w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_8mer", "7mer_m8 AA","7mer_m8 AU", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF type of 7mer_m8")

typeECDFRPF7m8w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/7mer_m8typeECDF_RPF.pdf", width=2, height = 2)
typeECDFRPF7m8w
dev.off()



#TE
TE7m8w <- TE
TE7m8w[TE7m8w$MREtype == "seed_7mer_m8" & TE7m8w$wobble == "Canonical", "MREtype"] <- "7mer_m8 AA"
TE7m8w[TE7m8w$MREtype == "seed_7mer_m8" & TE7m8w$wobble == "Non-Canonical", "MREtype"] <- "7mer_m8 AU"
table(TE7m8w$MREtype)

typeECDFTE7m8w <- ggplot(TE7m8w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_8mer", "7mer_m8 AA","7mer_m8 AU", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("TE type of 7mer_m8")

typeECDFTE7m8w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/7mer_m8typeECDF_TE.pdf", width=2, height = 2)
typeECDFTE7m8w
dev.off()


#MS
MS7m8w <- MS
MS7m8w[MS7m8w$MREtype == "seed_7mer_m8" & MS7m8w$wobble == "Canonical", "MREtype"] <- "7mer_m8 AA"
MS7m8w[MS7m8w$MREtype == "seed_7mer_m8" & MS7m8w$wobble == "Non-Canonical", "MREtype"] <- "7mer_m8 AU"
table(MS7m8w$MREtype)

typeECDFMS7m8w <- ggplot(MS7m8w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_8mer", "7mer_m8 AA","7mer_m8 AU", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("MS type of 7mer_m8")

typeECDFMS7m8w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/7mer_m8typeECDF_MS.pdf", width=2, height = 2)
typeECDFMS7m8w
dev.off()


```


## Significance
```{r significance7mer_m8}
RNA7m8wTTest1 <- t.test(RNA7m8w[RNA7m8w$MREtype=="seed_6mer", "log2FoldChange"], RNA7m8w[RNA7m8w$MREtype=="Non-target", "log2FoldChange"])
RNA7m8wTTest2 <- t.test(RNA7m8w[RNA7m8w$MREtype=="seed_7mer_a1", "log2FoldChange"], RNA7m8w[RNA7m8w$MREtype=="Non-target", "log2FoldChange"])
RNA7m8wTTest3 <- t.test(RNA7m8w[RNA7m8w$MREtype=="seed_8mer", "log2FoldChange"], RNA7m8w[RNA7m8w$MREtype=="Non-target", "log2FoldChange"])
RNA7m8wTTest4 <- t.test(RNA7m8w[RNA7m8w$MREtype=="seed_alt_38", "log2FoldChange"], RNA7m8w[RNA7m8w$MREtype=="Non-target", "log2FoldChange"])
RNA7m8wTTest5 <- t.test(RNA7m8w[RNA7m8w$MREtype=="7mer_m8 AA", "log2FoldChange"], RNA7m8w[RNA7m8w$MREtype=="Non-target", "log2FoldChange"])
RNA7m8wTTest6 <- t.test(RNA7m8w[RNA7m8w$MREtype=="7mer_m8 AU", "log2FoldChange"], RNA7m8w[RNA7m8w$MREtype=="Non-target", "log2FoldChange"])

RPF7m8wTTest1 <- t.test(RPF7m8w[RPF7m8w$MREtype=="seed_6mer", "log2FoldChange"], RPF7m8w[RPF7m8w$MREtype=="Non-target", "log2FoldChange"])
RPF7m8wTTest2 <- t.test(RPF7m8w[RPF7m8w$MREtype=="seed_7mer_a1", "log2FoldChange"], RPF7m8w[RPF7m8w$MREtype=="Non-target", "log2FoldChange"])
RPF7m8wTTest3 <- t.test(RPF7m8w[RPF7m8w$MREtype=="seed_8mer", "log2FoldChange"], RPF7m8w[RPF7m8w$MREtype=="Non-target", "log2FoldChange"])
RPF7m8wTTest4 <- t.test(RPF7m8w[RPF7m8w$MREtype=="seed_alt_38", "log2FoldChange"], RPF7m8w[RPF7m8w$MREtype=="Non-target", "log2FoldChange"])
RPF7m8wTTest5 <- t.test(RPF7m8w[RPF7m8w$MREtype=="7mer_m8 AA", "log2FoldChange"], RPF7m8w[RPF7m8w$MREtype=="Non-target", "log2FoldChange"])
RPF7m8wTTest6 <- t.test(RPF7m8w[RPF7m8w$MREtype=="7mer_m8 AU", "log2FoldChange"], RPF7m8w[RPF7m8w$MREtype=="Non-target", "log2FoldChange"])

TE7m8wTTest1 <- t.test(TE7m8w[TE7m8w$MREtype=="seed_6mer", "log2FoldChange"], TE7m8w[TE7m8w$MREtype=="Non-target", "log2FoldChange"])
TE7m8wTTest2 <- t.test(TE7m8w[TE7m8w$MREtype=="seed_7mer_a1", "log2FoldChange"], TE7m8w[TE7m8w$MREtype=="Non-target", "log2FoldChange"])
TE7m8wTTest3 <- t.test(TE7m8w[TE7m8w$MREtype=="seed_8mer", "log2FoldChange"], TE7m8w[TE7m8w$MREtype=="Non-target", "log2FoldChange"])
TE7m8wTTest4 <- t.test(TE7m8w[TE7m8w$MREtype=="seed_alt_38", "log2FoldChange"], TE7m8w[TE7m8w$MREtype=="Non-target", "log2FoldChange"])
TE7m8wTTest5 <- t.test(TE7m8w[TE7m8w$MREtype=="7mer_m8 AA", "log2FoldChange"], TE7m8w[TE7m8w$MREtype=="Non-target", "log2FoldChange"])
TE7m8wTTest6 <- t.test(TE7m8w[TE7m8w$MREtype=="7mer_m8 AU", "log2FoldChange"], TE7m8w[TE7m8w$MREtype=="Non-target", "log2FoldChange"])

MS7m8wTTest1 <- t.test(MS7m8w[MS7m8w$MREtype=="seed_6mer", "log2FoldChange"], MS7m8w[MS7m8w$MREtype=="Non-target", "log2FoldChange"])
MS7m8wTTest2 <- t.test(MS7m8w[MS7m8w$MREtype=="seed_7mer_a1", "log2FoldChange"], MS7m8w[MS7m8w$MREtype=="Non-target", "log2FoldChange"])
MS7m8wTTest3 <- t.test(MS7m8w[MS7m8w$MREtype=="seed_8mer", "log2FoldChange"], MS7m8w[MS7m8w$MREtype=="Non-target", "log2FoldChange"])
MS7m8wTTest4 <- t.test(MS7m8w[MS7m8w$MREtype=="seed_alt_38", "log2FoldChange"], MS7m8w[MS7m8w$MREtype=="Non-target", "log2FoldChange"])
MS7m8wTTest5 <- t.test(MS7m8w[MS7m8w$MREtype=="7mer_m8 AA", "log2FoldChange"], MS7m8w[MS7m8w$MREtype=="Non-target", "log2FoldChange"])
MS7m8wTTest6 <- t.test(MS7m8w[MS7m8w$MREtype=="7mer_m8 AU", "log2FoldChange"], MS7m8w[MS7m8w$MREtype=="Non-target", "log2FoldChange"])

#kolmogorov-smirnov
RNA7m8wksTest1 <- ks.test(RNA7m8w[RNA7m8w$MREtype=="seed_6mer", "log2FoldChange"], RNA7m8w[RNA7m8w$MREtype=="Non-target", "log2FoldChange"])
RNA7m8wksTest2 <- ks.test(RNA7m8w[RNA7m8w$MREtype=="seed_7mer_a1", "log2FoldChange"], RNA7m8w[RNA7m8w$MREtype=="Non-target", "log2FoldChange"])
RNA7m8wksTest3 <- ks.test(RNA7m8w[RNA7m8w$MREtype=="seed_8mer", "log2FoldChange"], RNA7m8w[RNA7m8w$MREtype=="Non-target", "log2FoldChange"])
RNA7m8wksTest4 <- ks.test(RNA7m8w[RNA7m8w$MREtype=="seed_alt_38", "log2FoldChange"], RNA7m8w[RNA7m8w$MREtype=="Non-target", "log2FoldChange"])
RNA7m8wksTest5 <- ks.test(RNA7m8w[RNA7m8w$MREtype=="7mer_m8 AA", "log2FoldChange"], RNA7m8w[RNA7m8w$MREtype=="Non-target", "log2FoldChange"])
RNA7m8wksTest6 <- ks.test(RNA7m8w[RNA7m8w$MREtype=="7mer_m8 AU", "log2FoldChange"], RNA7m8w[RNA7m8w$MREtype=="Non-target", "log2FoldChange"])

RPF7m8wksTest1 <- ks.test(RPF7m8w[RPF7m8w$MREtype=="seed_6mer", "log2FoldChange"], RPF7m8w[RPF7m8w$MREtype=="Non-target", "log2FoldChange"])
RPF7m8wksTest2 <- ks.test(RPF7m8w[RPF7m8w$MREtype=="seed_7mer_a1", "log2FoldChange"], RPF7m8w[RPF7m8w$MREtype=="Non-target", "log2FoldChange"])
RPF7m8wksTest3 <- ks.test(RPF7m8w[RPF7m8w$MREtype=="seed_8mer", "log2FoldChange"], RPF7m8w[RPF7m8w$MREtype=="Non-target", "log2FoldChange"])
RPF7m8wksTest4 <- ks.test(RPF7m8w[RPF7m8w$MREtype=="seed_alt_38", "log2FoldChange"], RPF7m8w[RPF7m8w$MREtype=="Non-target", "log2FoldChange"])
RPF7m8wksTest5 <- ks.test(RPF7m8w[RPF7m8w$MREtype=="7mer_m8 AA", "log2FoldChange"], RPF7m8w[RPF7m8w$MREtype=="Non-target", "log2FoldChange"])
RPF7m8wksTest6 <- ks.test(RPF7m8w[RPF7m8w$MREtype=="7mer_m8 AU", "log2FoldChange"], RPF7m8w[RPF7m8w$MREtype=="Non-target", "log2FoldChange"])

TE7m8wksTest1 <- ks.test(TE7m8w[TE7m8w$MREtype=="seed_6mer", "log2FoldChange"], TE7m8w[TE7m8w$MREtype=="Non-target", "log2FoldChange"])
TE7m8wksTest2 <- ks.test(TE7m8w[TE7m8w$MREtype=="seed_7mer_a1", "log2FoldChange"], TE7m8w[TE7m8w$MREtype=="Non-target", "log2FoldChange"])
TE7m8wksTest3 <- ks.test(TE7m8w[TE7m8w$MREtype=="seed_8mer", "log2FoldChange"], TE7m8w[TE7m8w$MREtype=="Non-target", "log2FoldChange"])
TE7m8wksTest4 <- ks.test(TE7m8w[TE7m8w$MREtype=="seed_alt_38", "log2FoldChange"], TE7m8w[TE7m8w$MREtype=="Non-target", "log2FoldChange"])
TE7m8wksTest5 <- ks.test(TE7m8w[TE7m8w$MREtype=="7mer_m8 AA", "log2FoldChange"], TE7m8w[TE7m8w$MREtype=="Non-target", "log2FoldChange"])
TE7m8wksTest6 <- ks.test(TE7m8w[TE7m8w$MREtype=="7mer_m8 AU", "log2FoldChange"], TE7m8w[TE7m8w$MREtype=="Non-target", "log2FoldChange"])

MS7m8wksTest1 <- ks.test(MS7m8w[MS7m8w$MREtype=="seed_6mer", "log2FoldChange"], MS7m8w[MS7m8w$MREtype=="Non-target", "log2FoldChange"])
MS7m8wksTest2 <- ks.test(MS7m8w[MS7m8w$MREtype=="seed_7mer_a1", "log2FoldChange"], MS7m8w[MS7m8w$MREtype=="Non-target", "log2FoldChange"])
MS7m8wksTest3 <- ks.test(MS7m8w[MS7m8w$MREtype=="seed_8mer", "log2FoldChange"], MS7m8w[MS7m8w$MREtype=="Non-target", "log2FoldChange"])
MS7m8wksTest4 <- ks.test(MS7m8w[MS7m8w$MREtype=="seed_alt_38", "log2FoldChange"], MS7m8w[MS7m8w$MREtype=="Non-target", "log2FoldChange"])
MS7m8wksTest5 <- ks.test(MS7m8w[MS7m8w$MREtype=="7mer_m8 AA", "log2FoldChange"], MS7m8w[MS7m8w$MREtype=="Non-target", "log2FoldChange"])
MS7m8wksTest6 <- ks.test(MS7m8w[MS7m8w$MREtype=="7mer_m8 AU", "log2FoldChange"], MS7m8w[MS7m8w$MREtype=="Non-target", "log2FoldChange"])


Type7m8wsig <- data.frame(c("RNA", "RNA", "RNA", "RNA", "RNA", "RNA", "RPF", "RPF", "RPF", "RPF", "RPF", "RPF",
                        "TE", "TE", "TE", "TE", "TE", "TE", "MS", "MS", "MS", "MS", "MS", "MS"),
                        c("seed_6mer", "seed_7mer_a1", "seed_8mer", "seed_alt_38", "7mer_m8 AA", "7mer_m8 AU",
                          "seed_6mer", "seed_7mer_a1", "seed_8mer", "seed_alt_38", "7mer_m8 AA", "7mer_m8 AU",
                          "seed_6mer", "seed_7mer_a1", "seed_8mer", "seed_alt_38", "7mer_m8 AA", "7mer_m8 AU", 
                          "seed_6mer", "seed_7mer_a1", "seed_8mer", "seed_alt_38", "7mer_m8 AA", "7mer_m8 AU"),
                        c(RNA7m8wTTest1$p.value, RNA7m8wTTest2$p.value, RNA7m8wTTest3$p.value, RNA7m8wTTest4$p.value, RNA7m8wTTest5$p.value, RNA7m8wTTest6$p.value,
                          RPF7m8wTTest1$p.value, RPF7m8wTTest2$p.value, RPF7m8wTTest3$p.value, RPF7m8wTTest4$p.value, RPF7m8wTTest5$p.value, RPF7m8wTTest6$p.value,
                          TE7m8wTTest1$p.value, TE7m8wTTest2$p.value, TE7m8wTTest3$p.value, TE7m8wTTest4$p.value, TE7m8wTTest5$p.value, TE7m8wTTest6$p.value,
                          MS7m8wTTest1$p.value, MS7m8wTTest2$p.value, MS7m8wTTest3$p.value, MS7m8wTTest4$p.value, MS7m8wTTest5$p.value, MS7m8wTTest6$p.value),
                        c(RNA7m8wksTest1$p.value, RNA7m8wksTest2$p.value, RNA7m8wksTest3$p.value, RNA7m8wksTest4$p.value, RNA7m8wksTest5$p.value, RNA7m8wksTest6$p.value,
                          RPF7m8wksTest1$p.value, RPF7m8wksTest2$p.value, RPF7m8wksTest3$p.value, RPF7m8wksTest4$p.value, RPF7m8wksTest5$p.value, RPF7m8wksTest6$p.value, 
                          TE7m8wksTest1$p.value, TE7m8wksTest2$p.value, TE7m8wksTest3$p.value, TE7m8wksTest4$p.value, TE7m8wksTest5$p.value, TE7m8wksTest6$p.value,
                          MS7m8wksTest1$p.value, MS7m8wksTest2$p.value, MS7m8wksTest3$p.value, MS7m8wksTest4$p.value, MS7m8wksTest5$p.value, MS7m8wksTest6$p.value))
colnames(Type7m8wsig) <- c("Experiment", "Subset", "Welch Two Sample t-test", "Asymptotic two-sample Kolmogorov-Smirnov test")
Type7m8wsig

#export
write_xlsx(Type7m8wsig, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/7m8wsignificance.xlsx")
```



# 6mer wobble vs non wobble ecdf
```{r 6merwooblenonwobble}
#RNA
RNA6w <- RNA
RNA6w[RNA6w$MREtype == "seed_6mer" & RNA6w$wobble == "Canonical", "MREtype"] <- "6mer AA"
RNA6w[RNA6w$MREtype == "seed_6mer" & RNA6w$wobble == "Non-Canonical", "MREtype"] <- "6mer AU"
table(RNA6w$MREtype)

typeECDFRNA6w <- ggplot(RNA6w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_8mer", "seed_7mer_a1", "seed_7mer_m8", "6mer AA","6mer AU", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA type of 6mer")

typeECDFRNA6w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/6mertypeECDF_RNA.pdf", width=2, height = 2)
typeECDFRNA6w
dev.off()



#RPF
RPF6w <- RPF
RPF6w[RPF6w$MREtype == "seed_6mer" & RPF6w$wobble == "Canonical", "MREtype"] <- "6mer AA"
RPF6w[RPF6w$MREtype == "seed_6mer" & RPF6w$wobble == "Non-Canonical", "MREtype"] <- "6mer AU"
table(RPF6w$MREtype)

typeECDFRPF6w <- ggplot(RPF6w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_8mer", "seed_7mer_a1", "seed_7mer_m8", "6mer AA","6mer AU", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF type of 6mer")

typeECDFRPF6w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/6mertypeECDF_RPF.pdf", width=2, height = 2)
typeECDFRPF6w
dev.off()



#TE
TE6w <- TE
TE6w[TE6w$MREtype == "seed_6mer" & TE6w$wobble == "Canonical", "MREtype"] <- "6mer AA"
TE6w[TE6w$MREtype == "seed_6mer" & TE6w$wobble == "Non-Canonical", "MREtype"] <- "6mer AU"
table(TE6w$MREtype)

typeECDFTE6w <- ggplot(TE6w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_8mer", "seed_7mer_a1", "seed_7mer_m8", "6mer AA","6mer AU", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("TE type of 6mer")

typeECDFTE6w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/6mertypeECDF_TE.pdf", width=2, height = 2)
typeECDFTE6w
dev.off()


#MS
MS6w <- MS
MS6w[MS6w$MREtype == "seed_6mer" & MS6w$wobble == "Canonical", "MREtype"] <- "6mer AA"
MS6w[MS6w$MREtype == "seed_6mer" & MS6w$wobble == "Non-Canonical", "MREtype"] <- "6mer AU"
table(MS6w$MREtype)

typeECDFMS6w <- ggplot(MS6w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_8mer", "seed_7mer_a1", "seed_7mer_m8", "6mer AA","6mer AU", "seed_alt_38", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("MS type of 6mer")

typeECDFMS6w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/6mertypeECDF_MS.pdf", width=2, height = 2)
typeECDFMS6w
dev.off()


```

## Significance
```{r significance6mer}
RNA6wTTest1 <- t.test(RNA6w[RNA6w$MREtype=="seed_8mer", "log2FoldChange"], RNA6w[RNA6w$MREtype=="Non-target", "log2FoldChange"])
RNA6wTTest2 <- t.test(RNA6w[RNA6w$MREtype=="seed_7mer_a1", "log2FoldChange"], RNA6w[RNA6w$MREtype=="Non-target", "log2FoldChange"])
RNA6wTTest3 <- t.test(RNA6w[RNA6w$MREtype=="seed_7mer_m8", "log2FoldChange"], RNA6w[RNA6w$MREtype=="Non-target", "log2FoldChange"])
RNA6wTTest4 <- t.test(RNA6w[RNA6w$MREtype=="seed_alt_38", "log2FoldChange"], RNA6w[RNA6w$MREtype=="Non-target", "log2FoldChange"])
RNA6wTTest5 <- t.test(RNA6w[RNA6w$MREtype=="6mer AA", "log2FoldChange"], RNA6w[RNA6w$MREtype=="Non-target", "log2FoldChange"])
RNA6wTTest6 <- t.test(RNA6w[RNA6w$MREtype=="6mer AU", "log2FoldChange"], RNA6w[RNA6w$MREtype=="Non-target", "log2FoldChange"])

RPF6wTTest1 <- t.test(RPF6w[RPF6w$MREtype=="seed_8mer", "log2FoldChange"], RPF6w[RPF6w$MREtype=="Non-target", "log2FoldChange"])
RPF6wTTest2 <- t.test(RPF6w[RPF6w$MREtype=="seed_7mer_a1", "log2FoldChange"], RPF6w[RPF6w$MREtype=="Non-target", "log2FoldChange"])
RPF6wTTest3 <- t.test(RPF6w[RPF6w$MREtype=="seed_7mer_m8", "log2FoldChange"], RPF6w[RPF6w$MREtype=="Non-target", "log2FoldChange"])
RPF6wTTest4 <- t.test(RPF6w[RPF6w$MREtype=="seed_alt_38", "log2FoldChange"], RPF6w[RPF6w$MREtype=="Non-target", "log2FoldChange"])
RPF6wTTest5 <- t.test(RPF6w[RPF6w$MREtype=="6mer AA", "log2FoldChange"], RPF6w[RPF6w$MREtype=="Non-target", "log2FoldChange"])
RPF6wTTest6 <- t.test(RPF6w[RPF6w$MREtype=="6mer AU", "log2FoldChange"], RPF6w[RPF6w$MREtype=="Non-target", "log2FoldChange"])

TE6wTTest1 <- t.test(TE6w[TE6w$MREtype=="seed_8mer", "log2FoldChange"], TE6w[TE6w$MREtype=="Non-target", "log2FoldChange"])
TE6wTTest2 <- t.test(TE6w[TE6w$MREtype=="seed_7mer_a1", "log2FoldChange"], TE6w[TE6w$MREtype=="Non-target", "log2FoldChange"])
TE6wTTest3 <- t.test(TE6w[TE6w$MREtype=="seed_7mer_m8", "log2FoldChange"], TE6w[TE6w$MREtype=="Non-target", "log2FoldChange"])
TE6wTTest4 <- t.test(TE6w[TE6w$MREtype=="seed_alt_38", "log2FoldChange"], TE6w[TE6w$MREtype=="Non-target", "log2FoldChange"])
TE6wTTest5 <- t.test(TE6w[TE6w$MREtype=="6mer AA", "log2FoldChange"], TE6w[TE6w$MREtype=="Non-target", "log2FoldChange"])
TE6wTTest6 <- t.test(TE6w[TE6w$MREtype=="6mer AU", "log2FoldChange"], TE6w[TE6w$MREtype=="Non-target", "log2FoldChange"])

MS6wTTest1 <- t.test(MS6w[MS6w$MREtype=="seed_8mer", "log2FoldChange"], MS6w[MS6w$MREtype=="Non-target", "log2FoldChange"])
MS6wTTest2 <- t.test(MS6w[MS6w$MREtype=="seed_7mer_a1", "log2FoldChange"], MS6w[MS6w$MREtype=="Non-target", "log2FoldChange"])
MS6wTTest3 <- t.test(MS6w[MS6w$MREtype=="seed_7mer_m8", "log2FoldChange"], MS6w[MS6w$MREtype=="Non-target", "log2FoldChange"])
MS6wTTest4 <- t.test(MS6w[MS6w$MREtype=="seed_alt_38", "log2FoldChange"], MS6w[MS6w$MREtype=="Non-target", "log2FoldChange"])
MS6wTTest5 <- t.test(MS6w[MS6w$MREtype=="6mer AA", "log2FoldChange"], MS6w[MS6w$MREtype=="Non-target", "log2FoldChange"])
MS6wTTest6 <- t.test(MS6w[MS6w$MREtype=="6mer AU", "log2FoldChange"], MS6w[MS6w$MREtype=="Non-target", "log2FoldChange"])

#kolmogorov-smirnov
RNA6wksTest1 <- ks.test(RNA6w[RNA6w$MREtype=="seed_8mer", "log2FoldChange"], RNA6w[RNA6w$MREtype=="Non-target", "log2FoldChange"])
RNA6wksTest2 <- ks.test(RNA6w[RNA6w$MREtype=="seed_7mer_a1", "log2FoldChange"], RNA6w[RNA6w$MREtype=="Non-target", "log2FoldChange"])
RNA6wksTest3 <- ks.test(RNA6w[RNA6w$MREtype=="seed_7mer_m8", "log2FoldChange"], RNA6w[RNA6w$MREtype=="Non-target", "log2FoldChange"])
RNA6wksTest4 <- ks.test(RNA6w[RNA6w$MREtype=="seed_alt_38", "log2FoldChange"], RNA6w[RNA6w$MREtype=="Non-target", "log2FoldChange"])
RNA6wksTest5 <- ks.test(RNA6w[RNA6w$MREtype=="6mer AA", "log2FoldChange"], RNA6w[RNA6w$MREtype=="Non-target", "log2FoldChange"])
RNA6wksTest6 <- ks.test(RNA6w[RNA6w$MREtype=="6mer AU", "log2FoldChange"], RNA6w[RNA6w$MREtype=="Non-target", "log2FoldChange"])

RPF6wksTest1 <- ks.test(RPF6w[RPF6w$MREtype=="seed_8mer", "log2FoldChange"], RPF6w[RPF6w$MREtype=="Non-target", "log2FoldChange"])
RPF6wksTest2 <- ks.test(RPF6w[RPF6w$MREtype=="seed_7mer_a1", "log2FoldChange"], RPF6w[RPF6w$MREtype=="Non-target", "log2FoldChange"])
RPF6wksTest3 <- ks.test(RPF6w[RPF6w$MREtype=="seed_7mer_m8", "log2FoldChange"], RPF6w[RPF6w$MREtype=="Non-target", "log2FoldChange"])
RPF6wksTest4 <- ks.test(RPF6w[RPF6w$MREtype=="seed_alt_38", "log2FoldChange"], RPF6w[RPF6w$MREtype=="Non-target", "log2FoldChange"])
RPF6wksTest5 <- ks.test(RPF6w[RPF6w$MREtype=="6mer AA", "log2FoldChange"], RPF6w[RPF6w$MREtype=="Non-target", "log2FoldChange"])
RPF6wksTest6 <- ks.test(RPF6w[RPF6w$MREtype=="6mer AU", "log2FoldChange"], RPF6w[RPF6w$MREtype=="Non-target", "log2FoldChange"])

TE6wksTest1 <- ks.test(TE6w[TE6w$MREtype=="seed_8mer", "log2FoldChange"], TE6w[TE6w$MREtype=="Non-target", "log2FoldChange"])
TE6wksTest2 <- ks.test(TE6w[TE6w$MREtype=="seed_7mer_a1", "log2FoldChange"], TE6w[TE6w$MREtype=="Non-target", "log2FoldChange"])
TE6wksTest3 <- ks.test(TE6w[TE6w$MREtype=="seed_7mer_m8", "log2FoldChange"], TE6w[TE6w$MREtype=="Non-target", "log2FoldChange"])
TE6wksTest4 <- ks.test(TE6w[TE6w$MREtype=="seed_alt_38", "log2FoldChange"], TE6w[TE6w$MREtype=="Non-target", "log2FoldChange"])
TE6wksTest5 <- ks.test(TE6w[TE6w$MREtype=="6mer AA", "log2FoldChange"], TE6w[TE6w$MREtype=="Non-target", "log2FoldChange"])
TE6wksTest6 <- ks.test(TE6w[TE6w$MREtype=="6mer AU", "log2FoldChange"], TE6w[TE6w$MREtype=="Non-target", "log2FoldChange"])

MS6wksTest1 <- ks.test(MS6w[MS6w$MREtype=="seed_8mer", "log2FoldChange"], MS6w[MS6w$MREtype=="Non-target", "log2FoldChange"])
MS6wksTest2 <- ks.test(MS6w[MS6w$MREtype=="seed_7mer_a1", "log2FoldChange"], MS6w[MS6w$MREtype=="Non-target", "log2FoldChange"])
MS6wksTest3 <- ks.test(MS6w[MS6w$MREtype=="seed_7mer_m8", "log2FoldChange"], MS6w[MS6w$MREtype=="Non-target", "log2FoldChange"])
MS6wksTest4 <- ks.test(MS6w[MS6w$MREtype=="seed_alt_38", "log2FoldChange"], MS6w[MS6w$MREtype=="Non-target", "log2FoldChange"])
MS6wksTest5 <- ks.test(MS6w[MS6w$MREtype=="6mer AA", "log2FoldChange"], MS6w[MS6w$MREtype=="Non-target", "log2FoldChange"])
MS6wksTest6 <- ks.test(MS6w[MS6w$MREtype=="6mer AU", "log2FoldChange"], MS6w[MS6w$MREtype=="Non-target", "log2FoldChange"])


Type6wsig <- data.frame(c("RNA", "RNA", "RNA", "RNA", "RNA", "RNA", "RPF", "RPF", "RPF", "RPF", "RPF", "RPF",
                        "TE", "TE", "TE", "TE", "TE", "TE", "MS", "MS", "MS", "MS", "MS", "MS"),
                        c("seed_8mer", "seed_7mer_a1", "seed_7mer_m8", "seed_alt_38", "6mer AA", "6mer AU",
                          "seed_8mer", "seed_7mer_a1", "seed_7mer_m8", "seed_alt_38", "6mer AA", "6mer AU",
                          "seed_8mer", "seed_7mer_a1", "seed_7mer_m8", "seed_alt_38", "6mer AA", "6mer AU", 
                          "seed_8mer", "seed_7mer_a1", "seed_7mer_m8", "seed_alt_38", "6mer AA", "6mer AU"),
                        c(RNA6wTTest1$p.value, RNA6wTTest2$p.value, RNA6wTTest3$p.value, RNA6wTTest4$p.value, RNA6wTTest5$p.value, RNA6wTTest6$p.value,
                          RPF6wTTest1$p.value, RPF6wTTest2$p.value, RPF6wTTest3$p.value, RPF6wTTest4$p.value, RPF6wTTest5$p.value, RPF6wTTest6$p.value,
                          TE6wTTest1$p.value, TE6wTTest2$p.value, TE6wTTest3$p.value, TE6wTTest4$p.value, TE6wTTest5$p.value, TE6wTTest6$p.value,
                          MS6wTTest1$p.value, MS6wTTest2$p.value, MS6wTTest3$p.value, MS6wTTest4$p.value, MS6wTTest5$p.value, MS6wTTest6$p.value),
                        c(RNA6wksTest1$p.value, RNA6wksTest2$p.value, RNA6wksTest3$p.value, RNA6wksTest4$p.value, RNA6wksTest5$p.value, RNA6wksTest6$p.value,
                          RPF6wksTest1$p.value, RPF6wksTest2$p.value, RPF6wksTest3$p.value, RPF6wksTest4$p.value, RPF6wksTest5$p.value, RPF6wksTest6$p.value, 
                          TE6wksTest1$p.value, TE6wksTest2$p.value, TE6wksTest3$p.value, TE6wksTest4$p.value, TE6wksTest5$p.value, TE6wksTest6$p.value,
                          MS6wksTest1$p.value, MS6wksTest2$p.value, MS6wksTest3$p.value, MS6wksTest4$p.value, MS6wksTest5$p.value, MS6wksTest6$p.value))
colnames(Type6wsig) <- c("Experiment", "Subset", "Welch Two Sample t-test", "Asymptotic two-sample Kolmogorov-Smirnov test")
Type6wsig

#export
write_xlsx(Type6wsig, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/6wsignificance.xlsx")
```



# alt_38 wobble vs non wobble ecdf
```{r alt_38wooblenonwobble}
#RNA
RNAa38w <- RNA
RNAa38w[RNAa38w$MREtype == "seed_alt_38" & RNAa38w$wobble == "Canonical", "MREtype"] <- "alt_38 AA"
RNAa38w[RNAa38w$MREtype == "seed_alt_38" & RNAa38w$wobble == "Non-Canonical", "MREtype"] <- "alt_38 AU"
table(RNAa38w$MREtype)

typeECDFRNAa38w <- ggplot(RNAa38w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "alt_38 AA","alt_38 AU", "seed_8mer", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA type of alt_38")

typeECDFRNAa38w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/alt_38typeECDF_RNA.pdf", width=2, height = 2)
typeECDFRNAa38w
dev.off()



#RPF
RPFa38w <- RPF
RPFa38w[RPFa38w$MREtype == "seed_alt_38" & RPFa38w$wobble == "Canonical", "MREtype"] <- "alt_38 AA"
RPFa38w[RPFa38w$MREtype == "seed_alt_38" & RPFa38w$wobble == "Non-Canonical", "MREtype"] <- "alt_38 AU"
table(RPFa38w$MREtype)

typeECDFRPFa38w <- ggplot(RPFa38w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "alt_38 AA","alt_38 AU", "seed_8mer", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF type of alt_38")

typeECDFRPFa38w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/alt_38typeECDF_RPF.pdf", width=2, height = 2)
typeECDFRPFa38w
dev.off()



#TE
TEa38w <- TE
TEa38w[TEa38w$MREtype == "seed_alt_38" & TEa38w$wobble == "Canonical", "MREtype"] <- "alt_38 AA"
TEa38w[TEa38w$MREtype == "seed_alt_38" & TEa38w$wobble == "Non-Canonical", "MREtype"] <- "alt_38 AU"
table(TEa38w$MREtype)

typeECDFTEa38w <- ggplot(TEa38w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "alt_38 AA","alt_38 AU", "seed_8mer", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("TE type of alt_38")

typeECDFTEa38w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/alt_38typeECDF_TE.pdf", width=2, height = 2)
typeECDFTEa38w
dev.off()


#MS
MSa38w <- MS
MSa38w[MSa38w$MREtype == "seed_alt_38" & MSa38w$wobble == "Canonical", "MREtype"] <- "alt_38 AA"
MSa38w[MSa38w$MREtype == "seed_alt_38" & MSa38w$wobble == "Non-Canonical", "MREtype"] <- "alt_38 AU"
table(MSa38w$MREtype)

typeECDFMSa38w <- ggplot(MSa38w, aes(as.numeric(log2FoldChange), 
                              colour=factor(MREtype, levels = c("Non-target", "No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "alt_38 AA","alt_38 AU", "seed_8mer", "Multiple")))) + 
  stat_ecdf(geom="step", linewidth=2) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("MS type of alt_38")

typeECDFMSa38w

#export
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/alt_38typeECDF_MS.pdf", width=2, height = 2)
typeECDFMSa38w
dev.off()


```



## Significance
```{r significancealt_38}
RNAa38wTTest1 <- t.test(RNAa38w[RNAa38w$MREtype=="seed_6mer", "log2FoldChange"], RNAa38w[RNAa38w$MREtype=="Non-target", "log2FoldChange"])
RNAa38wTTest2 <- t.test(RNAa38w[RNAa38w$MREtype=="seed_7mer_a1", "log2FoldChange"], RNAa38w[RNAa38w$MREtype=="Non-target", "log2FoldChange"])
RNAa38wTTest3 <- t.test(RNAa38w[RNAa38w$MREtype=="seed_7mer_m8", "log2FoldChange"], RNAa38w[RNAa38w$MREtype=="Non-target", "log2FoldChange"])
RNAa38wTTest4 <- t.test(RNAa38w[RNAa38w$MREtype=="seed_8mer", "log2FoldChange"], RNAa38w[RNAa38w$MREtype=="Non-target", "log2FoldChange"])
RNAa38wTTest5 <- t.test(RNAa38w[RNAa38w$MREtype=="alt_38 AA", "log2FoldChange"], RNAa38w[RNAa38w$MREtype=="Non-target", "log2FoldChange"])
RNAa38wTTest6 <- t.test(RNAa38w[RNAa38w$MREtype=="alt_38 AU", "log2FoldChange"], RNAa38w[RNAa38w$MREtype=="Non-target", "log2FoldChange"])

RPFa38wTTest1 <- t.test(RPFa38w[RPFa38w$MREtype=="seed_6mer", "log2FoldChange"], RPFa38w[RPFa38w$MREtype=="Non-target", "log2FoldChange"])
RPFa38wTTest2 <- t.test(RPFa38w[RPFa38w$MREtype=="seed_7mer_a1", "log2FoldChange"], RPFa38w[RPFa38w$MREtype=="Non-target", "log2FoldChange"])
RPFa38wTTest3 <- t.test(RPFa38w[RPFa38w$MREtype=="seed_7mer_m8", "log2FoldChange"], RPFa38w[RPFa38w$MREtype=="Non-target", "log2FoldChange"])
RPFa38wTTest4 <- t.test(RPFa38w[RPFa38w$MREtype=="seed_8mer", "log2FoldChange"], RPFa38w[RPFa38w$MREtype=="Non-target", "log2FoldChange"])
RPFa38wTTest5 <- t.test(RPFa38w[RPFa38w$MREtype=="alt_38 AA", "log2FoldChange"], RPFa38w[RPFa38w$MREtype=="Non-target", "log2FoldChange"])
RPFa38wTTest6 <- t.test(RPFa38w[RPFa38w$MREtype=="alt_38 AU", "log2FoldChange"], RPFa38w[RPFa38w$MREtype=="Non-target", "log2FoldChange"])

TEa38wTTest1 <- t.test(TEa38w[TEa38w$MREtype=="seed_6mer", "log2FoldChange"], TEa38w[TEa38w$MREtype=="Non-target", "log2FoldChange"])
TEa38wTTest2 <- t.test(TEa38w[TEa38w$MREtype=="seed_7mer_a1", "log2FoldChange"], TEa38w[TEa38w$MREtype=="Non-target", "log2FoldChange"])
TEa38wTTest3 <- t.test(TEa38w[TEa38w$MREtype=="seed_7mer_m8", "log2FoldChange"], TEa38w[TEa38w$MREtype=="Non-target", "log2FoldChange"])
TEa38wTTest4 <- t.test(TEa38w[TEa38w$MREtype=="seed_8mer", "log2FoldChange"], TEa38w[TEa38w$MREtype=="Non-target", "log2FoldChange"])
TEa38wTTest5 <- t.test(TEa38w[TEa38w$MREtype=="alt_38 AA", "log2FoldChange"], TEa38w[TEa38w$MREtype=="Non-target", "log2FoldChange"])
TEa38wTTest6 <- t.test(TEa38w[TEa38w$MREtype=="alt_38 AU", "log2FoldChange"], TEa38w[TEa38w$MREtype=="Non-target", "log2FoldChange"])

MSa38wTTest1 <- t.test(MSa38w[MSa38w$MREtype=="seed_6mer", "log2FoldChange"], MSa38w[MSa38w$MREtype=="Non-target", "log2FoldChange"])
MSa38wTTest2 <- t.test(MSa38w[MSa38w$MREtype=="seed_7mer_a1", "log2FoldChange"], MSa38w[MSa38w$MREtype=="Non-target", "log2FoldChange"])
MSa38wTTest3 <- t.test(MSa38w[MSa38w$MREtype=="seed_7mer_m8", "log2FoldChange"], MSa38w[MSa38w$MREtype=="Non-target", "log2FoldChange"])
MSa38wTTest4 <- t.test(MSa38w[MSa38w$MREtype=="seed_8mer", "log2FoldChange"], MSa38w[MSa38w$MREtype=="Non-target", "log2FoldChange"])
MSa38wTTest5 <- t.test(MSa38w[MSa38w$MREtype=="alt_38 AA", "log2FoldChange"], MSa38w[MSa38w$MREtype=="Non-target", "log2FoldChange"])
MSa38wTTest6 <- t.test(MSa38w[MSa38w$MREtype=="alt_38 AU", "log2FoldChange"], MSa38w[MSa38w$MREtype=="Non-target", "log2FoldChange"])

#kolmogorov-smirnov
RNAa38wksTest1 <- ks.test(RNAa38w[RNAa38w$MREtype=="seed_6mer", "log2FoldChange"], RNAa38w[RNAa38w$MREtype=="Non-target", "log2FoldChange"])
RNAa38wksTest2 <- ks.test(RNAa38w[RNAa38w$MREtype=="seed_7mer_a1", "log2FoldChange"], RNAa38w[RNAa38w$MREtype=="Non-target", "log2FoldChange"])
RNAa38wksTest3 <- ks.test(RNAa38w[RNAa38w$MREtype=="seed_7mer_m8", "log2FoldChange"], RNAa38w[RNAa38w$MREtype=="Non-target", "log2FoldChange"])
RNAa38wksTest4 <- ks.test(RNAa38w[RNAa38w$MREtype=="seed_8mer", "log2FoldChange"], RNAa38w[RNAa38w$MREtype=="Non-target", "log2FoldChange"])
RNAa38wksTest5 <- ks.test(RNAa38w[RNAa38w$MREtype=="alt_38 AA", "log2FoldChange"], RNAa38w[RNAa38w$MREtype=="Non-target", "log2FoldChange"])
RNAa38wksTest6 <- ks.test(RNAa38w[RNAa38w$MREtype=="alt_38 AU", "log2FoldChange"], RNAa38w[RNAa38w$MREtype=="Non-target", "log2FoldChange"])

RPFa38wksTest1 <- ks.test(RPFa38w[RPFa38w$MREtype=="seed_6mer", "log2FoldChange"], RPFa38w[RPFa38w$MREtype=="Non-target", "log2FoldChange"])
RPFa38wksTest2 <- ks.test(RPFa38w[RPFa38w$MREtype=="seed_7mer_a1", "log2FoldChange"], RPFa38w[RPFa38w$MREtype=="Non-target", "log2FoldChange"])
RPFa38wksTest3 <- ks.test(RPFa38w[RPFa38w$MREtype=="seed_7mer_m8", "log2FoldChange"], RPFa38w[RPFa38w$MREtype=="Non-target", "log2FoldChange"])
RPFa38wksTest4 <- ks.test(RPFa38w[RPFa38w$MREtype=="seed_8mer", "log2FoldChange"], RPFa38w[RPFa38w$MREtype=="Non-target", "log2FoldChange"])
RPFa38wksTest5 <- ks.test(RPFa38w[RPFa38w$MREtype=="alt_38 AA", "log2FoldChange"], RPFa38w[RPFa38w$MREtype=="Non-target", "log2FoldChange"])
RPFa38wksTest6 <- ks.test(RPFa38w[RPFa38w$MREtype=="alt_38 AU", "log2FoldChange"], RPFa38w[RPFa38w$MREtype=="Non-target", "log2FoldChange"])

TEa38wksTest1 <- ks.test(TEa38w[TEa38w$MREtype=="seed_6mer", "log2FoldChange"], TEa38w[TEa38w$MREtype=="Non-target", "log2FoldChange"])
TEa38wksTest2 <- ks.test(TEa38w[TEa38w$MREtype=="seed_7mer_a1", "log2FoldChange"], TEa38w[TEa38w$MREtype=="Non-target", "log2FoldChange"])
TEa38wksTest3 <- ks.test(TEa38w[TEa38w$MREtype=="seed_7mer_m8", "log2FoldChange"], TEa38w[TEa38w$MREtype=="Non-target", "log2FoldChange"])
TEa38wksTest4 <- ks.test(TEa38w[TEa38w$MREtype=="seed_8mer", "log2FoldChange"], TEa38w[TEa38w$MREtype=="Non-target", "log2FoldChange"])
TEa38wksTest5 <- ks.test(TEa38w[TEa38w$MREtype=="alt_38 AA", "log2FoldChange"], TEa38w[TEa38w$MREtype=="Non-target", "log2FoldChange"])
TEa38wksTest6 <- ks.test(TEa38w[TEa38w$MREtype=="alt_38 AU", "log2FoldChange"], TEa38w[TEa38w$MREtype=="Non-target", "log2FoldChange"])

MSa38wksTest1 <- ks.test(MSa38w[MSa38w$MREtype=="seed_6mer", "log2FoldChange"], MSa38w[MSa38w$MREtype=="Non-target", "log2FoldChange"])
MSa38wksTest2 <- ks.test(MSa38w[MSa38w$MREtype=="seed_7mer_a1", "log2FoldChange"], MSa38w[MSa38w$MREtype=="Non-target", "log2FoldChange"])
MSa38wksTest3 <- ks.test(MSa38w[MSa38w$MREtype=="seed_7mer_m8", "log2FoldChange"], MSa38w[MSa38w$MREtype=="Non-target", "log2FoldChange"])
MSa38wksTest4 <- ks.test(MSa38w[MSa38w$MREtype=="seed_8mer", "log2FoldChange"], MSa38w[MSa38w$MREtype=="Non-target", "log2FoldChange"])
MSa38wksTest5 <- ks.test(MSa38w[MSa38w$MREtype=="alt_38 AA", "log2FoldChange"], MSa38w[MSa38w$MREtype=="Non-target", "log2FoldChange"])
MSa38wksTest6 <- ks.test(MSa38w[MSa38w$MREtype=="alt_38 AU", "log2FoldChange"], MSa38w[MSa38w$MREtype=="Non-target", "log2FoldChange"])


Typea38wsig <- data.frame(c("RNA", "RNA", "RNA", "RNA", "RNA", "RNA", "RPF", "RPF", "RPF", "RPF", "RPF", "RPF",
                        "TE", "TE", "TE", "TE", "TE", "TE", "MS", "MS", "MS", "MS", "MS", "MS"),
                        c("seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "alt_38 AA", "alt_38 AU",
                          "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "alt_38 AA", "alt_38 AU",
                          "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "alt_38 AA", "alt_38 AU", 
                          "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "seed_8mer", "alt_38 AA", "alt_38 AU"),
                        c(RNAa38wTTest1$p.value, RNAa38wTTest2$p.value, RNAa38wTTest3$p.value, RNAa38wTTest4$p.value, RNAa38wTTest5$p.value, RNAa38wTTest6$p.value,
                          RPFa38wTTest1$p.value, RPFa38wTTest2$p.value, RPFa38wTTest3$p.value, RPFa38wTTest4$p.value, RPFa38wTTest5$p.value, RPFa38wTTest6$p.value,
                          TEa38wTTest1$p.value, TEa38wTTest2$p.value, TEa38wTTest3$p.value, TEa38wTTest4$p.value, TEa38wTTest5$p.value, TEa38wTTest6$p.value,
                          MSa38wTTest1$p.value, MSa38wTTest2$p.value, MSa38wTTest3$p.value, MSa38wTTest4$p.value, MSa38wTTest5$p.value, MSa38wTTest6$p.value),
                        c(RNAa38wksTest1$p.value, RNAa38wksTest2$p.value, RNAa38wksTest3$p.value, RNAa38wksTest4$p.value, RNAa38wksTest5$p.value, RNAa38wksTest6$p.value,
                          RPFa38wksTest1$p.value, RPFa38wksTest2$p.value, RPFa38wksTest3$p.value, RPFa38wksTest4$p.value, RPFa38wksTest5$p.value, RPFa38wksTest6$p.value, 
                          TEa38wksTest1$p.value, TEa38wksTest2$p.value, TEa38wksTest3$p.value, TEa38wksTest4$p.value, TEa38wksTest5$p.value, TEa38wksTest6$p.value,
                          MSa38wksTest1$p.value, MSa38wksTest2$p.value, MSa38wksTest3$p.value, MSa38wksTest4$p.value, MSa38wksTest5$p.value, MSa38wksTest6$p.value))
colnames(Typea38wsig) <- c("Experiment", "Subset", "Welch Two Sample t-test", "Asymptotic two-sample Kolmogorov-Smirnov test")
Typea38wsig

#export
write_xlsx(Typea38wsig, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/a38wsignificance.xlsx")
```


## density 8mer TE
```{R TEdensity8mer}
TEdens8mer <- ggplot(TE, aes(x=log2FoldChange, colour=factor(MREtype, levels = c("No seed", "seed_6mer", "seed_7mer_a1", "seed_7mer_m8", "8mer AA","8mer AU", "seed_alt_38", "Multiple", "Non-target")))) +
  geom_density() +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c( farbe1, farbe2, farbe3, farbe4, farbe5, farbe6, farbe7, farbe8, "black")) +
  theme_paper() +
  coord_cartesian(xlim = c(-1,1)) +
  ggtitle("TE")

TEdens8mer

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/8merDensity_TE.pdf", width=2, height = 2)
TEdens8mer
dev.off()

```

# mmsat4
```{r mmsat4}
mmsat4frame <- as.data.frame(MMSAT4)


#RNA
RNA$tvsmmsat4 <- "Non-target"
RNA$tvsmmsat4[sub("\\..*", "", RNA$Gene) %in% mmsat4frame$gene_id] <- "MMsat4"
RNA$tvsmmsat4[sub("\\..*", "", RNA$Gene) %in% largetframe$geneID] <- "miR-181 target"
RNA$tvsmmsat4[sub("\\..*", "", RNA$Gene) %in% largetframe$geneID & sub("\\..*", "", RNA$Gene) %in% mmsat4frame$gene_id] <- "both"
table(RNA$tvsmmsat4)

#RPF
RPF$tvsmmsat4 <- "Non-target"
RPF$tvsmmsat4[sub("\\..*", "", RPF$Gene) %in% mmsat4frame$gene_id] <- "MMsat4"
RPF$tvsmmsat4[sub("\\..*", "", RPF$Gene) %in% largetframe$geneID] <- "miR-181 target"
RPF$tvsmmsat4[sub("\\..*", "", RPF$Gene) %in% largetframe$geneID & sub("\\..*", "", RPF$Gene) %in% mmsat4frame$gene_id] <- "both"
table(RPF$tvsmmsat4)

#MS
MS$tvsmmsat4 <- "Non-target"
MS$tvsmmsat4[MS$gene_symbol %in% mmsat4frame$gene_name] <- "MMsat4"
MS$tvsmmsat4[MS$gene_symbol %in% largetframe$geneID] <- "miR-181 target"
MS$tvsmmsat4[MS$gene_symbol %in% largetframe$geneID & MS$gene_symbol %in% mmsat4frame$gene_name] <- "both"
table(MS$tvsmmsat4)

#TE
TE$tvsmmsat4 <- "Non-target"
TE$tvsmmsat4[sub("\\..*", "", TE$Gene) %in% mmsat4frame$gene_id] <- "MMsat4"
TE$tvsmmsat4[sub("\\..*", "", TE$Gene) %in% largetframe$geneID] <- "miR-181 target"
TE$tvsmmsat4[sub("\\..*", "", TE$Gene) %in% largetframe$geneID & sub("\\..*", "", TE$Gene) %in% mmsat4frame$gene_id] <- "both"
table(TE$tvsmmsat4)



#RNA
tolECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(tvsmmsat4, levels = c("Non-target", "MMsat4", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA Mmsat4 vs target")

tolECDFRNA


#RPF
tolECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(tvsmmsat4, levels = c("Non-target", "MMsat4", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF Mmsat4 vs target")

tolECDFRPF

#MS
tolECDFMS <- ggplot(MS, aes(as.numeric(log2FoldChange), colour=factor(tvsmmsat4, levels = c("Non-target", "MMsat4", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("MS Mmsat4 vs target")

tolECDFMS

#TE
tolECDFTE <- ggplot(TE, aes(as.numeric(log2FoldChange), colour=factor(tvsmmsat4, levels = c("Non-target", "MMsat4", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("TE Mmsat4 vs target")

tolECDFTE

#export RNA
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MMSAT4ECDF_RNA.pdf", width=2, height = 2)
tolECDFRNA
dev.off()


#export RPF
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MMSAT4ECDF_RPF.pdf", width=2, height = 2)
tolECDFRPF
dev.off()

#export MS
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MMSAT4ECDF_MS.pdf", width=2, height = 2)
tolECDFMS
dev.off()

#export TE
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MMSAT4ECDF_TE.pdf", width=2, height = 2)
tolECDFTE
dev.off()


table(RPF$tvsmmsat4)
```


## density mmsat4 TE
```{R TEdensityMMsat4}
TEdensmmsat4 <- ggplot(TE, aes(x=log2FoldChange, colour=factor(tvsmmsat4, levels = c("MMsat4", "miR-181 target", "both", "Non-target")))) +
  geom_density() +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c( farbe1, farbe3, RPFncol, "black")) +
  theme_paper() +
  coord_cartesian(xlim = c(-1,1)) +
  ggtitle("TE")

TEdensmmsat4

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MMsat4Density_TE.pdf", width=2, height = 2)
TEdensmmsat4
dev.off()

```

## Significance
```{r significancemmsat4}
RNAtvsmmsat4TTest1 <- t.test(RNA[RNA$tvsmmsat4=="MMsat4", "log2FoldChange"], RNA[RNA$tvsmmsat4=="Non-target", "log2FoldChange"])
RNAtvsmmsat4TTest2 <- t.test(RNA[RNA$tvsmmsat4=="both", "log2FoldChange"], RNA[RNA$tvsmmsat4=="Non-target", "log2FoldChange"])
RNAtvsmmsat4TTest3 <- t.test(RNA[RNA$tvsmmsat4=="miR-181 target", "log2FoldChange"], RNA[RNA$tvsmmsat4=="Non-target", "log2FoldChange"])

RPFtvsmmsat4TTest1 <- t.test(RPF[RPF$tvsmmsat4=="MMsat4", "log2FoldChange"], RPF[RPF$tvsmmsat4=="Non-target", "log2FoldChange"])
RPFtvsmmsat4TTest2 <- t.test(RPF[RPF$tvsmmsat4=="both", "log2FoldChange"], RPF[RPF$tvsmmsat4=="Non-target", "log2FoldChange"])
RPFtvsmmsat4TTest3 <- t.test(RPF[RPF$tvsmmsat4=="miR-181 target", "log2FoldChange"], RPF[RPF$tvsmmsat4=="Non-target", "log2FoldChange"])

TEtvsmmsat4TTest1 <- t.test(TE[TE$tvsmmsat4=="MMsat4", "log2FoldChange"], TE[TE$tvsmmsat4=="Non-target", "log2FoldChange"])
TEtvsmmsat4TTest2 <- t.test(TE[TE$tvsmmsat4=="both", "log2FoldChange"], TE[TE$tvsmmsat4=="Non-target", "log2FoldChange"])
TEtvsmmsat4TTest3 <- t.test(TE[TE$tvsmmsat4=="miR-181 target", "log2FoldChange"], TE[TE$tvsmmsat4=="Non-target", "log2FoldChange"])

# MStvsmmsat4TTest1 <- t.test(MS[MS$tvsmmsat4=="MMsat4", "log2FoldChange"], MS[MS$tvsmmsat4=="Non-target", "log2FoldChange"])
# MStvsmmsat4TTest2 <- t.test(MS[MS$tvsmmsat4=="both", "log2FoldChange"], MS[MS$tvsmmsat4=="Non-target", "log2FoldChange"])
# MStvsmmsat4TTest3 <- t.test(MS[MS$tvsmmsat4=="miR-181 target", "log2FoldChange"], MS[MS$tvsmmsat4=="Non-target", "log2FoldChange"])

#kolmogorov-smirnov
RNAtvsmmsat4ksTest1 <- ks.test(RNA[RNA$tvsmmsat4=="MMsat4", "log2FoldChange"], RNA[RNA$tvsmmsat4=="Non-target", "log2FoldChange"])
RNAtvsmmsat4ksTest2 <- ks.test(RNA[RNA$tvsmmsat4=="both", "log2FoldChange"], RNA[RNA$tvsmmsat4=="Non-target", "log2FoldChange"])
RNAtvsmmsat4ksTest3 <- ks.test(RNA[RNA$tvsmmsat4=="miR-181 target", "log2FoldChange"], RNA[RNA$tvsmmsat4=="Non-target", "log2FoldChange"])

RPFtvsmmsat4ksTest1 <- ks.test(RPF[RPF$tvsmmsat4=="MMsat4", "log2FoldChange"], RPF[RPF$tvsmmsat4=="Non-target", "log2FoldChange"])
RPFtvsmmsat4ksTest2 <- ks.test(RPF[RPF$tvsmmsat4=="both", "log2FoldChange"], RPF[RPF$tvsmmsat4=="Non-target", "log2FoldChange"])
RPFtvsmmsat4ksTest3 <- ks.test(RPF[RPF$tvsmmsat4=="miR-181 target", "log2FoldChange"], RPF[RPF$tvsmmsat4=="Non-target", "log2FoldChange"])

TEtvsmmsat4ksTest1 <- ks.test(TE[TE$tvsmmsat4=="MMsat4", "log2FoldChange"], TE[TE$tvsmmsat4=="Non-target", "log2FoldChange"])
TEtvsmmsat4ksTest2 <- ks.test(TE[TE$tvsmmsat4=="both", "log2FoldChange"], TE[TE$tvsmmsat4=="Non-target", "log2FoldChange"])
TEtvsmmsat4ksTest3 <- ks.test(TE[TE$tvsmmsat4=="miR-181 target", "log2FoldChange"], TE[TE$tvsmmsat4=="Non-target", "log2FoldChange"])

# MStvsmmsat4ksTest1 <- ks.test(MS[MS$tvsmmsat4=="MMsat4", "log2FoldChange"], MS[MS$tvsmmsat4=="Non-target", "log2FoldChange"])
# MStvsmmsat4ksTest2 <- ks.test(MS[MS$tvsmmsat4=="both", "log2FoldChange"], MS[MS$tvsmmsat4=="Non-target", "log2FoldChange"])
# MStvsmmsat4ksTest3 <- ks.test(MS[MS$tvsmmsat4=="miR-181 target", "log2FoldChange"], MS[MS$tvsmmsat4=="Non-target", "log2FoldChange"])


tvsmmsat4sig <- data.frame(c("RNA", "RNA", "RNA", "RPF", "RPF", "RPF", "TE", "TE", "TE" #,"MS", "MS", "MS"
                             ),
                        c("MMsat4", "both", "miR-181 target", "MMsat4", "both", "miR-181 target",
                          "MMsat4", "both", "miR-181 target"#,"MMsat4", "both", "miR-181 target"
                          ),
                        c(RNAtvsmmsat4TTest1$p.value, RNAtvsmmsat4TTest2$p.value, RNAtvsmmsat4TTest3$p.value,
                          RPFtvsmmsat4TTest1$p.value, RPFtvsmmsat4TTest2$p.value, RPFtvsmmsat4TTest3$p.value,
                          TEtvsmmsat4TTest1$p.value, TEtvsmmsat4TTest2$p.value, TEtvsmmsat4TTest3$p.value#, 
                          #MStvsmmsat4TTest1$p.value, MStvsmmsat4TTest2$p.value, MStvsmmsat4TTest3$p.value
                          ),
                        c(RNAtvsmmsat4ksTest1$p.value, RNAtvsmmsat4ksTest2$p.value, RNAtvsmmsat4ksTest3$p.value,
                          RPFtvsmmsat4ksTest1$p.value, RPFtvsmmsat4ksTest2$p.value, RPFtvsmmsat4ksTest3$p.value, 
                          TEtvsmmsat4ksTest1$p.value, TEtvsmmsat4ksTest2$p.value, TEtvsmmsat4ksTest3$p.value#,
                          #MStvsmmsat4ksTest1$p.value, MStvsmmsat4ksTest2$p.value, MStvsmmsat4ksTest3$p.value
                          ))
colnames(tvsmmsat4sig) <- c("Experiment", "Subset", "Welch Two Sample t-test", "Asymptotic two-sample Kolmogorov-Smirnov test")
tvsmmsat4sig

#export
write_xlsx(tvsmmsat4sig, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/tvsmmsat4significance.xlsx")
```



# MurSatRep1
```{r MurSatRep1}
MurSatRep1frame <- as.data.frame(MurSatRep1)

#RNA
RNA$tvsMurSatRep1 <- "Non-target"
RNA$tvsMurSatRep1[sub("\\..*", "", RNA$Gene) %in% MurSatRep1frame$gene_id] <- "MurSatRep1"
RNA$tvsMurSatRep1[sub("\\..*", "", RNA$Gene) %in% largetframe$geneID] <- "miR-181 target"
RNA$tvsMurSatRep1[sub("\\..*", "", RNA$Gene) %in% largetframe$geneID & sub("\\..*", "", RNA$Gene) %in% MurSatRep1frame$gene_id] <- "both"
table(RNA$tvsMurSatRep1)

#RPF
RPF$tvsMurSatRep1 <- "Non-target"
RPF$tvsMurSatRep1[sub("\\..*", "", RPF$Gene) %in% MurSatRep1frame$gene_id] <- "MurSatRep1"
RPF$tvsMurSatRep1[sub("\\..*", "", RPF$Gene) %in% largetframe$geneID] <- "miR-181 target"
RPF$tvsMurSatRep1[sub("\\..*", "", RPF$Gene) %in% largetframe$geneID & sub("\\..*", "", RPF$Gene) %in% MurSatRep1frame$gene_id] <- "both"
table(RPF$tvsMurSatRep1)

#MS
MS$tvsMurSatRep1 <- "Non-target"
MS$tvsMurSatRep1[MS$gene_symbol %in% MurSatRep1frame$gene_name] <- "MurSatRep1"
MS$tvsMurSatRep1[MS$gene_symbol %in% largetframe$geneID] <- "miR-181 target"
MS$tvsMurSatRep1[MS$gene_symbol %in% largetframe$geneID & MS$gene_symbol %in% MurSatRep1frame$gene_name] <- "both"
table(MS$tvsMurSatRep1)

#TE
TE$tvsMurSatRep1 <- "Non-target"
TE$tvsMurSatRep1[sub("\\..*", "", TE$Gene) %in% MurSatRep1frame$gene_id] <- "MurSatRep1"
TE$tvsMurSatRep1[sub("\\..*", "", TE$Gene) %in% largetframe$geneID] <- "miR-181 target"
TE$tvsMurSatRep1[sub("\\..*", "", TE$Gene) %in% largetframe$geneID & sub("\\..*", "", TE$Gene) %in% MurSatRep1frame$gene_id] <- "both"
table(TE$tvsMurSatRep1)


#RNA
murECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(tvsMurSatRep1, levels = c("Non-target", "MurSatRep1", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA MurSatRep1 vs target")

murECDFRNA


#RPF
murECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(tvsMurSatRep1, levels = c("Non-target", "MurSatRep1", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF MurSatRep1 vs target")

murECDFRPF

#MS
murECDFMS <- ggplot(MS, aes(as.numeric(log2FoldChange), colour=factor(tvsMurSatRep1, levels = c("Non-target", "MurSatRep1", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("MS MurSatRep1 vs target")

murECDFMS

#TE
murECDFTE <- ggplot(TE, aes(as.numeric(log2FoldChange), colour=factor(tvsMurSatRep1, levels = c("Non-target", "MurSatRep1", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("TE MurSatRep1 vs target")

murECDFTE



#export RNA
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/mursatrep1ECDF_RNA.pdf", width=2, height = 2)
murECDFRNA
dev.off()

#export RPF
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/mursatrep1ECDF_RPF.pdf", width=2, height = 2)
murECDFRPF
dev.off()

#export MS
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/mursatrep1ECDF_MS.pdf", width=2, height = 2)
murECDFMS
dev.off()

#export TE
pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/mursatrep1ECDF_TE.pdf", width=2, height = 2)
murECDFTE
dev.off()


table(RPF$tvsMurSatRep1)
```
## density mursatrep1 TE
```{R TEdensitymursatrep1}
TEdensmursatrep1 <- ggplot(TE, aes(x=log2FoldChange, colour=factor(tvsMurSatRep1, levels = c( "MurSatRep1", "miR-181 target", "both", "Non-target")))) +
  geom_density() +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c(farbe1, farbe3, RPFncol, "black")) +
  theme_paper() +
  coord_cartesian(xlim = c(-1,1)) +
  ggtitle("TE")

TEdensmursatrep1

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MurSatRep1Density_TE.pdf", width=2, height = 2)
TEdensmursatrep1
dev.off()

```


## Significance
```{r significanceMurSatRep1}
RNAtvsMurSatRep1TTest1 <- t.test(RNA[RNA$tvsMurSatRep1=="MurSatRep1", "log2FoldChange"], RNA[RNA$tvsMurSatRep1=="Non-target", "log2FoldChange"])
RNAtvsMurSatRep1TTest2 <- t.test(RNA[RNA$tvsMurSatRep1=="both", "log2FoldChange"], RNA[RNA$tvsMurSatRep1=="Non-target", "log2FoldChange"])
RNAtvsMurSatRep1TTest3 <- t.test(RNA[RNA$tvsMurSatRep1=="miR-181 target", "log2FoldChange"], RNA[RNA$tvsMurSatRep1=="Non-target", "log2FoldChange"])

RPFtvsMurSatRep1TTest1 <- t.test(RPF[RPF$tvsMurSatRep1=="MurSatRep1", "log2FoldChange"], RPF[RPF$tvsMurSatRep1=="Non-target", "log2FoldChange"])
RPFtvsMurSatRep1TTest2 <- t.test(RPF[RPF$tvsMurSatRep1=="both", "log2FoldChange"], RPF[RPF$tvsMurSatRep1=="Non-target", "log2FoldChange"])
RPFtvsMurSatRep1TTest3 <- t.test(RPF[RPF$tvsMurSatRep1=="miR-181 target", "log2FoldChange"], RPF[RPF$tvsMurSatRep1=="Non-target", "log2FoldChange"])

TEtvsMurSatRep1TTest1 <- t.test(TE[TE$tvsMurSatRep1=="MurSatRep1", "log2FoldChange"], TE[TE$tvsMurSatRep1=="Non-target", "log2FoldChange"])
TEtvsMurSatRep1TTest2 <- t.test(TE[TE$tvsMurSatRep1=="both", "log2FoldChange"], TE[TE$tvsMurSatRep1=="Non-target", "log2FoldChange"])
TEtvsMurSatRep1TTest3 <- t.test(TE[TE$tvsMurSatRep1=="miR-181 target", "log2FoldChange"], TE[TE$tvsMurSatRep1=="Non-target", "log2FoldChange"])

# MStvsMurSatRep1TTest1 <- t.test(MS[MS$tvsMurSatRep1=="MurSatRep1", "log2FoldChange"], MS[MS$tvsMurSatRep1=="Non-target", "log2FoldChange"])
# MStvsMurSatRep1TTest2 <- t.test(MS[MS$tvsMurSatRep1=="both", "log2FoldChange"], MS[MS$tvsMurSatRep1=="Non-target", "log2FoldChange"])
# MStvsMurSatRep1TTest3 <- t.test(MS[MS$tvsMurSatRep1=="miR-181 target", "log2FoldChange"], MS[MS$tvsMurSatRep1=="Non-target", "log2FoldChange"])

#kolmogorov-smirnov
RNAtvsMurSatRep1ksTest1 <- ks.test(RNA[RNA$tvsMurSatRep1=="MurSatRep1", "log2FoldChange"], RNA[RNA$tvsMurSatRep1=="Non-target", "log2FoldChange"])
RNAtvsMurSatRep1ksTest2 <- ks.test(RNA[RNA$tvsMurSatRep1=="both", "log2FoldChange"], RNA[RNA$tvsMurSatRep1=="Non-target", "log2FoldChange"])
RNAtvsMurSatRep1ksTest3 <- ks.test(RNA[RNA$tvsMurSatRep1=="miR-181 target", "log2FoldChange"], RNA[RNA$tvsMurSatRep1=="Non-target", "log2FoldChange"])

RPFtvsMurSatRep1ksTest1 <- ks.test(RPF[RPF$tvsMurSatRep1=="MurSatRep1", "log2FoldChange"], RPF[RPF$tvsMurSatRep1=="Non-target", "log2FoldChange"])
RPFtvsMurSatRep1ksTest2 <- ks.test(RPF[RPF$tvsMurSatRep1=="both", "log2FoldChange"], RPF[RPF$tvsMurSatRep1=="Non-target", "log2FoldChange"])
RPFtvsMurSatRep1ksTest3 <- ks.test(RPF[RPF$tvsMurSatRep1=="miR-181 target", "log2FoldChange"], RPF[RPF$tvsMurSatRep1=="Non-target", "log2FoldChange"])

TEtvsMurSatRep1ksTest1 <- ks.test(TE[TE$tvsMurSatRep1=="MurSatRep1", "log2FoldChange"], TE[TE$tvsMurSatRep1=="Non-target", "log2FoldChange"])
TEtvsMurSatRep1ksTest2 <- ks.test(TE[TE$tvsMurSatRep1=="both", "log2FoldChange"], TE[TE$tvsMurSatRep1=="Non-target", "log2FoldChange"])
TEtvsMurSatRep1ksTest3 <- ks.test(TE[TE$tvsMurSatRep1=="miR-181 target", "log2FoldChange"], TE[TE$tvsMurSatRep1=="Non-target", "log2FoldChange"])

# MStvsMurSatRep1ksTest1 <- ks.test(MS[MS$tvsMurSatRep1=="MurSatRep1", "log2FoldChange"], MS[MS$tvsMurSatRep1=="Non-target", "log2FoldChange"])
# MStvsMurSatRep1ksTest2 <- ks.test(MS[MS$tvsMurSatRep1=="both", "log2FoldChange"], MS[MS$tvsMurSatRep1=="Non-target", "log2FoldChange"])
# MStvsMurSatRep1ksTest3 <- ks.test(MS[MS$tvsMurSatRep1=="miR-181 target", "log2FoldChange"], MS[MS$tvsMurSatRep1=="Non-target", "log2FoldChange"])


tvsMurSatRep1sig <- data.frame(c("RNA", "RNA", "RNA", "RPF", "RPF", "RPF", "TE", "TE", "TE" #,"MS", "MS", "MS"
                             ),
                        c("MurSatRep1", "both", "miR-181 target", "MurSatRep1", "both", "miR-181 target",
                          "MurSatRep1", "both", "miR-181 target"#,"MurSatRep1", "both", "miR-181 target"
                          ),
                        c(RNAtvsMurSatRep1TTest1$p.value, RNAtvsMurSatRep1TTest2$p.value, RNAtvsMurSatRep1TTest3$p.value,
                          RPFtvsMurSatRep1TTest1$p.value, RPFtvsMurSatRep1TTest2$p.value, RPFtvsMurSatRep1TTest3$p.value,
                          TEtvsMurSatRep1TTest1$p.value, TEtvsMurSatRep1TTest2$p.value, TEtvsMurSatRep1TTest3$p.value#, 
                          #MStvsMurSatRep1TTest1$p.value, MStvsMurSatRep1TTest2$p.value, MStvsMurSatRep1TTest3$p.value
                          ),
                        c(RNAtvsMurSatRep1ksTest1$p.value, RNAtvsMurSatRep1ksTest2$p.value, RNAtvsMurSatRep1ksTest3$p.value,
                          RPFtvsMurSatRep1ksTest1$p.value, RPFtvsMurSatRep1ksTest2$p.value, RPFtvsMurSatRep1ksTest3$p.value, 
                          TEtvsMurSatRep1ksTest1$p.value, TEtvsMurSatRep1ksTest2$p.value, TEtvsMurSatRep1ksTest3$p.value#,
                          #MStvsMurSatRep1ksTest1$p.value, MStvsMurSatRep1ksTest2$p.value, MStvsMurSatRep1ksTest3$p.value
                          ))
colnames(tvsMurSatRep1sig) <- c("Experiment", "Subset", "Welch Two Sample t-test", "Asymptotic two-sample Kolmogorov-Smirnov test")
tvsMurSatRep1sig

#export
write_xlsx(tvsMurSatRep1sig, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/tvsMurSatRep1significance.xlsx")
```


## metagene plot
```{r metageneplot}
MMSAT4$id <- "MMsat4"
MurSatRep1$id <- "MurSatRep1"

metaob <- c(MMSAT4, MurSatRep1)

metael <- metaGeneProfile(object = metaob, annotation = gff23path, group = "id")
metasat4 <- metaGeneProfile(object = MMSAT4, annotation = gff23path)
metamur <- metaGeneProfile(object = MurSatRep1, annotation = gff23path)

metasat4
metamur

metael


pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/metagene_profile_both.pdf", width=2, height = 2)
metael
dev.off()


# metagene plot that only contains genes that also have Mres 

mmsat4MRE <- subsetByOverlaps(MMSAT4 , makeGRangesFromDataFrame(mmsat4frame))
MurSatRep1MRE <- subsetByOverlaps(MurSatRep1 , makeGRangesFromDataFrame(MurSatRep1frame))

mmsat4MRE$id <- "MMsat4"
MurSatRep1MRE$id <- "MurSatRep1"

metaob2 <- c(mmsat4MRE, MurSatRep1MRE)

metael2 <- metaGeneProfile(object = metaob2, annotation = gff23path, group = "id")
metael2

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/metagene_profile_both_onlywithMRE.pdf", width=2, height = 2)
metael2
dev.off()

```



## venn diagram
```{r vennmmsat4}

vset <- list(mmsat4frame[!duplicated(mmsat4frame$gene_id) & mmsat4frame$gene_id %in% largetframe$geneID,"gene_id"], MurSatRep1frame[!duplicated(MurSatRep1frame$gene_id) & MurSatRep1frame$gene_id %in% largetframe$geneID,"gene_id"])
names(vset) <- c("MMSAT4", "MurSatRep1")

vplot <- plot(euler(vset, shape = "ellipse"), quantities = TRUE, main="Genes by contained motifes" , fills=c(farbe1, farbe2)) 


pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/Venn_motifes.pdf", width=2, height = 2)
vplot
dev.off()


vsetECDF <- list(mmsat4frame[!duplicated(mmsat4frame$gene_id) & mmsat4frame$gene_id %in% largetframe$geneID & mmsat4frame$gene_id %in% sub("\\..*", "", RNA$Gene) & mmsat4frame$gene_id %in% sub("\\..*", "", RPF$Gene) ,"gene_id"], MurSatRep1frame[!duplicated(MurSatRep1frame$gene_id) & MurSatRep1frame$gene_id %in% largetframe$geneID & MurSatRep1frame$gene_id %in% sub("\\..*", "", RNA$Gene) & MurSatRep1frame$gene_id %in% sub("\\..*", "", RPF$Gene),"gene_id"])
names(vsetECDF) <- c("MMSAT4", "MurSatRep1")

vplotecdf <- plot(euler(vsetECDF, shape = "ellipse"), quantities = TRUE, main="Genes by contained motifes" , fills=c(farbe1, farbe2)) 


pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/Venn_motifes_likeECDF.pdf", width=2, height = 2)
vplotecdf
dev.off()

```


# histogram MREs per gene with repeat
```{r mresPerGeneRep}
histmremmsat4 <- as.data.frame(table(largetframe[largetframe$geneID %in% mmsat4frame$gene_id, "geneID"]))
dim(histmremmsat4)

mmsat4GMRElist <- as.data.frame(table(largetframe[largetframe$geneID %in% mmsat4frame$gene_id, "geneName"]))
colnames(mmsat4GMRElist) = c("Gene", "Number of MREs")
write.csv(mmsat4GMRElist, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/genelist_MMsat4withMRE.csv")

histmremmsat4P <- ggplot(histmremmsat4, aes(y=Freq)) + 
  geom_histogram(binwidth = 1, colour="black", fill="#B4B4B4") +
  theme_paper() +
  scale_x_continuous(expand = c(0,0)) +
  scale_y_reverse()

histmremmsat4P

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MRE_histogram_mres_per_MMSAT4.pdf", width=2, height = 2)
histmremmsat4P
dev.off()

histmremursat <- as.data.frame(table(largetframe[largetframe$geneID %in% MurSatRep1frame$gene_id, "geneID"]))
dim(histmremursat)

mursatrep1GMRElist <- as.data.frame(table(largetframe[largetframe$geneID %in% MurSatRep1frame$gene_id, "geneName"]))
colnames(mursatrep1GMRElist) = c("Gene", "Number of MREs")
write.csv(mursatrep1GMRElist, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/genelist_MursatRep1withMRE.csv")

histmremursatP<- ggplot(histmremursat, aes(y=Freq)) + 
  geom_histogram(binwidth = 1, colour="black", fill="#B4B4B4") +
  theme_paper() +
  scale_x_continuous(expand = c(0,0)) +
  scale_y_reverse()

histmremursatP

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MRE_histogram_mres_per_MurSatRep1.pdf", width=2, height = 2)
histmremursatP
dev.off()

#like ecdf (occurs in RNA and RPF)
histmremmsat4ECDF <- as.data.frame(table(largetframe[largetframe$geneID %in% mmsat4frame$gene_id & largetframe$geneID %in% sub("\\..*", "", RNA$Gene) & largetframe$geneID %in% sub("\\..*", "", RPF$Gene), "geneID"]))
dim(histmremmsat4ECDF)

histmremmsat4ECDFP <- ggplot(histmremmsat4ECDF, aes(y=Freq)) + 
  geom_histogram(binwidth = 1, colour="black", fill="#B4B4B4") +
  theme_paper() +
  scale_x_continuous(expand = c(0,0)) +
  scale_y_reverse()

histmremmsat4ECDFP

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MRE_histogram_mres_per_MMSAT4_likeECDF.pdf", width=2, height = 2)
histmremmsat4ECDFP
dev.off()

histmremursatECDF <- as.data.frame(table(largetframe[largetframe$geneID %in% MurSatRep1frame$gene_id & largetframe$geneID %in% sub("\\..*", "", RNA$Gene) & largetframe$geneID %in% sub("\\..*", "", RPF$Gene), "geneID"]))
dim(histmremursatECDF)


histmremursatECDFP<- ggplot(histmremursatECDF, aes(y=Freq)) + 
  geom_histogram(binwidth = 1, colour="black", fill="#B4B4B4") +
  theme_paper() +
  scale_x_continuous(expand = c(0,0)) +
  scale_y_reverse()

histmremursatECDFP

pdf("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/MRE_histogram_mres_per_MurSatRep1_likeECDF.pdf", width=2, height = 2)
histmremursatECDFP
dev.off()

```





## Sequence overlap mmsat4 mursatrep1
Just check to make sure these are not the same
```{r seqol}
olmmsat4 <- subsetByOverlaps(MMSAT4, MurSatRep1)
olmursatrep1 <- subsetByOverlaps(MurSatRep1, MMSAT4)


#make pie charts
pimmsat <- data.frame(c("MMsat4", "Overlap"),
                      c(length(MMSAT4)-length(olmmsat4), length(olmmsat4)))
colnames(pimmsat) <- c("Group", "Count")

pimursatrep1 <- data.frame(c("MurSatRep1", "Overlap"),
                      c(length(MurSatRep1)-length(olmursatrep1), length(olmursatrep1)))
colnames(pimursatrep1) <- c("Group", "Count")

pimmsatP <- ggplot(pimmsat, aes(x="", y=Count, fill=Group)) + 
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  geom_text(aes(label = Count),
            position = position_stack(vjust = 0.5)) 

pimmsatP

pimursatrep1P <- ggplot(pimursatrep1, aes(x="", y=Count, fill=Group)) + 
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  geom_text(aes(label = Count),
            position = position_stack(vjust = 0.5)) 

pimursatrep1P

```


# MAplots of targets with satellites
```{r masat}
#mmsat4 RNA
MAmmsat4RNA <- ggplot(RNA, aes(x=log10(baseMean), y=log2FoldChange, fill=factor(tvsmmsat4, levels = c("Non-target", "miR-181 target", "MMsat4", "both"))), colour="black") + 
  geom_point(shape=21, size=1) + 
  scale_fill_manual(values = c(farbeneg, farbe1, farbe2, farbe3)) +
  geom_point(data = RNA[RNA$tvsmmsat4=="Non-target",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour=farbeneg, fill=farbeneg, size=1) +
  geom_point(data = RNA[RNA$tvsmmsat4=="miR-181 target",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe1, size=2) +
  geom_point(data = RNA[RNA$tvsmmsat4=="MMsat4",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe2, size=2) +
  geom_point(data = RNA[RNA$tvsmmsat4=="both",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe3, size=3) +
  geom_hline(yintercept = 0, linetype="dashed", colour="black")+
  theme_paper() +
  coord_cartesian(ylim = c(-2,2), xlim = c(1,5))
MAmmsat4RNA

##mmsat4 RPF
MAmmsat4RPF <- ggplot(RPF, aes(x=log10(baseMean), y=log2FoldChange, fill=factor(tvsmmsat4, levels = c("Non-target", "miR-181 target", "MMsat4", "both"))), colour="black") + 
  geom_point(shape=21, size=1) + 
  scale_fill_manual(values = c(farbeneg, farbe1, farbe2, farbe3)) +
  geom_point(data = RPF[RPF$tvsmmsat4=="Non-target",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour=farbeneg, fill=farbeneg, size=1) +
  geom_point(data = RPF[RPF$tvsmmsat4=="miR-181 target",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe1, size=2) +
  geom_point(data = RPF[RPF$tvsmmsat4=="MMsat4",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe2, size=2) +
  geom_point(data = RPF[RPF$tvsmmsat4=="both",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe3, size=3) +
  geom_hline(yintercept = 0, linetype="dashed", colour="black")+
  theme_paper() +
  coord_cartesian(ylim = c(-2,2), xlim = c(1,5))
MAmmsat4RPF

#Mursatrep1 RNA
MAmursatrep1RNA <- ggplot(RNA, aes(x=log10(baseMean), y=log2FoldChange, fill=factor(tvsMurSatRep1, levels = c("Non-target", "miR-181 target", "MurSatRep1", "both"))), colour="black") + 
  geom_point(shape=21, size=1) + 
  scale_fill_manual(values = c(farbeneg, farbe1, farbe2, farbe3)) +
  geom_point(data = RNA[RNA$tvsMurSatRep1=="Non-target",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour=farbeneg, fill=farbeneg, size=1) +
  geom_point(data = RNA[RNA$tvsMurSatRep1=="miR-181 target",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe1, size=2) +
  geom_point(data = RNA[RNA$tvsMurSatRep1=="MurSatRep1",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe2, size=2) +
  geom_point(data = RNA[RNA$tvsMurSatRep1=="both",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe3, size=3) +
  geom_hline(yintercept = 0, linetype="dashed", colour="black")+
  theme_paper() +
  coord_cartesian(ylim = c(-2,2), xlim = c(1,5))
MAmursatrep1RNA

#Mursatrep1 RPF
MAmursatrep1RPF <- ggplot(RPF, aes(x=log10(baseMean), y=log2FoldChange, fill=factor(tvsMurSatRep1, levels = c("Non-target", "miR-181 target", "MurSatRep1", "both"))), colour="black") + 
  geom_point(shape=21, size=1) + 
  scale_fill_manual(values = c(farbeneg, farbe1, farbe2, farbe3)) +
  geom_point(data = RPF[RPF$tvsMurSatRep1=="Non-target",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour=farbeneg, fill=farbeneg, size=1) +
  geom_point(data = RPF[RPF$tvsMurSatRep1=="miR-181 target",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe1, size=2) +
  geom_point(data = RPF[RPF$tvsMurSatRep1=="MurSatRep1",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe2, size=2) +
  geom_point(data = RPF[RPF$tvsMurSatRep1=="both",], aes(x=log10(baseMean), y=log2FoldChange), shape=21, colour="black", fill=farbe3, size=3) +
  geom_hline(yintercept = 0, linetype="dashed", colour="black")+
  theme_paper() +
  coord_cartesian(ylim = c(-2,2), xlim = c(1,5))
MAmursatrep1RPF


```


# Session info
```{r sessioninfo}
sessionInfo()
```






