---
title: "Fig4 ECDF plots"
author: "Nikita Verheyden"
date: "2023-05-16"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


# Setup

dir
```{r dir}
setwd("D:/Krueger_Lab/Publications/miR181_paper/Figure4")
```

## packages
```{r packages}
library(ggplot2)
library(rtracklayer)
library(dplyr)


```

## data
```{r data}
#Ribo profiling
RNA <- read.csv("D:/Krueger_Lab/Publications/miR181_paper/Figure3/RNA_masterframe.csv")
RPF <- read.csv("D:/Krueger_Lab/Publications/miR181_paper/Figure3/RPF_masterframe.csv")

#load the gtf file to compare genes
gff23 <- import.gff3("D:/Krueger_Lab/Ribo_Profiling/run15112022M23/ref_genome/gencode.vM23.annotation.gff3.gz")

#targets
larget <- readRDS("D:/Krueger_Lab/Publications/miR181_paper/Figure3/mir181_bs_with_seeds.rds")
largetframe <- as.data.frame(larget)
#MMsat4
repeat_masker <- readRDS("D:/Krueger_Lab/Publications/miR181_paper/Figure2/MMsat4/repeat_masker.rds")
MMSAT4 <- repeat_masker[repeat_masker$repName == "MMSAT4"]

```
## colours
```{r colours}
#colours
farbeneg <- "#b4b4b4"
farbe1 <- "#0073C2FF"
farbe2 <- "#EFC000FF"
farbe3 <- "#CD534CFF"
farbe4 <- "#7AA6DCFF"
farbe5 <- "#868686FF"
farbe6 <- "#003C67FF"
farbe7 <- "#8F7700FF"
farbe8 <- "#3B3B3BFF"
farbe9 <- "#A73030FF"
farbe10 <- "#4A6990FF"
farbe11 <- "#FF6F00FF"
farbe12 <- "#C71000FF"
farbe13 <- "#008EA0FF"
farbe14 <- "#8A4198FF"
farbe15 <- "#5A9599FF"
farbe16 <- "#FF6348FF"
  
RNApcol <- "#b56504"
RNAncol <- "#027d73"
RPFpcol <- "#c4c404"
RPFncol <- "#8d0391"
```


# inspect targetdata

We're keeping all of those targets for now but will analyze the in ecdf plots
```{r set}
table(largetframe$set)

colnames(largetframe)
``` 


# ECDF plots

each code chunk is a split of the main target table that is then used for a specific ecdf plot


## datasets

```{r datasets, fig.show="hold", out.width="50%"}
#RNA
RNA$targetset <- "Non-target"
RNA$targetset[RNA$gene_symbol %in% largetframe[largetframe$set == "ago_bs_mir181_chi", "geneName"]] <- "ago_bs_mir181_chi"
RNA$targetset[RNA$gene_symbol %in% largetframe[largetframe$set == "mir181_enriched", "geneName"]] <- "mir181_enriched"
RNA$targetset[RNA$gene_symbol %in% largetframe[largetframe$set == "ago_bs_mir181_chi&mir181_enriched", "geneName"]] <- "both"

table(RNA$targetset)

#RPF
RPF$targetset <- "Non-target"
RPF$targetset[RPF$gene_symbol %in% largetframe[largetframe$set == "ago_bs_mir181_chi", "geneName"]] <- "ago_bs_mir181_chi"
RPF$targetset[RPF$gene_symbol %in% largetframe[largetframe$set == "mir181_enriched", "geneName"]] <- "mir181_enriched"
RPF$targetset[RPF$gene_symbol %in% largetframe[largetframe$set == "ago_bs_mir181_chi&mir181_enriched", "geneName"]] <- "both"

table(RPF$targetset)


# ecdf plots
#RNA
setECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(targetset, levels = c("Non-target", "ago_bs_mir181_chi", "mir181_enriched", "both")))) + 
  stat_ecdf(geom="step", size=1) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + theme_bw() +
  theme(legend.position = c(0.8, 0.35), legend.title = element_blank(), 
        legend.background = element_rect(colour = "transparent", fill="transparent"), 
        axis.title=element_text(size=16),plot.title = element_text(size=16, face = "bold"), aspect.ratio = 0.8) +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA region only single BS")

setECDFRNA

#RPF
setECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(targetset, levels = c("Non-target", "ago_bs_mir181_chi", "mir181_enriched", "both")))) + 
  stat_ecdf(geom="step", size=1) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + theme_bw() +
  theme(legend.position = c(0.8, 0.35), legend.title = element_blank(), 
        legend.background = element_rect(colour = "transparent", fill="transparent"), 
        axis.title=element_text(size=16),plot.title = element_text(size=16, face = "bold"), aspect.ratio = 0.8) +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF region only single BS")

setECDFRPF

```


## region (single targets)
```{r region, fig.show="hold", out.width="50%"}
#get number of binding sites per gene to be able to sort for singles
bsnum <- as.data.frame(table(largetframe$geneName))
colnames(bsnum) <- c("geneName", "BS_number")

#RNA
RNA$region_single <- "Non-target"
RNA$region_single[RNA$gene_symbol %in% largetframe[largetframe$region == "utr5", "geneName"]] <- "5'UTR"
RNA$region_single[RNA$gene_symbol %in% largetframe[largetframe$region == "cds", "geneName"]] <- "CDS"
RNA$region_single[RNA$gene_symbol %in% largetframe[largetframe$region == "utr3", "geneName"]] <- "3'UTR"
RNA$region_single[RNA$gene_symbol %in% bsnum[bsnum$BS_number > 1, "geneName"]] <- "multiple"

table(RNA$region_single)

#RPF
RPF$region_single <- "Non-target"
RPF$region_single[RPF$gene_symbol %in% largetframe[largetframe$region == "utr5", "geneName"]] <- "5'UTR"
RPF$region_single[RPF$gene_symbol %in% largetframe[largetframe$region == "cds", "geneName"]] <- "CDS"
RPF$region_single[RPF$gene_symbol %in% largetframe[largetframe$region == "utr3", "geneName"]] <- "3'UTR"
RPF$region_single[RPF$gene_symbol %in% bsnum[bsnum$BS_number > 1, "geneName"]] <- "multiple"

table(RPF$region_single)


# ECDF plots
#RNA
regsingECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(region_single, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "multiple")))) + 
  stat_ecdf(geom="step", size=1) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbeneg)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + theme_bw() +
  theme(legend.position = c(0.8, 0.35), legend.title = element_blank(), 
        legend.background = element_rect(colour = "transparent", fill="transparent"), 
        axis.title=element_text(size=16),plot.title = element_text(size=16, face = "bold"), aspect.ratio = 0.8) +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA region only single BS")

regsingECDFRNA

#RPF
regsingECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(region_single, levels = c("Non-target", "5'UTR", "CDS", "3'UTR", "multiple")))) + 
  stat_ecdf(geom="step", size=1) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3, farbeneg)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + theme_bw() +
  theme(legend.position = c(0.8, 0.35), legend.title = element_blank(), 
        legend.background = element_rect(colour = "transparent", fill="transparent"), 
        axis.title=element_text(size=16),plot.title = element_text(size=16, face = "bold"), aspect.ratio = 0.8) +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF region only single BS")

regsingECDFRPF 

```

## number of target sites

```{r targetnum, fig.show="hold", out.width="50%"}
colnames(bsnum) <- c("gene_symbol", "BS_number")

#RNA
RNAnum <- left_join(RNA, bsnum, by="gene_symbol")
RNAnum$BS_number[is.na(RNAnum$BS_number)] <- "Non-target"
RNAnum$BS_num_plot <- ifelse(RNAnum$BS_number == "Non-target", "Non-target",
                             ifelse(RNAnum$BS_number == 1, "One bs",
                                    ifelse(RNAnum$BS_number == 2, "Two bs", "More")))


#RPF
RPFnum <- left_join(RPF, bsnum, by="gene_symbol")
RPFnum$BS_number[is.na(RPFnum$BS_number)] <- "Non-target"
RPFnum$BS_num_plot <- ifelse(RPFnum$BS_number == "Non-target", "Non-target",
                             ifelse(RPFnum$BS_number == 1, "One bs",
                                    ifelse(RPFnum$BS_number == 2, "Two bs", "More")))

#ecdf plots
#RNA
numECDFRNA <- ggplot(RNAnum, aes(as.numeric(log2FoldChange), colour=factor(BS_num_plot, levels = c("Non-target", "One bs", "Two bs", "More")))) + 
  stat_ecdf(geom="step", size=1) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + theme_bw() +
  theme(legend.position = c(0.8, 0.35), legend.title = element_blank(), 
        legend.background = element_rect(colour = "transparent", fill="transparent"), 
        axis.title=element_text(size=16),plot.title = element_text(size=16, face = "bold"), aspect.ratio = 0.8) +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA number of BS")

numECDFRNA

#RPF
numECDFRPF <- ggplot(RPFnum, aes(as.numeric(log2FoldChange), colour=factor(BS_num_plot, levels = c("Non-target", "One bs", "Two bs", "More")))) + 
  stat_ecdf(geom="step", size=1) +
  scale_colour_manual(values = c("black", farbe1, farbe2, farbe3)) +
  coord_cartesian(xlim = c(-0.75, 0.75)) + theme_bw() +
  theme(legend.position = c(0.8, 0.35), legend.title = element_blank(), 
        legend.background = element_rect(colour = "transparent", fill="transparent"), 
        axis.title=element_text(size=16),plot.title = element_text(size=16, face = "bold"), aspect.ratio = 0.8) +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF number of BS")

numECDFRPF

```




# session info
```{r sessioninfo}
sessionInfo()
```


