---
title: "Annotation file preprocessing for binding sites"
author: "Melina Klostermann"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  BiocStyle::html_document:
    toc_float: TRUE
    code_folding: hide
    toc: TRUE
    number_sections: yes
    fig_caption: yes
  # pdf_document:
  #   fig_caption: true
  #   fig_crop: true
  #   toc: true
  #   toc_depth: 1
  #   number_sections: true
---

```{r setup, include=FALSE}
require("knitr")
knitr::opts_chunk$set(warning=FALSE, message=FALSE, cache=F, cache.lazy = FALSE)
tidy.opts=list(width.cutoff=80,tidy=TRUE, echo=FALSE)
```

# Libraries and Settings

```{r libraries_settings}
# ----------------------------------------
# libraries
# ----------------------------------------

library(GenomicRanges)
library(GenomicFeatures)
library(rtracklayer)
library(knitr)
library(dplyr)
library(kableExtra)

# ----------------------------------------
# settings
# ----------------------------------------
here <- here::here()

out <- paste0(here,"/Supporting_scripts/annotation_preprocessing/")

```

# What was done?

- Annotations are downloaded as gff3 from GENCODE (Release m23 GRCm38.p6). This release is relatively old (05.2019), but it is consitent with current miRbase Release 22, which uses the GRCm38 (mm10) genome assembly. 
- Annotations are kept as is with no further filtering.
\newline
- A txDb file of the annotation is made.
- Also the gene annotations are extracted as GRanges object.
\newline
- Both files are saved and later reloaded in the binding site definition scripts.


# Annotation txdb

```{r}
# ----------------------------------------
# Process whole annotation
# ----------------------------------------

# load annotation
anno <- readGFFAsGRanges("/Users/melinaklostermann/Documents/projects/anno/gencodevM23/gencode.vM23.annotation.gff3")

# shorten gene ids
anno$geneID <- sub("\\..*", "", anno$gene_id)

# make txdb and save
annoDb = makeTxDbFromGRanges(anno)
saveDb(annoDb, paste0(out, "annotation.db"))
saveRDS(anno, paste0(out, "annotation.rds"))


kable(head(anno)) %>%
  kable_material(c("striped", "hover")) %>%
  scroll_box(width = "100%", height = "500px")
```

# Get genes

```{r}
# ----------------------------------------
# Retrieve genes
# ----------------------------------------

# get genes
gns = genes(annoDb)

# get back meta data
idx = match(gns$gene_id, anno$gene_id)
mcols(gns) = cbind(mcols(gns), mcols(anno)[idx,])

# clean gene names and meta data
names(gns) = sub("\\..*", "", names(gns))
meta = data.frame(gene_id = gns$gene_id, gene_name = gns$gene_name, gene_type = gns$gene_type)
mcols(gns) = meta
gns$geneID = names(gns)



# save genes
saveRDS(gns, paste0(out, "gene_annotation.rds"))
rtracklayer::export.bed(gns, paste0(out, "gene_annotation.bed"))


kable(head(gns))  %>%
  kable_material(c("striped", "hover")) %>%
  scroll_box(width = "100%", height = "500px")
```

# Session Info

```{r}
sessionInfo()


```