---
title: "Sattelite_repeats_ECDF_plots"
author:
- name: Nikita Verheyden
  affiliation: JLU Giessen
  date: "`r Sys.Date()`"
output:
  BiocStyle::html_document:
      toc: TRUE
      toc_float: TRUE
      code_folding: hide

header-includes:
- \makeatletter\renewcommand*{\fps@figure}{h}\makeatother
- \usepackage{placeins}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


# Setup

dir
```{r dir}
setwd("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/06_RF_MMSAT4")
```


## Packages
```{r packages}
source("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Supporting_scripts/themes/theme_paper.R")
library(ggplot2)
library(rtracklayer)
library(dplyr)
library(tibble)
library(Rtsne)
library(cliProfiler)
library(GenomicRanges)
library(cowplot)
library(eulerr)
library(readxl)
library(biomaRt)
library(writexl)
library(ggrastr)

```

## Data
```{r data}
#Ribo profiling
RNA <- read.csv("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/01_Ribosome_profiling_pipeline/RNA_masterframe.csv")
RPF <- read.csv("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/01_Ribosome_profiling_pipeline/RPF_masterframe.csv")

#load the gtf file to compare genes
gff23path <- file.path("C:/Users/nikit/Documents/Krueger_Lab/Ribo_Profiling/run15112022M23/ref_genome", "gencode.vM23.annotation.gff3.gz")

#targets
larget <- readRDS("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper_at_submission/Figure4/mir181_bs_with_seeds_transcripts.rds")
largetframe <- as.data.frame(larget)
table(largetframe$set)
largetframe <- largetframe %>%
  subset(set %in% c("ago_bs_mir181_chi&mir181_enriched", "mir181_enriched"))


#Translational efficiency
TE <- read.csv("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Supporting_scripts/deltaTE/TE_m23_genesInRNAandRPF.csv")



#melina repeats
melrep <- readRDS("C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure3/rep_on_transcripts.rds")
MMSAT4 <- melrep[melrep$repName == "MMSAT4"]
MurSatRep1 <- melrep[melrep$repName == "MurSatRep1"]
```


## colours
```{r colours}
#colours
farbeneg <- "#b4b4b4"
farbe1 <- "#0073C2FF"
farbe2 <- "#EFC000FF"
farbe3 <- "#CD534CFF"
farbe4 <- "#7AA6DCFF"
farbe5 <- "#868686FF"
farbe6 <- "#003C67FF"
farbe7 <- "#8F7700FF"
farbe8 <- "#3B3B3BFF"
farbe9 <- "#A73030FF"
farbe10 <- "#4A6990FF"
farbe11 <- "#FF6F00FF"
farbe12 <- "#C71000FF"
farbe13 <- "#008EA0FF"
farbe14 <- "#8A4198FF"
farbe15 <- "#5A9599FF"
farbe16 <- "#FF6348FF"
  
RNApcol <- "#b56504"
RNAncol <- "#027d73"
RPFpcol <- "#c4c404"
RPFncol <- "#8d0391"
```

## axis limits
```{r axislim}
xlim <- 0.5
zoomfac <- 0.45
```


# mmsat4
```{r mmsat4}
mmsat4frame <- as.data.frame(MMSAT4)


#RNA
RNA$tvsmmsat4 <- "Non-target"
RNA$tvsmmsat4[sub("\\..*", "", RNA$Gene) %in% mmsat4frame$gene_id] <- "MMsat4"
RNA$tvsmmsat4[sub("\\..*", "", RNA$Gene) %in% largetframe$geneID] <- "miR-181 target"
RNA$tvsmmsat4[sub("\\..*", "", RNA$Gene) %in% largetframe$geneID & sub("\\..*", "", RNA$Gene) %in% mmsat4frame$gene_id] <- "both"
table(RNA$tvsmmsat4)

#RPF
RPF$tvsmmsat4 <- "Non-target"
RPF$tvsmmsat4[sub("\\..*", "", RPF$Gene) %in% mmsat4frame$gene_id] <- "MMsat4"
RPF$tvsmmsat4[sub("\\..*", "", RPF$Gene) %in% largetframe$geneID] <- "miR-181 target"
RPF$tvsmmsat4[sub("\\..*", "", RPF$Gene) %in% largetframe$geneID & sub("\\..*", "", RPF$Gene) %in% mmsat4frame$gene_id] <- "both"
table(RPF$tvsmmsat4)


#TE
TE$tvsmmsat4 <- "Non-target"
TE$tvsmmsat4[sub("\\..*", "", TE$Gene) %in% mmsat4frame$gene_id] <- "MMsat4"
TE$tvsmmsat4[sub("\\..*", "", TE$Gene) %in% largetframe$geneID] <- "miR-181 target"
TE$tvsmmsat4[sub("\\..*", "", TE$Gene) %in% largetframe$geneID & sub("\\..*", "", TE$Gene) %in% mmsat4frame$gene_id] <- "both"
table(TE$tvsmmsat4)



#RNA
tolECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(tvsmmsat4, levels = c("Non-target", "MMsat4", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RNAncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA Mmsat4 vs target")

tolECDFRNA


#RPF
tolECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(tvsmmsat4, levels = c("Non-target", "MMsat4", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF Mmsat4 vs target")

tolECDFRPF


#TE
tolECDFTE <- ggplot(TE, aes(as.numeric(log2FoldChange), colour=factor(tvsmmsat4, levels = c("Non-target", "MMsat4", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("TE Mmsat4 vs target")

tolECDFTE

#export RNA
pdf("MMSAT4ECDF_RNA.pdf", width=2, height = 2)
tolECDFRNA
dev.off()


#export RPF
pdf("MMSAT4ECDF_RPF.pdf", width=2, height = 2)
tolECDFRPF
dev.off()


#export TE
pdf("MMSAT4ECDF_TE.pdf", width=2, height = 2)
tolECDFTE
dev.off()


table(RPF$tvsmmsat4)
table(RNA$tvsmmsat4)
table(TE$tvsmmsat4)
```


## density mmsat4 TE
```{R TEdensityMMsat4}
TEdensmmsat4 <- ggplot(TE, aes(x=log2FoldChange, colour=factor(tvsmmsat4, levels = c("MMsat4", "miR-181 target", "both", "Non-target")))) +
  geom_density() +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c( farbe1, farbe3, RPFncol, "black")) +
  theme_paper() +
  coord_cartesian(xlim = c(-1,1)) +
  ggtitle("TE")

TEdensmmsat4

pdf("MMsat4Density_TE.pdf", width=2, height = 2)
TEdensmmsat4
dev.off()

```

## Significance
```{r significancemmsat4}
RNAtvsmmsat4TTest1 <- t.test(RNA[RNA$tvsmmsat4=="MMsat4", "log2FoldChange"], RNA[RNA$tvsmmsat4=="Non-target", "log2FoldChange"])
RNAtvsmmsat4TTest2 <- t.test(RNA[RNA$tvsmmsat4=="both", "log2FoldChange"], RNA[RNA$tvsmmsat4=="Non-target", "log2FoldChange"])
RNAtvsmmsat4TTest3 <- t.test(RNA[RNA$tvsmmsat4=="miR-181 target", "log2FoldChange"], RNA[RNA$tvsmmsat4=="Non-target", "log2FoldChange"])

RPFtvsmmsat4TTest1 <- t.test(RPF[RPF$tvsmmsat4=="MMsat4", "log2FoldChange"], RPF[RPF$tvsmmsat4=="Non-target", "log2FoldChange"])
RPFtvsmmsat4TTest2 <- t.test(RPF[RPF$tvsmmsat4=="both", "log2FoldChange"], RPF[RPF$tvsmmsat4=="Non-target", "log2FoldChange"])
RPFtvsmmsat4TTest3 <- t.test(RPF[RPF$tvsmmsat4=="miR-181 target", "log2FoldChange"], RPF[RPF$tvsmmsat4=="Non-target", "log2FoldChange"])

TEtvsmmsat4TTest1 <- t.test(TE[TE$tvsmmsat4=="MMsat4", "log2FoldChange"], TE[TE$tvsmmsat4=="Non-target", "log2FoldChange"])
TEtvsmmsat4TTest2 <- t.test(TE[TE$tvsmmsat4=="both", "log2FoldChange"], TE[TE$tvsmmsat4=="Non-target", "log2FoldChange"])
TEtvsmmsat4TTest3 <- t.test(TE[TE$tvsmmsat4=="miR-181 target", "log2FoldChange"], TE[TE$tvsmmsat4=="Non-target", "log2FoldChange"])


#kolmogorov-smirnov
RNAtvsmmsat4ksTest1 <- ks.test(RNA[RNA$tvsmmsat4=="MMsat4", "log2FoldChange"], RNA[RNA$tvsmmsat4=="Non-target", "log2FoldChange"])
RNAtvsmmsat4ksTest2 <- ks.test(RNA[RNA$tvsmmsat4=="both", "log2FoldChange"], RNA[RNA$tvsmmsat4=="Non-target", "log2FoldChange"])
RNAtvsmmsat4ksTest3 <- ks.test(RNA[RNA$tvsmmsat4=="miR-181 target", "log2FoldChange"], RNA[RNA$tvsmmsat4=="Non-target", "log2FoldChange"])

RPFtvsmmsat4ksTest1 <- ks.test(RPF[RPF$tvsmmsat4=="MMsat4", "log2FoldChange"], RPF[RPF$tvsmmsat4=="Non-target", "log2FoldChange"])
RPFtvsmmsat4ksTest2 <- ks.test(RPF[RPF$tvsmmsat4=="both", "log2FoldChange"], RPF[RPF$tvsmmsat4=="Non-target", "log2FoldChange"])
RPFtvsmmsat4ksTest3 <- ks.test(RPF[RPF$tvsmmsat4=="miR-181 target", "log2FoldChange"], RPF[RPF$tvsmmsat4=="Non-target", "log2FoldChange"])

TEtvsmmsat4ksTest1 <- ks.test(TE[TE$tvsmmsat4=="MMsat4", "log2FoldChange"], TE[TE$tvsmmsat4=="Non-target", "log2FoldChange"])
TEtvsmmsat4ksTest2 <- ks.test(TE[TE$tvsmmsat4=="both", "log2FoldChange"], TE[TE$tvsmmsat4=="Non-target", "log2FoldChange"])
TEtvsmmsat4ksTest3 <- ks.test(TE[TE$tvsmmsat4=="miR-181 target", "log2FoldChange"], TE[TE$tvsmmsat4=="Non-target", "log2FoldChange"])



tvsmmsat4sig <- data.frame(c("RNA", "RNA", "RNA", "RPF", "RPF", "RPF", "TE", "TE", "TE" 
                             ),
                        c("MMsat4", "both", "miR-181 target", "MMsat4", "both", "miR-181 target",
                          "MMsat4", "both", "miR-181 target"
                          ),
                        c(RNAtvsmmsat4TTest1$p.value, RNAtvsmmsat4TTest2$p.value, RNAtvsmmsat4TTest3$p.value,
                          RPFtvsmmsat4TTest1$p.value, RPFtvsmmsat4TTest2$p.value, RPFtvsmmsat4TTest3$p.value,
                          TEtvsmmsat4TTest1$p.value, TEtvsmmsat4TTest2$p.value, TEtvsmmsat4TTest3$p.value
                          ),
                        c(RNAtvsmmsat4ksTest1$p.value, RNAtvsmmsat4ksTest2$p.value, RNAtvsmmsat4ksTest3$p.value,
                          RPFtvsmmsat4ksTest1$p.value, RPFtvsmmsat4ksTest2$p.value, RPFtvsmmsat4ksTest3$p.value, 
                          TEtvsmmsat4ksTest1$p.value, TEtvsmmsat4ksTest2$p.value, TEtvsmmsat4ksTest3$p.value
                          ))
colnames(tvsmmsat4sig) <- c("Experiment", "Subset", "Welch Two Sample t-test", "Asymptotic two-sample Kolmogorov-Smirnov test")
tvsmmsat4sig

#export
write_xlsx(tvsmmsat4sig, "tvsmmsat4significance.xlsx")
```



# MurSatRep1
```{r MurSatRep1}
MurSatRep1frame <- as.data.frame(MurSatRep1)

#RNA
RNA$tvsMurSatRep1 <- "Non-target"
RNA$tvsMurSatRep1[sub("\\..*", "", RNA$Gene) %in% MurSatRep1frame$gene_id] <- "MurSatRep1"
RNA$tvsMurSatRep1[sub("\\..*", "", RNA$Gene) %in% largetframe$geneID] <- "miR-181 target"
RNA$tvsMurSatRep1[sub("\\..*", "", RNA$Gene) %in% largetframe$geneID & sub("\\..*", "", RNA$Gene) %in% MurSatRep1frame$gene_id] <- "both"
table(RNA$tvsMurSatRep1)

#RPF
RPF$tvsMurSatRep1 <- "Non-target"
RPF$tvsMurSatRep1[sub("\\..*", "", RPF$Gene) %in% MurSatRep1frame$gene_id] <- "MurSatRep1"
RPF$tvsMurSatRep1[sub("\\..*", "", RPF$Gene) %in% largetframe$geneID] <- "miR-181 target"
RPF$tvsMurSatRep1[sub("\\..*", "", RPF$Gene) %in% largetframe$geneID & sub("\\..*", "", RPF$Gene) %in% MurSatRep1frame$gene_id] <- "both"
table(RPF$tvsMurSatRep1)


#TE
TE$tvsMurSatRep1 <- "Non-target"
TE$tvsMurSatRep1[sub("\\..*", "", TE$Gene) %in% MurSatRep1frame$gene_id] <- "MurSatRep1"
TE$tvsMurSatRep1[sub("\\..*", "", TE$Gene) %in% largetframe$geneID] <- "miR-181 target"
TE$tvsMurSatRep1[sub("\\..*", "", TE$Gene) %in% largetframe$geneID & sub("\\..*", "", TE$Gene) %in% MurSatRep1frame$gene_id] <- "both"
table(TE$tvsMurSatRep1)


#RNA
murECDFRNA <- ggplot(RNA, aes(as.numeric(log2FoldChange), colour=factor(tvsMurSatRep1, levels = c("Non-target", "MurSatRep1", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RNA MurSatRep1 vs target")

murECDFRNA


#RPF
murECDFRPF <- ggplot(RPF, aes(as.numeric(log2FoldChange), colour=factor(tvsMurSatRep1, levels = c("Non-target", "MurSatRep1", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("RPF MurSatRep1 vs target")

murECDFRPF



#TE
murECDFTE <- ggplot(TE, aes(as.numeric(log2FoldChange), colour=factor(tvsMurSatRep1, levels = c("Non-target", "MurSatRep1", "miR-181 target", "both")))) + 
  stat_ecdf(geom="step", linewidth=2*zoomfac) +
  geom_vline(xintercept = 0, linewidth=1*zoomfac, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c("black", farbe1, farbe3, RPFncol)) +
  coord_cartesian(xlim = c(-xlim, xlim)) + 
  theme_paper() +
  scale_y_continuous("Cumulative density") + scale_x_continuous("log2FoldChange") +
  ggtitle("TE MurSatRep1 vs target")

murECDFTE



#export RNA
pdf("mursatrep1ECDF_RNA.pdf", width=2, height = 2)
murECDFRNA
dev.off()

#export RPF
pdf("mursatrep1ECDF_RPF.pdf", width=2, height = 2)
murECDFRPF
dev.off()

#export TE
pdf("mursatrep1ECDF_TE.pdf", width=2, height = 2)
murECDFTE
dev.off()


table(RPF$tvsMurSatRep1)
table(RNA$tvsMurSatRep1)
table(TE$tvsMurSatRep1)

```

## density mursatrep1 TE
```{R TEdensitymursatrep1}
TEdensmursatrep1 <- ggplot(TE, aes(x=log2FoldChange, colour=factor(tvsMurSatRep1, levels = c( "MurSatRep1", "miR-181 target", "both", "Non-target")))) +
  geom_density() +
  geom_vline(xintercept = 0, linewidth=1, linetype="dashed", colour=farbeneg) +
  scale_colour_manual(values = c(farbe1, farbe3, RPFncol, "black")) +
  theme_paper() +
  coord_cartesian(xlim = c(-1,1)) +
  ggtitle("TE")

TEdensmursatrep1

pdf("MurSatRep1Density_TE.pdf", width=2, height = 2)
TEdensmursatrep1
dev.off()

```


## Significance
```{r significanceMurSatRep1}
RNAtvsMurSatRep1TTest1 <- t.test(RNA[RNA$tvsMurSatRep1=="MurSatRep1", "log2FoldChange"], RNA[RNA$tvsMurSatRep1=="Non-target", "log2FoldChange"])
RNAtvsMurSatRep1TTest2 <- t.test(RNA[RNA$tvsMurSatRep1=="both", "log2FoldChange"], RNA[RNA$tvsMurSatRep1=="Non-target", "log2FoldChange"])
RNAtvsMurSatRep1TTest3 <- t.test(RNA[RNA$tvsMurSatRep1=="miR-181 target", "log2FoldChange"], RNA[RNA$tvsMurSatRep1=="Non-target", "log2FoldChange"])

RPFtvsMurSatRep1TTest1 <- t.test(RPF[RPF$tvsMurSatRep1=="MurSatRep1", "log2FoldChange"], RPF[RPF$tvsMurSatRep1=="Non-target", "log2FoldChange"])
RPFtvsMurSatRep1TTest2 <- t.test(RPF[RPF$tvsMurSatRep1=="both", "log2FoldChange"], RPF[RPF$tvsMurSatRep1=="Non-target", "log2FoldChange"])
RPFtvsMurSatRep1TTest3 <- t.test(RPF[RPF$tvsMurSatRep1=="miR-181 target", "log2FoldChange"], RPF[RPF$tvsMurSatRep1=="Non-target", "log2FoldChange"])

TEtvsMurSatRep1TTest1 <- t.test(TE[TE$tvsMurSatRep1=="MurSatRep1", "log2FoldChange"], TE[TE$tvsMurSatRep1=="Non-target", "log2FoldChange"])
TEtvsMurSatRep1TTest2 <- t.test(TE[TE$tvsMurSatRep1=="both", "log2FoldChange"], TE[TE$tvsMurSatRep1=="Non-target", "log2FoldChange"])
TEtvsMurSatRep1TTest3 <- t.test(TE[TE$tvsMurSatRep1=="miR-181 target", "log2FoldChange"], TE[TE$tvsMurSatRep1=="Non-target", "log2FoldChange"])


#kolmogorov-smirnov
RNAtvsMurSatRep1ksTest1 <- ks.test(RNA[RNA$tvsMurSatRep1=="MurSatRep1", "log2FoldChange"], RNA[RNA$tvsMurSatRep1=="Non-target", "log2FoldChange"])
RNAtvsMurSatRep1ksTest2 <- ks.test(RNA[RNA$tvsMurSatRep1=="both", "log2FoldChange"], RNA[RNA$tvsMurSatRep1=="Non-target", "log2FoldChange"])
RNAtvsMurSatRep1ksTest3 <- ks.test(RNA[RNA$tvsMurSatRep1=="miR-181 target", "log2FoldChange"], RNA[RNA$tvsMurSatRep1=="Non-target", "log2FoldChange"])

RPFtvsMurSatRep1ksTest1 <- ks.test(RPF[RPF$tvsMurSatRep1=="MurSatRep1", "log2FoldChange"], RPF[RPF$tvsMurSatRep1=="Non-target", "log2FoldChange"])
RPFtvsMurSatRep1ksTest2 <- ks.test(RPF[RPF$tvsMurSatRep1=="both", "log2FoldChange"], RPF[RPF$tvsMurSatRep1=="Non-target", "log2FoldChange"])
RPFtvsMurSatRep1ksTest3 <- ks.test(RPF[RPF$tvsMurSatRep1=="miR-181 target", "log2FoldChange"], RPF[RPF$tvsMurSatRep1=="Non-target", "log2FoldChange"])

TEtvsMurSatRep1ksTest1 <- ks.test(TE[TE$tvsMurSatRep1=="MurSatRep1", "log2FoldChange"], TE[TE$tvsMurSatRep1=="Non-target", "log2FoldChange"])
TEtvsMurSatRep1ksTest2 <- ks.test(TE[TE$tvsMurSatRep1=="both", "log2FoldChange"], TE[TE$tvsMurSatRep1=="Non-target", "log2FoldChange"])
TEtvsMurSatRep1ksTest3 <- ks.test(TE[TE$tvsMurSatRep1=="miR-181 target", "log2FoldChange"], TE[TE$tvsMurSatRep1=="Non-target", "log2FoldChange"])



tvsMurSatRep1sig <- data.frame(c("RNA", "RNA", "RNA", "RPF", "RPF", "RPF", "TE", "TE", "TE" 
                             ),
                        c("MurSatRep1", "both", "miR-181 target", "MurSatRep1", "both", "miR-181 target",
                          "MurSatRep1", "both", "miR-181 target"
                          ),
                        c(RNAtvsMurSatRep1TTest1$p.value, RNAtvsMurSatRep1TTest2$p.value, RNAtvsMurSatRep1TTest3$p.value,
                          RPFtvsMurSatRep1TTest1$p.value, RPFtvsMurSatRep1TTest2$p.value, RPFtvsMurSatRep1TTest3$p.value,
                          TEtvsMurSatRep1TTest1$p.value, TEtvsMurSatRep1TTest2$p.value, TEtvsMurSatRep1TTest3$p.value
                          ),
                        c(RNAtvsMurSatRep1ksTest1$p.value, RNAtvsMurSatRep1ksTest2$p.value, RNAtvsMurSatRep1ksTest3$p.value,
                          RPFtvsMurSatRep1ksTest1$p.value, RPFtvsMurSatRep1ksTest2$p.value, RPFtvsMurSatRep1ksTest3$p.value, 
                          TEtvsMurSatRep1ksTest1$p.value, TEtvsMurSatRep1ksTest2$p.value, TEtvsMurSatRep1ksTest3$p.value
                          ))
colnames(tvsMurSatRep1sig) <- c("Experiment", "Subset", "Welch Two Sample t-test", "Asymptotic two-sample Kolmogorov-Smirnov test")
tvsMurSatRep1sig

#export
write_xlsx(tvsMurSatRep1sig, "tvsMurSatRep1significance.xlsx")
```


## metagene plot
```{r metageneplot}
MMSAT4$id <- "MMsat4"
MurSatRep1$id <- "MurSatRep1"

metaob <- c(MMSAT4, MurSatRep1)

metael <- metaGeneProfile(object = metaob, annotation = gff23path, group = "id")
metasat4 <- metaGeneProfile(object = MMSAT4, annotation = gff23path)
metamur <- metaGeneProfile(object = MurSatRep1, annotation = gff23path)

metasat4
metamur

metael


pdf("metagene_profile_both.pdf", width=2, height = 2)
metael
dev.off()



```



## venn diagram
```{r vennmmsat4}

vset <- list(mmsat4frame[!duplicated(mmsat4frame$gene_id) & mmsat4frame$gene_id %in% largetframe$geneID,"gene_id"], MurSatRep1frame[!duplicated(MurSatRep1frame$gene_id) & MurSatRep1frame$gene_id %in% largetframe$geneID,"gene_id"])
names(vset) <- c("MMSAT4", "MurSatRep1")

vplot <- plot(euler(vset, shape = "ellipse"), quantities = TRUE, main="Genes by contained motifes" , fills=c(farbe1, farbe2)) 


pdf("Venn_motifes.pdf", width=2, height = 2)
vplot
dev.off()

```


# histogram MREs per gene with repeat
```{r mresPerGeneRep}
histmremmsat4 <- as.data.frame(table(largetframe[largetframe$geneID %in% mmsat4frame$gene_id, "geneID"]))
dim(histmremmsat4)

mmsat4GMRElist <- as.data.frame(table(largetframe[largetframe$geneID %in% mmsat4frame$gene_id, "geneName"]))
colnames(mmsat4GMRElist) = c("Gene", "Number of MREs")
write.csv(mmsat4GMRElist, "C:/Users/nikit/Documents/Krueger_Lab/Publications/miR181_paper/Figure4/genelist_MMsat4withMRE.csv")

histmremmsat4P <- ggplot(histmremmsat4, aes(y=Freq)) + 
  geom_histogram(binwidth = 1, colour="black", fill="#B4B4B4") +
  theme_paper() +
  scale_x_continuous(expand = c(0,0)) +
  scale_y_reverse()

histmremmsat4P

pdf("MRE_histogram_mres_per_MMSAT4.pdf", width=2, height = 2)
histmremmsat4P
dev.off()

```




#ma plots with significnace and "both" dots mmsat4 and mursatrep1
```{r masig}
ylim <- 2
xlim <- 10^6
xmin <- 10


lfcsig <- 0
pvsig <- 0.05

#RNA
#make columns with identifiers
RNAdat <- RNA[RNA$baseMean >= xmin,]
RNAdat$ycutoff <- "in"
RNAdat$ycutoff[RNAdat$log2FoldChange > ylim] <- "high"
RNAdat$ycutoff[RNAdat$log2FoldChange < -ylim] <- "low"

RNAdat$xcutoff <- "in"
RNAdat$xcutoff[RNAdat$baseMean > xlim] <- "high"

#implement cutoff
RNAdat$log2FoldChange[RNAdat$log2FoldChange > ylim] <- ylim
RNAdat$log2FoldChange[RNAdat$log2FoldChange < -ylim] <- -ylim
RNAdat$baseMean[RNAdat$baseMean > xlim] <- xlim




#RPF
#make columns with identifiers
RPFdat <- RPF[RPF$baseMean >= xmin,]
RPFdat$ycutoff <- "in"
RPFdat$ycutoff[RPFdat$log2FoldChange > ylim] <- "high"
RPFdat$ycutoff[RPFdat$log2FoldChange < -ylim] <- "low"

RPFdat$xcutoff <- "in"
RPFdat$xcutoff[RPFdat$baseMean > xlim] <- "high"

#implement cutoff
RPFdat$log2FoldChange[RPFdat$log2FoldChange > ylim] <- ylim
RPFdat$log2FoldChange[RPFdat$log2FoldChange < -ylim] <- -ylim
RPFdat$baseMean[RPFdat$baseMean > xlim] <- xlim

#RNA MMsat4
MAmmsat4RNAsig <- ggplot(RNAdat[!(RNAdat$xcutoff == "high"),], aes(y=log2FoldChange, x=log10(baseMean))) + geom_point_rast(shape=21, size = 0.5, 
     aes(fill = factor(LFCandPADJSig), colour=factor(LFCandPADJSig))) +
  scale_fill_manual(values = c(farbeneg, RNAncol, RNApcol)) + scale_colour_manual(values = c(farbeneg, "black", "black")) +
  geom_point(data=RNAdat[RNAdat$LFCandPADJSig == "Significant down" & RNAdat$ycutoff =="in" & RNAdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=21, size = 1, colour="black", fill=RNAncol) +
  geom_point(data=RNAdat[RNAdat$LFCandPADJSig == "Significant up" & RNAdat$ycutoff =="in" & RNAdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=21, size = 1, colour="black", fill=RNApcol) +
  geom_point(data=RNAdat[RNAdat$LFCandPADJSig == "Significant up" & RNAdat$ycutoff =="high" & RNAdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=24, size = 1, colour="black", fill=RNApcol) +
  geom_point(data=RNAdat[RNAdat$LFCandPADJSig == "Significant down" & RNAdat$ycutoff =="low" & RNAdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=25, size = 1, colour="black", fill=RNAncol) +
  geom_point(data=RNAdat[RNAdat$tvsmmsat4 == "both" & RNAdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=21, size = 1, colour="black", fill=RPFncol) +
  theme_paper() +
  geom_point(data=RNAdat[RNAdat$xcutoff == "high",], aes(y=log2FoldChange, x=log10(baseMean)), shape="\u25BA", size = 3, colour=farbeneg) +
  geom_hline(yintercept = 0, linetype="solid", colour="black") + 
  geom_hline(yintercept = lfcsig, linetype="dashed", colour="black") + 
  geom_hline(yintercept = -lfcsig, linetype="dashed", colour="black") + 
  scale_y_continuous("Fold change miR-181a/b-1 KO/WT (log2)", limits = c(-ylim, ylim)) + scale_x_continuous("Base mean expression ratio (log10)", limits = c(log10(xmin), log10(xlim))) +
  guides(color = guide_legend(override.aes = list(size = 3))) + 
  ggtitle("RNA MMsat4")

MAmmsat4RNAsig

pdf("MAplot_RNA_mmsat4.pdf", width = 1.8, height = 2)
MAmmsat4RNAsig
dev.off()

#RNA Mursatrep1
Mursatrep1RNAsig <- ggplot(RNAdat[!(RNAdat$xcutoff == "high"),], aes(y=log2FoldChange, x=log10(baseMean))) + geom_point_rast(shape=21, size = 0.5, 
     aes(fill = factor(LFCandPADJSig), colour=factor(LFCandPADJSig))) +
  scale_fill_manual(values = c(farbeneg, RNAncol, RNApcol)) + scale_colour_manual(values = c(farbeneg, "black", "black")) +
  geom_point(data=RNAdat[RNAdat$LFCandPADJSig == "Significant down" & RNAdat$ycutoff =="in" & RNAdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=21, size = 1, colour="black", fill=RNAncol) +
  geom_point(data=RNAdat[RNAdat$LFCandPADJSig == "Significant up" & RNAdat$ycutoff =="in" & RNAdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=21, size = 1, colour="black", fill=RNApcol) +
  geom_point(data=RNAdat[RNAdat$LFCandPADJSig == "Significant up" & RNAdat$ycutoff =="high" & RNAdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=24, size = 1, colour="black", fill=RNApcol) +
  geom_point(data=RNAdat[RNAdat$LFCandPADJSig == "Significant down" & RNAdat$ycutoff =="low" & RNAdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=25, size = 1, colour="black", fill=RNAncol) +
  geom_point(data=RNAdat[RNAdat$tvsMurSatRep1 == "both" & RNAdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=21, size = 1, colour="black", fill=RPFncol) +
  theme_paper() +
  geom_point(data=RNAdat[RNAdat$xcutoff == "high",], aes(y=log2FoldChange, x=log10(baseMean)), shape="\u25BA", size = 3, colour=farbeneg) +
  geom_hline(yintercept = 0, linetype="solid", colour="black") + 
  geom_hline(yintercept = lfcsig, linetype="dashed", colour="black") + 
  geom_hline(yintercept = -lfcsig, linetype="dashed", colour="black") + 
  scale_y_continuous("Fold change miR-181a/b-1 KO/WT (log2)", limits = c(-ylim, ylim)) + scale_x_continuous("Base mean expression ratio (log10)", limits = c(log10(xmin), log10(xlim))) +
  guides(color = guide_legend(override.aes = list(size = 3)))+ 
  ggtitle("RNA Mursatrep1")

Mursatrep1RNAsig

pdf("MAplot_RNA_Mursatrep1.pdf", width = 1.8, height = 2)
Mursatrep1RNAsig
dev.off()

#RPF MMsat4
MAmmsat4RPFsig <- ggplot(RPFdat[!(RPFdat$xcutoff == "high"),], aes(y=log2FoldChange, x=log10(baseMean))) + geom_point_rast(shape=21, size = 0.5, 
     aes(fill = factor(LFCandPADJSig), colour=factor(LFCandPADJSig))) +
  scale_fill_manual(values = c(farbeneg, RNAncol, RNApcol)) + scale_colour_manual(values = c(farbeneg, "black", "black")) +
  geom_point(data=RPFdat[RPFdat$LFCandPADJSig == "Significant down" & RPFdat$ycutoff =="in" & RPFdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=21, size = 1, colour="black", fill=RNAncol) +
  geom_point(data=RPFdat[RPFdat$LFCandPADJSig == "Significant up" & RPFdat$ycutoff =="in" & RPFdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=21, size = 1, colour="black", fill=RNApcol) +
  geom_point(data=RPFdat[RPFdat$LFCandPADJSig == "Significant up" & RPFdat$ycutoff =="high" & RPFdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=24, size = 1, colour="black", fill=RNApcol) +
  geom_point(data=RPFdat[RPFdat$LFCandPADJSig == "Significant down" & RPFdat$ycutoff =="low" & RPFdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=25, size = 1, colour="black", fill=RNAncol) +
  geom_point(data=RPFdat[RPFdat$tvsmmsat4 == "both" & RPFdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=21, size = 1, colour="black", fill=RPFncol) +
  theme_paper() +
  geom_point(data=RPFdat[RPFdat$xcutoff == "high",], aes(y=log2FoldChange, x=log10(baseMean)), shape="\u25BA", size = 3, colour=farbeneg) +
  geom_hline(yintercept = 0, linetype="solid", colour="black") + 
  geom_hline(yintercept = lfcsig, linetype="dashed", colour="black") + 
  geom_hline(yintercept = -lfcsig, linetype="dashed", colour="black") + 
  scale_y_continuous("Fold change miR-181a/b-1 KO/WT (log2)", limits = c(-ylim, ylim)) + scale_x_continuous("Base mean expression ratio (log10)", limits = c(log10(xmin), log10(xlim))) +
  guides(color = guide_legend(override.aes = list(size = 3)))+ 
  ggtitle("RPF MMsat4")

MAmmsat4RPFsig

pdf("MAplot_RPF_mmsat4.pdf", width = 1.8, height = 2)
MAmmsat4RPFsig
dev.off()

#RPF Mursatrep1
Mursatrep1RPFsig <- ggplot(RPFdat[!(RPFdat$xcutoff == "high"),], aes(y=log2FoldChange, x=log10(baseMean))) + geom_point_rast(shape=21, size = 0.5, 
     aes(fill = factor(LFCandPADJSig), colour=factor(LFCandPADJSig))) +
  scale_fill_manual(values = c(farbeneg, RNAncol, RNApcol)) + scale_colour_manual(values = c(farbeneg, "black", "black")) +
  geom_point(data=RPFdat[RPFdat$LFCandPADJSig == "Significant down" & RPFdat$ycutoff =="in" & RPFdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=21, size = 1, colour="black", fill=RNAncol) +
  geom_point(data=RPFdat[RPFdat$LFCandPADJSig == "Significant up" & RPFdat$ycutoff =="in" & RPFdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=21, size = 1, colour="black", fill=RNApcol) +
  geom_point(data=RPFdat[RPFdat$LFCandPADJSig == "Significant up" & RPFdat$ycutoff =="high" & RPFdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=24, size = 1, colour="black", fill=RNApcol) +
  geom_point(data=RPFdat[RPFdat$LFCandPADJSig == "Significant down" & RPFdat$ycutoff =="low" & RPFdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=25, size = 1, colour="black", fill=RNAncol) +
  geom_point(data=RPFdat[RPFdat$tvsMurSatRep1 == "both" & RPFdat$xcutoff == "in",], aes(y=log2FoldChange, x=log10(baseMean)), shape=21, size = 1, colour="black", fill=RPFncol) +
  theme_paper() +
  geom_point(data=RPFdat[RPFdat$xcutoff == "high",], aes(y=log2FoldChange, x=log10(baseMean)), shape="\u25BA", size = 3, colour=farbeneg) +
  geom_hline(yintercept = 0, linetype="solid", colour="black") + 
  geom_hline(yintercept = lfcsig, linetype="dashed", colour="black") + 
  geom_hline(yintercept = -lfcsig, linetype="dashed", colour="black") + 
  scale_y_continuous("Fold change miR-181a/b-1 KO/WT (log2)", limits = c(-ylim, ylim)) + scale_x_continuous("Base mean expression ratio (log10)", limits = c(log10(xmin), log10(xlim))) +
  guides(color = guide_legend(override.aes = list(size = 3)))+ 
  ggtitle("RPF Mursatrep1")

Mursatrep1RPFsig

pdf("MAplot_RPF_Mursatrep1.pdf",width = 1.8, height = 2)
Mursatrep1RPFsig
dev.off()
```





# Session info
```{r sessioninfo}
sessionInfo()
```

